
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://monarch-initiative.github.io/dismech/elements/">
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../about/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.0">
    
    
      
        <title>Schema - dismech</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.618322db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../css/pandas-dataframe.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#disorder-mechanisms-knowledge-base-schema" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="dismech" class="md-header__button md-logo" aria-label="dismech" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            dismech
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Schema
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/monarch-initiative/dismech" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="dismech" class="md-nav__button md-logo" aria-label="dismech" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    dismech
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/monarch-initiative/dismech" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Schema
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Schema
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Classes
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Slots
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Enumerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Types
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      
        Subsets
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    About
  

    
  </span>
  
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Classes
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Slots
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Enumerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Types
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      
        Subsets
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="disorder-mechanisms-knowledge-base-schema">Disorder Mechanisms Knowledge Base Schema</h1>
<p>Disease Pathophysiology Knowledge Base Schema</p>
<p>URI: https://w3id.org/monarch-initiative/dismech</p>
<p>Name: dismech</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AgentLifeCycle/">AgentLifeCycle</a></td>
<td></td>
</tr>
<tr>
<td><a href="AgentLifeCycleStage/">AgentLifeCycleStage</a></td>
<td></td>
</tr>
<tr>
<td><a href="AnimalModel/">AnimalModel</a></td>
<td></td>
</tr>
<tr>
<td><a href="Any/">Any</a></td>
<td></td>
</tr>
<tr>
<td><a href="Assay/">Assay</a></td>
<td></td>
</tr>
<tr>
<td><a href="AssociationMetric/">AssociationMetric</a></td>
<td>Quantitative association metric and its uncertainty</td>
</tr>
<tr>
<td><a href="AssociationSignal/">AssociationSignal</a></td>
<td>An association signal from EHR, registry, or computational sources, optionall...</td>
</tr>
<tr>
<td><a href="AssociationStatistics/">AssociationStatistics</a></td>
<td>Statistical summary with evidence for an association signal</td>
</tr>
<tr>
<td><a href="Biochemical/">Biochemical</a></td>
<td></td>
</tr>
<tr>
<td><a href="CausalEdge/">CausalEdge</a></td>
<td>A reference to a downstream effect or consequence in a causal relationship</td>
</tr>
<tr>
<td><a href="ClassificationAssignment/">ClassificationAssignment</a></td>
<td>Base class for classification assignments with evidence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ChannelopathyAssignment/">ChannelopathyAssignment</a></td>
<td>Channelopathy organ system classification assignment</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="HarrisonsChapterAssignment/">HarrisonsChapterAssignment</a></td>
<td>Harrison's internal medicine chapter classification assignment</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ICDOMorphologyAssignment/">ICDOMorphologyAssignment</a></td>
<td>ICD-O morphology classification assignment for neoplastic diseases</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="IUISAssignment/">IUISAssignment</a></td>
<td>IUIS primary immunodeficiency classification assignment</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="LysosomalStorageAssignment/">LysosomalStorageAssignment</a></td>
<td>Lysosomal storage disease biochemical classification assignment</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="MechanisticNosologyAssignment/">MechanisticNosologyAssignment</a></td>
<td>Mechanistic/pathway-based disease classification assignment</td>
</tr>
<tr>
<td><a href="ClinicalTrial/">ClinicalTrial</a></td>
<td>A clinical trial relevant to treatment or research of a disease</td>
</tr>
<tr>
<td><a href="ComorbidityAssociation/">ComorbidityAssociation</a></td>
<td>An association between two conditions, including directionality, evidence, an...</td>
</tr>
<tr>
<td><a href="ComorbidityHypothesis/">ComorbidityHypothesis</a></td>
<td>Mechanistic hypothesis for a comorbidity association, with rich text and embe...</td>
</tr>
<tr>
<td><a href="ComputationalModel/">ComputationalModel</a></td>
<td>A computational or in-silico model relevant to understanding disease mechanis...</td>
</tr>
<tr>
<td><a href="CriteriaSet/">CriteriaSet</a></td>
<td>A named criteria grouping within a definition</td>
</tr>
<tr>
<td><a href="CurationEvent/">CurationEvent</a></td>
<td>A single curation event in the audit trail</td>
</tr>
<tr>
<td><a href="Dataset/">Dataset</a></td>
<td>A reference to a publicly available omics or phenotype dataset</td>
</tr>
<tr>
<td><a href="Definition/">Definition</a></td>
<td>A diagnostic or phenotype definition for the disease</td>
</tr>
<tr>
<td><a href="Demographics/">Demographics</a></td>
<td>Demographic stratification for an association signal</td>
</tr>
<tr>
<td><a href="Descriptor/">Descriptor</a></td>
<td>Base class for structured descriptors that allow a preferred term, optional d...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="AnatomicalEntityDescriptor/">AnatomicalEntityDescriptor</a></td>
<td>A descriptor for anatomical locations, bindable to UBERON</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="AssayDescriptor/">AssayDescriptor</a></td>
<td>A descriptor for assays, bindable to OBI</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="BiologicalProcessDescriptor/">BiologicalProcessDescriptor</a></td>
<td>A descriptor for biological processes, bindable to Gene Ontology (GO)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="BiomarkerDescriptor/">BiomarkerDescriptor</a></td>
<td>A descriptor for biomarkers, bindable to NCIT</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="CellTypeDescriptor/">CellTypeDescriptor</a></td>
<td>A descriptor for cell types, bindable to Cell Ontology (CL)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="CellularComponentDescriptor/">CellularComponentDescriptor</a></td>
<td>A descriptor for cellular components, bindable to GO cellular component</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ChemicalEntityDescriptor/">ChemicalEntityDescriptor</a></td>
<td>A descriptor for chemical entities, bindable to CHEBI</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ConditionDescriptor/">ConditionDescriptor</a></td>
<td>A descriptor for a condition or disease, optionally bound to MONDO</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="CriteriaItem/">CriteriaItem</a></td>
<td>A criterion element (clinical feature, test result, imaging requirement)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="DiseaseDescriptor/">DiseaseDescriptor</a></td>
<td>A descriptor for the focal disease, bindable to MONDO</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="EnvironmentDescriptor/">EnvironmentDescriptor</a></td>
<td>A descriptor for environmental contexts/settings, bindable to ENVO</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ExposureDescriptor/">ExposureDescriptor</a></td>
<td>A descriptor for exposure events, bindable to ECTO or XCO</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="GeneDescriptor/">GeneDescriptor</a></td>
<td>A descriptor for genes, bindable to HGNC or other gene databases</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="GeneProductDescriptor/">GeneProductDescriptor</a></td>
<td>A descriptor for gene products (proteins, fusion proteins, oncoproteins), bin...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="HistopathologyFindingDescriptor/">HistopathologyFindingDescriptor</a></td>
<td>A descriptor for histopathologic findings, bindable to NCIT Morphologic Findi...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="InheritanceDescriptor/">InheritanceDescriptor</a></td>
<td>A descriptor for inheritance patterns, bindable to HPO mode of inheritance te...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="LifeCycleStageDescriptor/">LifeCycleStageDescriptor</a></td>
<td>A descriptor for parasite life cycle stages, bindable to OPL</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="OrganismDescriptor/">OrganismDescriptor</a></td>
<td>A descriptor for organisms, bindable to NCBITaxon</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="HostDescriptor/">HostDescriptor</a></td>
<td>A descriptor for hosts in an infectious agent life cycle</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="PhenotypeDescriptor/">PhenotypeDescriptor</a></td>
<td>A descriptor for phenotypes, bindable to Human Phenotype Ontology (HP)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ProteinComplexDescriptor/">ProteinComplexDescriptor</a></td>
<td>A descriptor for protein complexes that gene products participate in, bindabl...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="RegimenDescriptor/">RegimenDescriptor</a></td>
<td>A descriptor for treatment regimens, bindable to NCIT</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="SampleTypeDescriptor/">SampleTypeDescriptor</a></td>
<td>A descriptor for biological sample types (tissue and/or cell type)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="TreatmentDescriptor/">TreatmentDescriptor</a></td>
<td>A descriptor for treatments/medical actions, bindable to Medical Action Ontol...</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="TriggerDescriptor/">TriggerDescriptor</a></td>
<td>A descriptor for triggers/causes</td>
</tr>
<tr>
<td><a href="Diagnosis/">Diagnosis</a></td>
<td></td>
</tr>
<tr>
<td><a href="DifferentialDiagnosis/">DifferentialDiagnosis</a></td>
<td>A disease or condition that presents similarly to the focal disease and must ...</td>
</tr>
<tr>
<td><a href="Disease/">Disease</a></td>
<td></td>
</tr>
<tr>
<td><a href="DiseaseClassifications/">DiseaseClassifications</a></td>
<td>Container for all classification assignments for a disease</td>
</tr>
<tr>
<td><a href="DiseaseCollection/">DiseaseCollection</a></td>
<td></td>
</tr>
<tr>
<td><a href="DiseaseMappings/">DiseaseMappings</a></td>
<td>Container for external identifier mappings for a disease</td>
</tr>
<tr>
<td><a href="Environmental/">Environmental</a></td>
<td>An environmental factor, exposure, or context relevant to disease</td>
</tr>
<tr>
<td><a href="EpidemiologyInfo/">EpidemiologyInfo</a></td>
<td></td>
</tr>
<tr>
<td><a href="EvidenceItem/">EvidenceItem</a></td>
<td></td>
</tr>
<tr>
<td><a href="Finding/">Finding</a></td>
<td>A key finding or claim extracted from a source (publication or dataset)</td>
</tr>
<tr>
<td><a href="FunctionalEffect/">FunctionalEffect</a></td>
<td></td>
</tr>
<tr>
<td><a href="Genetic/">Genetic</a></td>
<td></td>
</tr>
<tr>
<td><a href="GeneticContext/">GeneticContext</a></td>
<td>A structured description of a genetic context that modifies phenotype frequen...</td>
</tr>
<tr>
<td><a href="GOEnrichment/">GOEnrichment</a></td>
<td>GO enrichment results for an association signal</td>
</tr>
<tr>
<td><a href="GOEnrichmentTerm/">GOEnrichmentTerm</a></td>
<td>GO term enrichment result with statistical metrics</td>
</tr>
<tr>
<td><a href="HistopathologyFinding/">HistopathologyFinding</a></td>
<td>A histopathologic finding from microscopic examination of tissue</td>
</tr>
<tr>
<td><a href="InfectiousAgent/">InfectiousAgent</a></td>
<td></td>
</tr>
<tr>
<td><a href="Inheritance/">Inheritance</a></td>
<td></td>
</tr>
<tr>
<td><a href="MappingConsistency/">MappingConsistency</a></td>
<td>Consistency assertion for a mapping relative to another source</td>
</tr>
<tr>
<td><a href="Mechanism/">Mechanism</a></td>
<td></td>
</tr>
<tr>
<td><a href="MechanisticHypothesis/">MechanisticHypothesis</a></td>
<td>Disease-level hypothesis metadata used to organize downstream causal edges in...</td>
</tr>
<tr>
<td><a href="ModelingConsideration/">ModelingConsideration</a></td>
<td></td>
</tr>
<tr>
<td><a href="OnsetDescriptor/">OnsetDescriptor</a></td>
<td>Structured description of age of onset</td>
</tr>
<tr>
<td><a href="Pathophysiology/">Pathophysiology</a></td>
<td></td>
</tr>
<tr>
<td><a href="Phenotype/">Phenotype</a></td>
<td></td>
</tr>
<tr>
<td><a href="PhenotypeContext/">PhenotypeContext</a></td>
<td>A context-specific annotation qualifying how a phenotype manifests under part...</td>
</tr>
<tr>
<td><a href="Prevalence/">Prevalence</a></td>
<td></td>
</tr>
<tr>
<td><a href="ProgressionInfo/">ProgressionInfo</a></td>
<td></td>
</tr>
<tr>
<td><a href="PublicationReference/">PublicationReference</a></td>
<td>A reference to a publication with associated findings</td>
</tr>
<tr>
<td><a href="Qualifier/">Qualifier</a></td>
<td>A predicate-value pair for formal post-composition</td>
</tr>
<tr>
<td><a href="Stage/">Stage</a></td>
<td></td>
</tr>
<tr>
<td><a href="Subtype/">Subtype</a></td>
<td></td>
</tr>
<tr>
<td><a href="Term/">Term</a></td>
<td>A structured reference to an ontology term</td>
</tr>
<tr>
<td><a href="TermMapping/">TermMapping</a></td>
<td>Mapping from this disease entry to an external term or code</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ICD10CMMapping/">ICD10CMMapping</a></td>
<td>ICD-10-CM diagnosis code mapping</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ICD11FMapping/">ICD11FMapping</a></td>
<td>ICD-11 Foundation diagnosis code mapping</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="MondoMapping/">MondoMapping</a></td>
<td>MONDO disease ontology mapping</td>
</tr>
<tr>
<td><a href="Transmission/">Transmission</a></td>
<td></td>
</tr>
<tr>
<td><a href="Treatment/">Treatment</a></td>
<td></td>
</tr>
<tr>
<td><a href="TreatmentMechanismTarget/">TreatmentMechanismTarget</a></td>
<td>Links a treatment to a specific pathophysiology mechanism node it targets</td>
</tr>
<tr>
<td><a href="UpstreamConditionHypothesis/">UpstreamConditionHypothesis</a></td>
<td>Hypothesized upstream condition that may explain both A and B</td>
</tr>
<tr>
<td><a href="Variant/">Variant</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="a_before_b/">a_before_b</a></td>
<td>Probability or fraction of A before B in an EHR signal</td>
</tr>
<tr>
<td><a href="accession/">accession</a></td>
<td>Dataset accession identifier as a CURIE (e</td>
</tr>
<tr>
<td><a href="additional_requirements/">additional_requirements</a></td>
<td>Additional requirements used in a criteria set</td>
</tr>
<tr>
<td><a href="age_range/">age_range</a></td>
<td>Age range or stratification, if applicable</td>
</tr>
<tr>
<td><a href="agent_life_cycle/">agent_life_cycle</a></td>
<td></td>
</tr>
<tr>
<td><a href="allele_type/">allele_type</a></td>
<td>Type of allele or mutation (e</td>
</tr>
<tr>
<td><a href="alleles/">alleles</a></td>
<td></td>
</tr>
<tr>
<td><a href="animal_models/">animal_models</a></td>
<td></td>
</tr>
<tr>
<td><a href="applies_to_subtypes/">applies_to_subtypes</a></td>
<td>Disease subtypes for which this hypothesis is intended to apply</td>
</tr>
<tr>
<td><a href="assays/">assays</a></td>
<td></td>
</tr>
<tr>
<td><a href="associated_phenotypes/">associated_phenotypes</a></td>
<td></td>
</tr>
<tr>
<td><a href="association/">association</a></td>
<td></td>
</tr>
<tr>
<td><a href="association_signals/">association_signals</a></td>
<td>Association signals from EHR, registry, or computational sources</td>
</tr>
<tr>
<td><a href="b_before_a/">b_before_a</a></td>
<td>Probability or fraction of B before A in an EHR signal</td>
</tr>
<tr>
<td><a href="background/">background</a></td>
<td></td>
</tr>
<tr>
<td><a href="base_model/">base_model</a></td>
<td>Parent/base model this is derived from (e</td>
</tr>
<tr>
<td><a href="biochemical/">biochemical</a></td>
<td></td>
</tr>
<tr>
<td><a href="biological_processes/">biological_processes</a></td>
<td></td>
</tr>
<tr>
<td><a href="biomarker_term/">biomarker_term</a></td>
<td>Ontology term for a biomarker (from NCIT)</td>
</tr>
<tr>
<td><a href="categories/">categories</a></td>
<td></td>
</tr>
<tr>
<td><a href="category/">category</a></td>
<td></td>
</tr>
<tr>
<td><a href="causal_link_type/">causal_link_type</a></td>
<td>Whether this edge is direct or indirect, and whether omitted intermediates ar...</td>
</tr>
<tr>
<td><a href="cell_type_term/">cell_type_term</a></td>
<td>CL term for the cell type</td>
</tr>
<tr>
<td><a href="cell_types/">cell_types</a></td>
<td></td>
</tr>
<tr>
<td><a href="cellular_components/">cellular_components</a></td>
<td></td>
</tr>
<tr>
<td><a href="channelopathy_category/">channelopathy_category</a></td>
<td>Channelopathy organ system classification</td>
</tr>
<tr>
<td><a href="chemical_entities/">chemical_entities</a></td>
<td></td>
</tr>
<tr>
<td><a href="chemicals/">chemicals</a></td>
<td></td>
</tr>
<tr>
<td><a href="children/">children</a></td>
<td>Names of other subtypes in this list that are members/children of this groupi...</td>
</tr>
<tr>
<td><a href="classification/">classification</a></td>
<td>Classification scheme this subtype belongs to (e</td>
</tr>
<tr>
<td><a href="classification_value/">classification_value</a></td>
<td>The classification value assigned</td>
</tr>
<tr>
<td><a href="classifications/">classifications</a></td>
<td>Classification assignments for this disease from various nosologies</td>
</tr>
<tr>
<td><a href="clinical_significance/">clinical_significance</a></td>
<td></td>
</tr>
<tr>
<td><a href="clinical_trials/">clinical_trials</a></td>
<td>Clinical trials relevant to disease treatment and research</td>
</tr>
<tr>
<td><a href="code/">code</a></td>
<td>Code within the specified code system</td>
</tr>
<tr>
<td><a href="code_system/">code_system</a></td>
<td>Code system used for a condition reference (e</td>
</tr>
<tr>
<td><a href="combined_score/">combined_score</a></td>
<td>Combined score used by an enrichment method</td>
</tr>
<tr>
<td><a href="complementation_group/">complementation_group</a></td>
<td>Complementation group designation (e</td>
</tr>
<tr>
<td><a href="components/">components</a></td>
<td>Component conditions that make up a composite descriptor</td>
</tr>
<tr>
<td><a href="composition/">composition</a></td>
<td>Composition type for a composite condition descriptor</td>
</tr>
<tr>
<td><a href="computational_models/">computational_models</a></td>
<td>Computational models (metabolic, mechanistic, ML, digital twins) for this dis...</td>
</tr>
<tr>
<td><a href="conditions/">conditions</a></td>
<td>Experimental conditions or disease states represented</td>
</tr>
<tr>
<td><a href="consequence/">consequence</a></td>
<td></td>
</tr>
<tr>
<td><a href="consequences/">consequences</a></td>
<td></td>
</tr>
<tr>
<td><a href="consistency/">consistency</a></td>
<td>Consistency assertions for this mapping against other sources</td>
</tr>
<tr>
<td><a href="consistent/">consistent</a></td>
<td>Consistency status for a mapping relative to a reference source</td>
</tr>
<tr>
<td><a href="context/">context</a></td>
<td></td>
</tr>
<tr>
<td><a href="core_clinical_characteristics/">core_clinical_characteristics</a></td>
<td>Core clinical characteristics used in a criteria set</td>
</tr>
<tr>
<td><a href="creation_date/">creation_date</a></td>
<td>ISO 8601/RFC 3339 timestamp string for when this disease entry was first crea...</td>
</tr>
<tr>
<td><a href="criteria_sets/">criteria_sets</a></td>
<td>Named criteria groupings within a definition</td>
</tr>
<tr>
<td><a href="curation_action/">curation_action</a></td>
<td>Type of curation action performed</td>
</tr>
<tr>
<td><a href="curation_description/">curation_description</a></td>
<td>Human-readable description of changes made</td>
</tr>
<tr>
<td><a href="curation_history/">curation_history</a></td>
<td>Audit trail of AI-assisted curation events</td>
</tr>
<tr>
<td><a href="curation_model/">curation_model</a></td>
<td>Model identifier used for curation (e</td>
</tr>
<tr>
<td><a href="curation_status/">curation_status</a></td>
<td>Curation workflow status</td>
</tr>
<tr>
<td><a href="curation_timestamp/">curation_timestamp</a></td>
<td>ISO 8601 timestamp of the curation event</td>
</tr>
<tr>
<td><a href="data_type/">data_type</a></td>
<td>The type of omics or other data in the dataset</td>
</tr>
<tr>
<td><a href="datasets/">datasets</a></td>
<td>Publicly available datasets relevant to disease research</td>
</tr>
<tr>
<td><a href="de_novo_rate/">de_novo_rate</a></td>
<td>Estimated percentage of de novo cases (e</td>
</tr>
<tr>
<td><a href="definition_type/">definition_type</a></td>
<td>The type of definition or criteria set</td>
</tr>
<tr>
<td><a href="definitions/">definitions</a></td>
<td>Definitions or diagnostic criteria for this disease</td>
</tr>
<tr>
<td><a href="demographics/">demographics</a></td>
<td>Demographic stratification for an association signal</td>
</tr>
<tr>
<td><a href="description/">description</a></td>
<td></td>
</tr>
<tr>
<td><a href="diagnosis/">diagnosis</a></td>
<td></td>
</tr>
<tr>
<td><a href="diagnosis_term/">diagnosis_term</a></td>
<td>The MAXO term for this diagnostic procedure</td>
</tr>
<tr>
<td><a href="diagnostic/">diagnostic</a></td>
<td></td>
</tr>
<tr>
<td><a href="differential_diagnoses/">differential_diagnoses</a></td>
<td>Differential diagnoses - similar diseases that must be ruled out</td>
</tr>
<tr>
<td><a href="directionality/">directionality</a></td>
<td>Direction of a comorbidity/trajectory association</td>
</tr>
<tr>
<td><a href="disease_a/">disease_a</a></td>
<td>First disease in a comorbidity pair</td>
</tr>
<tr>
<td><a href="disease_b/">disease_b</a></td>
<td>Second disease in a comorbidity pair</td>
</tr>
<tr>
<td><a href="disease_term/">disease_term</a></td>
<td>The MONDO disease term for this disease</td>
</tr>
<tr>
<td><a href="diseases/">diseases</a></td>
<td></td>
</tr>
<tr>
<td><a href="distinguishing_features/">distinguishing_features</a></td>
<td>Key clinical, laboratory, imaging, or epidemiological features that help diff...</td>
</tr>
<tr>
<td><a href="downstream/">downstream</a></td>
<td></td>
</tr>
<tr>
<td><a href="duration/">duration</a></td>
<td></td>
</tr>
<tr>
<td><a href="duration_days/">duration_days</a></td>
<td></td>
</tr>
<tr>
<td><a href="effect/">effect</a></td>
<td></td>
</tr>
<tr>
<td><a href="environment_context/">environment_context</a></td>
<td>The ENVO term for the environmental context/setting</td>
</tr>
<tr>
<td><a href="environmental/">environmental</a></td>
<td></td>
</tr>
<tr>
<td><a href="epidemiology/">epidemiology</a></td>
<td></td>
</tr>
<tr>
<td><a href="evidence/">evidence</a></td>
<td></td>
</tr>
<tr>
<td><a href="evidence_source/">evidence_source</a></td>
<td>Origin of the evidence item (human clinical, model organism, in vitro, or com...</td>
</tr>
<tr>
<td><a href="examples/">examples</a></td>
<td></td>
</tr>
<tr>
<td><a href="exclusion_criteria/">exclusion_criteria</a></td>
<td>Exclusion criteria for a definition or criteria set</td>
</tr>
<tr>
<td><a href="explanation/">explanation</a></td>
<td></td>
</tr>
<tr>
<td><a href="exposure_term/">exposure_term</a></td>
<td>The ECTO/XCO term for this exposure event</td>
</tr>
<tr>
<td><a href="exposures/">exposures</a></td>
<td>Environmental exposures studied in the dataset</td>
</tr>
<tr>
<td><a href="expressivity/">expressivity</a></td>
<td>Expressivity classification for this inheritance pattern</td>
</tr>
<tr>
<td><a href="factors/">factors</a></td>
<td></td>
</tr>
<tr>
<td><a href="fdr/">fdr</a></td>
<td>FDR-adjusted p-value for an association or enrichment</td>
</tr>
<tr>
<td><a href="features/">features</a></td>
<td></td>
</tr>
<tr>
<td><a href="finding_term/">finding_term</a></td>
<td>Ontology term for a histopathologic finding (from NCIT or HP)</td>
</tr>
<tr>
<td><a href="findings/">findings</a></td>
<td>Key findings or claims extracted from this source (publication or dataset)</td>
</tr>
<tr>
<td><a href="frequency/">frequency</a></td>
<td></td>
</tr>
<tr>
<td><a href="function/">function</a></td>
<td></td>
</tr>
<tr>
<td><a href="functional_effects/">functional_effects</a></td>
<td></td>
</tr>
<tr>
<td><a href="functional_impact/">functional_impact</a></td>
<td>Functional consequence of the genetic variant (e</td>
</tr>
<tr>
<td><a href="gene/">gene</a></td>
<td></td>
</tr>
<tr>
<td><a href="gene_products/">gene_products</a></td>
<td>Gene products (proteins, fusion proteins, oncoproteins) involved in this path...</td>
</tr>
<tr>
<td><a href="gene_term/">gene_term</a></td>
<td>The HGNC term for this gene</td>
</tr>
<tr>
<td><a href="genes/">genes</a></td>
<td></td>
</tr>
<tr>
<td><a href="genetic/">genetic</a></td>
<td></td>
</tr>
<tr>
<td><a href="genetic_context/">genetic_context</a></td>
<td>The genetic context under which this qualification applies</td>
</tr>
<tr>
<td><a href="genotype/">genotype</a></td>
<td></td>
</tr>
<tr>
<td><a href="geography/">geography</a></td>
<td></td>
</tr>
<tr>
<td><a href="go_enrichment/">go_enrichment</a></td>
<td>GO enrichment results associated with a signal</td>
</tr>
<tr>
<td><a href="go_terms/">go_terms</a></td>
<td>GO term enrichment results</td>
</tr>
<tr>
<td><a href="harrisons_chapter/">harrisons_chapter</a></td>
<td>Harrison's internal medicine chapter classification</td>
</tr>
<tr>
<td><a href="has_subtypes/">has_subtypes</a></td>
<td></td>
</tr>
<tr>
<td><a href="histopathology/">histopathology</a></td>
<td>Histopathologic findings including microscopic morphology, architectural patt...</td>
</tr>
<tr>
<td><a href="hosts/">hosts</a></td>
<td></td>
</tr>
<tr>
<td><a href="hypotheses/">hypotheses</a></td>
<td>Mechanistic or causal hypotheses about the association</td>
</tr>
<tr>
<td><a href="hypothesis_group_id/">hypothesis_group_id</a></td>
<td>Stable identifier for a disease-level mechanistic hypothesis grouping</td>
</tr>
<tr>
<td><a href="hypothesis_groups/">hypothesis_groups</a></td>
<td>Hypothesis identifiers or grouping labels that this edge belongs to</td>
</tr>
<tr>
<td><a href="hypothesis_label/">hypothesis_label</a></td>
<td>Human-readable label/title for a mechanistic hypothesis</td>
</tr>
<tr>
<td><a href="icd10cm_mappings/">icd10cm_mappings</a></td>
<td>ICD-10-CM code mappings for this disease</td>
</tr>
<tr>
<td><a href="icd11f_mappings/">icd11f_mappings</a></td>
<td>ICD-11 Foundation code mappings for this disease</td>
</tr>
<tr>
<td><a href="icdo_morphology/">icdo_morphology</a></td>
<td>ICD-O morphology classification (for neoplastic diseases)</td>
</tr>
<tr>
<td><a href="id/">id</a></td>
<td>Ontology term identifier (CURIE)</td>
</tr>
<tr>
<td><a href="identifiers/">identifiers</a></td>
<td></td>
</tr>
<tr>
<td><a href="imaging_requirements/">imaging_requirements</a></td>
<td>Imaging requirements used in a criteria set</td>
</tr>
<tr>
<td><a href="inclusion_criteria/">inclusion_criteria</a></td>
<td>Inclusion criteria for a definition or criteria set</td>
</tr>
<tr>
<td><a href="incubation_days/">incubation_days</a></td>
<td></td>
</tr>
<tr>
<td><a href="incubation_years/">incubation_years</a></td>
<td></td>
</tr>
<tr>
<td><a href="infectious_agent/">infectious_agent</a></td>
<td></td>
</tr>
<tr>
<td><a href="infectious_agent_term/">infectious_agent_term</a></td>
<td>The NCBITaxon term for this infectious agent</td>
</tr>
<tr>
<td><a href="inheritance/">inheritance</a></td>
<td></td>
</tr>
<tr>
<td><a href="inheritance_term/">inheritance_term</a></td>
<td>The HPO mode of inheritance term for this inheritance pattern</td>
</tr>
<tr>
<td><a href="intermediate_mechanisms/">intermediate_mechanisms</a></td>
<td>Known or suspected intermediate mechanisms omitted from this edge for graph c...</td>
</tr>
<tr>
<td><a href="iuis_category/">iuis_category</a></td>
<td>IUIS primary immunodeficiency classification</td>
</tr>
<tr>
<td><a href="label/">label</a></td>
<td>Human-readable label for the ontology term</td>
</tr>
<tr>
<td><a href="laboratory_requirements/">laboratory_requirements</a></td>
<td>Laboratory or serologic requirements used in a criteria set</td>
</tr>
<tr>
<td><a href="laterality/">laterality</a></td>
<td>Laterality qualifier (left, right, or bilateral)</td>
</tr>
<tr>
<td><a href="life_cycle_stage_term/">life_cycle_stage_term</a></td>
<td>The OPL term for this agent life cycle stage</td>
</tr>
<tr>
<td><a href="life_cycle_stages/">life_cycle_stages</a></td>
<td></td>
</tr>
<tr>
<td><a href="literature_evidence/">literature_evidence</a></td>
<td>Literature-based evidence items for this association</td>
</tr>
<tr>
<td><a href="located_in/">located_in</a></td>
<td>Anatomical location where this entity/process occurs or procedure is performe...</td>
</tr>
<tr>
<td><a href="locations/">locations</a></td>
<td></td>
</tr>
<tr>
<td><a href="lysosomal_storage_category/">lysosomal_storage_category</a></td>
<td>Lysosomal storage disease biochemical classification</td>
</tr>
<tr>
<td><a href="mapping_justification/">mapping_justification</a></td>
<td>Brief rationale or justification for the mapping</td>
</tr>
<tr>
<td><a href="mapping_notes/">mapping_notes</a></td>
<td>Notes on code-to-concept mapping decisions for this signal</td>
</tr>
<tr>
<td><a href="mapping_predicate/">mapping_predicate</a></td>
<td>Relationship between this disease and the mapped term (e</td>
</tr>
<tr>
<td><a href="mapping_source/">mapping_source</a></td>
<td>Source of the mapping (e</td>
</tr>
<tr>
<td><a href="mappings/">mappings</a></td>
<td>External identifier mappings for this disease (SSSOM-inspired)</td>
</tr>
<tr>
<td><a href="markers/">markers</a></td>
<td></td>
</tr>
<tr>
<td><a href="max_age_years/">max_age_years</a></td>
<td>Maximum (latest) age of onset in years</td>
</tr>
<tr>
<td><a href="maximum_value/">maximum_value</a></td>
<td></td>
</tr>
<tr>
<td><a href="mean_age_years/">mean_age_years</a></td>
<td>Mean age of onset in years, as reported in a cohort study</td>
</tr>
<tr>
<td><a href="mean_range/">mean_range</a></td>
<td></td>
</tr>
<tr>
<td><a href="mechanism/">mechanism</a></td>
<td></td>
</tr>
<tr>
<td><a href="mechanisms/">mechanisms</a></td>
<td></td>
</tr>
<tr>
<td><a href="mechanistic_category/">mechanistic_category</a></td>
<td>Mechanistic/pathway-based disease classification</td>
</tr>
<tr>
<td><a href="mechanistic_hypotheses/">mechanistic_hypotheses</a></td>
<td>Disease-level mechanistic hypotheses that group and annotate causal edges</td>
</tr>
<tr>
<td><a href="method/">method</a></td>
<td>Method or pipeline name</td>
</tr>
<tr>
<td><a href="metric_ci_lower/">metric_ci_lower</a></td>
<td>Lower confidence interval bound</td>
</tr>
<tr>
<td><a href="metric_ci_upper/">metric_ci_upper</a></td>
<td>Upper confidence interval bound</td>
</tr>
<tr>
<td><a href="metric_type/">metric_type</a></td>
<td>Metric type (e</td>
</tr>
<tr>
<td><a href="metric_value/">metric_value</a></td>
<td>Metric value</td>
</tr>
<tr>
<td><a href="metrics/">metrics</a></td>
<td>Quantitative association metrics</td>
</tr>
<tr>
<td><a href="min_age_years/">min_age_years</a></td>
<td>Minimum (earliest) age of onset in years</td>
</tr>
<tr>
<td><a href="minimum_required/">minimum_required</a></td>
<td>Minimum number of criteria required in this criteria set</td>
</tr>
<tr>
<td><a href="minimum_value/">minimum_value</a></td>
<td></td>
</tr>
<tr>
<td><a href="model_format/">model_format</a></td>
<td>File format (e</td>
</tr>
<tr>
<td><a href="model_id/">model_id</a></td>
<td>Identifier within the repository (e</td>
</tr>
<tr>
<td><a href="model_software/">model_software</a></td>
<td>Software/toolbox for running the model (e</td>
</tr>
<tr>
<td><a href="model_type/">model_type</a></td>
<td>Type of computational model</td>
</tr>
<tr>
<td><a href="modeling_considerations/">modeling_considerations</a></td>
<td></td>
</tr>
<tr>
<td><a href="modifier/">modifier</a></td>
<td>Directional or qualitative modifier for a descriptor (e</td>
</tr>
<tr>
<td><a href="mondo_mappings/">mondo_mappings</a></td>
<td>MONDO disease ontology mappings for this disease</td>
</tr>
<tr>
<td><a href="name/">name</a></td>
<td></td>
</tr>
<tr>
<td><a href="notes/">notes</a></td>
<td></td>
</tr>
<tr>
<td><a href="onset/">onset</a></td>
<td>Structured age of onset descriptor</td>
</tr>
<tr>
<td><a href="onset_category/">onset_category</a></td>
<td>HPO onset category (e</td>
</tr>
<tr>
<td><a href="organism/">organism</a></td>
<td>The organism from which samples were derived</td>
</tr>
<tr>
<td><a href="overlap/">overlap</a></td>
<td>Overlap fraction for an enrichment term</td>
</tr>
<tr>
<td><a href="p_value/">p_value</a></td>
<td>P-value for an association or enrichment</td>
</tr>
<tr>
<td><a href="parent_of_origin_effect/">parent_of_origin_effect</a></td>
<td>Parent-of-origin effect or bias (e</td>
</tr>
<tr>
<td><a href="parents/">parents</a></td>
<td></td>
</tr>
<tr>
<td><a href="pathophysiology/">pathophysiology</a></td>
<td></td>
</tr>
<tr>
<td><a href="pathways/">pathways</a></td>
<td></td>
</tr>
<tr>
<td><a href="penetrance/">penetrance</a></td>
<td>Penetrance classification for this inheritance pattern</td>
</tr>
<tr>
<td><a href="penetrance_percentage/">penetrance_percentage</a></td>
<td>Estimated penetrance percentage or range (e</td>
</tr>
<tr>
<td><a href="percentage/">percentage</a></td>
<td></td>
</tr>
<tr>
<td><a href="perturbations/">perturbations</a></td>
<td>Gene knockouts, reaction deletions, or parameter changes modeling the disease</td>
</tr>
<tr>
<td><a href="phase/">phase</a></td>
<td></td>
</tr>
<tr>
<td><a href="phenotype_contexts/">phenotype_contexts</a></td>
<td>Context-specific qualifications of this phenotype's frequency, severity, or o...</td>
</tr>
<tr>
<td><a href="phenotype_term/">phenotype_term</a></td>
<td>The HP term for this phenotype</td>
</tr>
<tr>
<td><a href="phenotypes/">phenotypes</a></td>
<td></td>
</tr>
<tr>
<td><a href="platform/">platform</a></td>
<td>Sequencing or array platform used</td>
</tr>
<tr>
<td><a href="population/">population</a></td>
<td>Population or cohort description (e</td>
</tr>
<tr>
<td><a href="predicate/">predicate</a></td>
<td>The relationship/predicate in a qualifier (e</td>
</tr>
<tr>
<td><a href="preferred_term/">preferred_term</a></td>
<td>The preferred human-readable term for this descriptor</td>
</tr>
<tr>
<td><a href="presence/">presence</a></td>
<td></td>
</tr>
<tr>
<td><a href="prevalence/">prevalence</a></td>
<td></td>
</tr>
<tr>
<td><a href="progression/">progression</a></td>
<td></td>
</tr>
<tr>
<td><a href="protein_complexes/">protein_complexes</a></td>
<td>Protein complexes that gene products participate in</td>
</tr>
<tr>
<td><a href="publication/">publication</a></td>
<td>Associated publication (PMID)</td>
</tr>
<tr>
<td><a href="qualifiers/">qualifiers</a></td>
<td>List of predicate-value pairs for formal post-composition</td>
</tr>
<tr>
<td><a href="reference/">reference</a></td>
<td>The authoritative reference (publication) for this evidence item</td>
</tr>
<tr>
<td><a href="references/">references</a></td>
<td>Top-level list of references with their key findings for this disease</td>
</tr>
<tr>
<td><a href="regimen_term/">regimen_term</a></td>
<td>The NCIT term for this treatment regimen</td>
</tr>
<tr>
<td><a href="repository_url/">repository_url</a></td>
<td>URL to model repository (GitHub, BiGG, VMH, BioModels)</td>
</tr>
<tr>
<td><a href="results/">results</a></td>
<td></td>
</tr>
<tr>
<td><a href="review_notes/">review_notes</a></td>
<td></td>
</tr>
<tr>
<td><a href="role/">role</a></td>
<td></td>
</tr>
<tr>
<td><a href="same_time/">same_time</a></td>
<td>Probability or fraction of A and B occurring in the same time window</td>
</tr>
<tr>
<td><a href="sample_count/">sample_count</a></td>
<td>Total number of samples in the dataset</td>
</tr>
<tr>
<td><a href="sample_types/">sample_types</a></td>
<td>Types of biological samples in the dataset</td>
</tr>
<tr>
<td><a href="scope/">scope</a></td>
<td>Scope or population for which the definition applies (e</td>
</tr>
<tr>
<td><a href="sequelae/">sequelae</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequence_length/">sequence_length</a></td>
<td></td>
</tr>
<tr>
<td><a href="severity/">severity</a></td>
<td></td>
</tr>
<tr>
<td><a href="sex/">sex</a></td>
<td>Sex-specific stratum, if applicable</td>
</tr>
<tr>
<td><a href="shared_upstream_hypotheses/">shared_upstream_hypotheses</a></td>
<td>Suspected upstream conditions that may explain both A and B</td>
</tr>
<tr>
<td><a href="signal_disorder_a_id/">signal_disorder_a_id</a></td>
<td>Original identifier for disorder A in this signal (CURIE, e</td>
</tr>
<tr>
<td><a href="signal_disorder_b_id/">signal_disorder_b_id</a></td>
<td>Original identifier for disorder B in this signal (CURIE, e</td>
</tr>
<tr>
<td><a href="slug/">slug</a></td>
<td>Canonical slug identifier (Title_Case_With_Underscores) for leaf conditions</td>
</tr>
<tr>
<td><a href="snippet/">snippet</a></td>
<td>An exact excerpt/quote from the referenced publication that supports or refut...</td>
</tr>
<tr>
<td><a href="source/">source</a></td>
<td>Source dataset or provenance label</td>
</tr>
<tr>
<td><a href="species/">species</a></td>
<td></td>
</tr>
<tr>
<td><a href="specificity/">specificity</a></td>
<td></td>
</tr>
<tr>
<td><a href="stages/">stages</a></td>
<td></td>
</tr>
<tr>
<td><a href="statement/">statement</a></td>
<td>A key finding or claim from the publication</td>
</tr>
<tr>
<td><a href="statistics/">statistics</a></td>
<td>Statistical summary for an association signal</td>
</tr>
<tr>
<td><a href="status/">status</a></td>
<td>Status or state of a clinical trial or other process</td>
</tr>
<tr>
<td><a href="substages/">substages</a></td>
<td></td>
</tr>
<tr>
<td><a href="subtype/">subtype</a></td>
<td></td>
</tr>
<tr>
<td><a href="subtype_frequency/">subtype_frequency</a></td>
<td>Frequency of this subtype among all cases of the parent disease (e</td>
</tr>
<tr>
<td><a href="subtype_term/">subtype_term</a></td>
<td>The MONDO term for a disease subtype</td>
</tr>
<tr>
<td><a href="subtypes/">subtypes</a></td>
<td></td>
</tr>
<tr>
<td><a href="supporting_text/">supporting_text</a></td>
<td>Exact excerpt/quote from the publication supporting the statement</td>
</tr>
<tr>
<td><a href="supports/">supports</a></td>
<td></td>
</tr>
<tr>
<td><a href="synonyms/">synonyms</a></td>
<td></td>
</tr>
<tr>
<td><a href="target/">target</a></td>
<td>The name of the target element in a causal relationship</td>
</tr>
<tr>
<td><a href="target_mechanisms/">target_mechanisms</a></td>
<td>Pathophysiology mechanism nodes that this treatment targets or modulates</td>
</tr>
<tr>
<td><a href="target_phenotypes/">target_phenotypes</a></td>
<td>Phenotypes that this treatment or trial addresses or targets</td>
</tr>
<tr>
<td><a href="term/">term</a></td>
<td>Optional structured ontology term reference</td>
</tr>
<tr>
<td><a href="therapeutic_agent/">therapeutic_agent</a></td>
<td>The drug, chemical, or therapeutic agent used in a treatment</td>
</tr>
<tr>
<td><a href="tissue_term/">tissue_term</a></td>
<td>UBERON term for the tissue</td>
</tr>
<tr>
<td><a href="title/">title</a></td>
<td>Title of the publication</td>
</tr>
<tr>
<td><a href="transmission/">transmission</a></td>
<td></td>
</tr>
<tr>
<td><a href="treatment_effect/">treatment_effect</a></td>
<td>How the treatment affects the targeted mechanism</td>
</tr>
<tr>
<td><a href="treatment_term/">treatment_term</a></td>
<td>The MAXO term for this treatment/medical action</td>
</tr>
<tr>
<td><a href="treatments/">treatments</a></td>
<td></td>
</tr>
<tr>
<td><a href="triggers/">triggers</a></td>
<td></td>
</tr>
<tr>
<td><a href="type/">type</a></td>
<td></td>
</tr>
<tr>
<td><a href="unit/">unit</a></td>
<td></td>
</tr>
<tr>
<td><a href="updated_date/">updated_date</a></td>
<td>ISO 8601/RFC 3339 timestamp string for when this disease entry was last updat...</td>
</tr>
<tr>
<td><a href="upstream_disorder/">upstream_disorder</a></td>
<td>Upstream disorder referenced in a hypothesis</td>
</tr>
<tr>
<td><a href="value/">value</a></td>
<td>The value/filler in a qualifier</td>
</tr>
<tr>
<td><a href="variants/">variants</a></td>
<td></td>
</tr>
<tr>
<td><a href="vectors/">vectors</a></td>
<td></td>
</tr>
<tr>
<td><a href="zygosity/">zygosity</a></td>
<td>Zygosity context</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AnatomicalEntityTerm/">AnatomicalEntityTerm</a></td>
<td>A term representing an anatomical entity</td>
</tr>
<tr>
<td><a href="AssayTerm/">AssayTerm</a></td>
<td>A term representing an assay</td>
</tr>
<tr>
<td><a href="AssociationMetricTypeEnum/">AssociationMetricTypeEnum</a></td>
<td>Type of association metric</td>
</tr>
<tr>
<td><a href="AssociationSignalMethodEnum/">AssociationSignalMethodEnum</a></td>
<td>Method used to derive an association signal</td>
</tr>
<tr>
<td><a href="AssociationSignalSourceEnum/">AssociationSignalSourceEnum</a></td>
<td>Source of an association signal</td>
</tr>
<tr>
<td><a href="BiologicalProcessTerm/">BiologicalProcessTerm</a></td>
<td>A term representing a biological process or pathway</td>
</tr>
<tr>
<td><a href="BiomarkerTerm/">BiomarkerTerm</a></td>
<td>A biomarker term from NCIT</td>
</tr>
<tr>
<td><a href="CausalLinkTypeEnum/">CausalLinkTypeEnum</a></td>
<td>Degree of mechanistic directness represented by a causal edge</td>
</tr>
<tr>
<td><a href="CellTypeTerm/">CellTypeTerm</a></td>
<td>A cell type</td>
</tr>
<tr>
<td><a href="CellularComponentTerm/">CellularComponentTerm</a></td>
<td>A term representing a cellular component</td>
</tr>
<tr>
<td><a href="ChannelopathyOrganSystemEnum/">ChannelopathyOrganSystemEnum</a></td>
<td>Classification categories for channelopathies by affected organ system</td>
</tr>
<tr>
<td><a href="ChemicalEntityTerm/">ChemicalEntityTerm</a></td>
<td>A term representing a chemical entity</td>
</tr>
<tr>
<td><a href="ClinicalSignificanceEnum/">ClinicalSignificanceEnum</a></td>
<td>The clinical significance of a variant for a condition (ACMG guidelines)</td>
</tr>
<tr>
<td><a href="ClinicalTrialPhaseEnum/">ClinicalTrialPhaseEnum</a></td>
<td>Clinical trial phase categories per FDA/NIH standards</td>
</tr>
<tr>
<td><a href="ClinicalTrialStatusEnum/">ClinicalTrialStatusEnum</a></td>
<td>Clinical trial recruitment and status categories per ClinicalTrials</td>
</tr>
<tr>
<td><a href="ComorbidityDirectionEnum/">ComorbidityDirectionEnum</a></td>
<td>Directionality of a comorbidity/trajectory association</td>
</tr>
<tr>
<td><a href="ComputationalModelTypeEnum/">ComputationalModelTypeEnum</a></td>
<td>Type of computational or in-silico model</td>
</tr>
<tr>
<td><a href="ConditionCompositionEnum/">ConditionCompositionEnum</a></td>
<td>Composition type for a composite condition descriptor</td>
</tr>
<tr>
<td><a href="CurationActionEnum/">CurationActionEnum</a></td>
<td>Simple action types for curation audit trail</td>
</tr>
<tr>
<td><a href="CurationStatusEnum/">CurationStatusEnum</a></td>
<td>Curation workflow status for an association</td>
</tr>
<tr>
<td><a href="DatasetTypeEnum/">DatasetTypeEnum</a></td>
<td>Type of dataset or data resource</td>
</tr>
<tr>
<td><a href="DefinitionTypeEnum/">DefinitionTypeEnum</a></td>
<td>The type of definition or criteria set</td>
</tr>
<tr>
<td><a href="DiseaseTerm/">DiseaseTerm</a></td>
<td>A disease or medical condition</td>
</tr>
<tr>
<td><a href="EnvironmentTerm/">EnvironmentTerm</a></td>
<td>A term representing an environmental context, material, or feature (from ENVO...</td>
</tr>
<tr>
<td><a href="EvidenceItemSupportEnum/">EvidenceItemSupportEnum</a></td>
<td>The level of support for an evidence item</td>
</tr>
<tr>
<td><a href="EvidenceSourceEnum/">EvidenceSourceEnum</a></td>
<td>The provenance/source of the evidence item</td>
</tr>
<tr>
<td><a href="ExposureTerm/">ExposureTerm</a></td>
<td>A term representing an exposure event (from ECTO or XCO)</td>
</tr>
<tr>
<td><a href="ExpressivityEnum/">ExpressivityEnum</a></td>
<td>Expressivity classification for inheritance</td>
</tr>
<tr>
<td><a href="FrequencyEnum/">FrequencyEnum</a></td>
<td>The frequency of an event or phenomenon</td>
</tr>
<tr>
<td><a href="GeneProductTerm/">GeneProductTerm</a></td>
<td>A gene product term from NCIT</td>
</tr>
<tr>
<td><a href="GeneTerm/">GeneTerm</a></td>
<td>A gene term from HGNC</td>
</tr>
<tr>
<td><a href="GeographyTerm/">GeographyTerm</a></td>
<td>A place or location</td>
</tr>
<tr>
<td><a href="HarrisonsChapterEnum/">HarrisonsChapterEnum</a></td>
<td>Traditional internal medicine chapter groupings for disease classification</td>
</tr>
<tr>
<td><a href="HistopathologyFindingTerm/">HistopathologyFindingTerm</a></td>
<td>A histopathologic finding term from NCIT</td>
</tr>
<tr>
<td><a href="ICD10CMTerm/">ICD10CMTerm</a></td>
<td>An ICD-10-CM diagnosis code</td>
</tr>
<tr>
<td><a href="ICD11FTerm/">ICD11FTerm</a></td>
<td>An ICD-11 Foundation diagnosis code</td>
</tr>
<tr>
<td><a href="ICDOMorphologyEnum/">ICDOMorphologyEnum</a></td>
<td>ICD-O morphology axis classification for cancer subtypes</td>
</tr>
<tr>
<td><a href="InheritanceTerm/">InheritanceTerm</a></td>
<td>A term representing mode of inheritance</td>
</tr>
<tr>
<td><a href="IUISCategoryEnum/">IUISCategoryEnum</a></td>
<td>IUIS classification tables for inborn errors of immunity (IEI)</td>
</tr>
<tr>
<td><a href="LateralityEnum/">LateralityEnum</a></td>
<td>Laterality qualifier for anatomical structures or procedures</td>
</tr>
<tr>
<td><a href="LifeCycleStageTerm/">LifeCycleStageTerm</a></td>
<td>A parasite life cycle stage term (from OPL)</td>
</tr>
<tr>
<td><a href="LysosomalStorageEnum/">LysosomalStorageEnum</a></td>
<td>Classification of lysosomal storage diseases by accumulated substrate type</td>
</tr>
<tr>
<td><a href="MappingConsistencyEnum/">MappingConsistencyEnum</a></td>
<td>Consistency of a mapping relative to another reference source</td>
</tr>
<tr>
<td><a href="MechanisticHypothesisStatusEnum/">MechanisticHypothesisStatusEnum</a></td>
<td>Curation/maturity status for a disease-level mechanistic hypothesis</td>
</tr>
<tr>
<td><a href="MechanisticNosologyEnum/">MechanisticNosologyEnum</a></td>
<td>Classification of diseases by molecular mechanism or affected cellular system</td>
</tr>
<tr>
<td><a href="ModifierEnum/">ModifierEnum</a></td>
<td>Qualifiers for direction, intensity, or pathological state of a descriptor</td>
</tr>
<tr>
<td><a href="OnsetEnum/">OnsetEnum</a></td>
<td>Age of onset categories from HPO</td>
</tr>
<tr>
<td><a href="OrganismTerm/">OrganismTerm</a></td>
<td>A term representing an organism from NCBITaxon</td>
</tr>
<tr>
<td><a href="PenetranceEnum/">PenetranceEnum</a></td>
<td>Penetrance classification for inheritance</td>
</tr>
<tr>
<td><a href="PhaseTerm/">PhaseTerm</a></td>
<td>A phase or stage</td>
</tr>
<tr>
<td><a href="PhenotypeCategoryEnum/">PhenotypeCategoryEnum</a></td>
<td>Broad phenotype categories from the Human Phenotype Ontology</td>
</tr>
<tr>
<td><a href="PhenotypeTerm/">PhenotypeTerm</a></td>
<td>A term representing a phenotype or disease manifestation</td>
</tr>
<tr>
<td><a href="ProteinComplexTerm/">ProteinComplexTerm</a></td>
<td>A term representing a protein complex</td>
</tr>
<tr>
<td><a href="RegimenTerm/">RegimenTerm</a></td>
<td>A term representing a treatment regimen (from NCIT)</td>
</tr>
<tr>
<td><a href="SexEnum/">SexEnum</a></td>
<td>Sex-specific stratum</td>
</tr>
<tr>
<td><a href="TreatmentActionTerm/">TreatmentActionTerm</a></td>
<td>A term representing a medical action or treatment (from MAXO)</td>
</tr>
<tr>
<td><a href="TreatmentEffectEnum/">TreatmentEffectEnum</a></td>
<td>How a treatment affects a pathophysiology mechanism node</td>
</tr>
<tr>
<td><a href="TriggerTerm/">TriggerTerm</a></td>
<td>A trigger</td>
</tr>
<tr>
<td><a href="ZygosityEnum/">ZygosityEnum</a></td>
<td>Zygosity categories from GENO</td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="Boolean/">Boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="Curie/">Curie</a></td>
<td>a compact URI</td>
</tr>
<tr>
<td><a href="Date/">Date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="DateOrDatetime/">DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="Datetime/">Datetime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="Decimal/">Decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="Double/">Double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="Float/">Float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="FrequencyQuantity/">FrequencyQuantity</a></td>
<td></td>
</tr>
<tr>
<td><a href="Integer/">Integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="Jsonpath/">Jsonpath</a></td>
<td>A string encoding a JSON Path</td>
</tr>
<tr>
<td><a href="Jsonpointer/">Jsonpointer</a></td>
<td>A string encoding a JSON Pointer</td>
</tr>
<tr>
<td><a href="Ncname/">Ncname</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="Nodeidentifier/">Nodeidentifier</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="Objectidentifier/">Objectidentifier</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="PMID/">PMID</a></td>
<td></td>
</tr>
<tr>
<td><a href="Sparqlpath/">Sparqlpath</a></td>
<td>A string encoding a SPARQL Property Path</td>
</tr>
<tr>
<td><a href="String/">String</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="Time/">Time</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="Uri/">Uri</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="Uriorcurie/">Uriorcurie</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.7a47a382.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.e71a0d61.min.js"></script>
      
    
  </body>
</html>