---
reference_id: "PMID:24088064"
title: The Mediator complex and transcription regulation.
authors:
- Poss ZC
- Ebmeier CC
- Taatjes DJ
journal: Crit Rev Biochem Mol Biol
year: '2013'
doi: 10.3109/10409238.2013.840259
content_type: full_text_xml
---

# The Mediator complex and transcription regulation.
**Authors:** Poss ZC, Ebmeier CC, Taatjes DJ
**Journal:** Crit Rev Biochem Mol Biol (2013)
**DOI:** [10.3109/10409238.2013.840259](https://doi.org/10.3109/10409238.2013.840259)

## Content

1. Crit Rev Biochem Mol Biol. 2013 Nov-Dec;48(6):575-608. doi: 
10.3109/10409238.2013.840259. Epub 2013 Oct 3.

The Mediator complex and transcription regulation.

Poss ZC(1), Ebmeier CC, Taatjes DJ.

Author information:
(1)Department of Chemistry and Biochemistry, University of Colorado , Boulder, 
CO , USA.

The Mediator complex is a multi-subunit assembly that appears to be required for 
regulating expression of most RNA polymerase II (pol II) transcripts, which 
include protein-coding and most non-coding RNA genes. Mediator and pol II 
function within the pre-initiation complex (PIC), which consists of Mediator, 
pol II, TFIIA, TFIIB, TFIID, TFIIE, TFIIF and TFIIH and is approximately 4.0 MDa 
in size. Mediator serves as a central scaffold within the PIC and helps regulate 
pol II activity in ways that remain poorly understood. Mediator is also 
generally targeted by sequence-specific, DNA-binding transcription factors (TFs) 
that work to control gene expression programs in response to developmental or 
environmental cues. At a basic level, Mediator functions by relaying signals 
from TFs directly to the pol II enzyme, thereby facilitating TF-dependent 
regulation of gene expression. Thus, Mediator is essential for converting 
biological inputs (communicated by TFs) to physiological responses (via changes 
in gene expression). In this review, we summarize an expansive body of research 
on the Mediator complex, with an emphasis on yeast and mammalian complexes. We 
focus on the basics that underlie Mediator function, such as its structure and 
subunit composition, and describe its broad regulatory influence on gene 
expression, ranging from chromatin architecture to transcription initiation and 
elongation, to mRNA processing. We also describe factors that influence Mediator 
structure and activity, including TFs, non-coding RNAs and the CDK8 module.

DOI: 10.3109/10409238.2013.840259
PMCID: PMC3852498
PMID: 24088064 [Indexed for MEDLINE]

Expression of most non-coding RNA genes and all protein-coding genes is controlled by the RNA polymerase II (pol II) enzyme; however, pol II does not initiate promoter-specific transcription on its own. Rather, pol II functions and is regulated within a macromolecular assembly known as the pre-initiation complex (PIC), consisting of TFIIA, TFIIB, TFIID, TFIIE, TFIIF, TFIIH, pol II and Mediator (Hahn, 2004; Thomas & Chiang, 2006). Among the PIC components, Mediator was the last to be discovered. Using primarily yeast genetics and biochemistry, the Young and Kornberg labs converged on a factor/activity that interacted with the pol II enzyme and was needed for activator-dependent transcription in vitro and in vivo (Flanagan et al., 1991; Kelleher-III et al., 1990; Koleske & Young, 1994; Nonet & Young, 1989; Thompson et al., 1993). This factor ultimately became known as the Mediator complex (Conaway & Conaway, 2011; Kornberg, 2005). The isolation of human Mediator complexes relied in large part on biochemical purifications via different transcription factor (TF) activation domains (Boyer et al., 1999; Fondell et al., 1996; Ito et al., 1999; Naar et al., 1999; Rachez et al., 1999; Ryu et al., 1999), which led to acronyms such as TRAP (thyroid hormone receptor associated proteins) and ARC (activator recruited cofactor). Collectively, these complexes are now generally called Mediator and share a unified subunit nomenclature (Bourbon et al., 2004).

Mediator is not required for transcription per se, and over evolutionary time (Figure 1), it emerged in eukaryotic organisms. Throughout evolution, Mediator sequences have diverged rapidly, such that identity or similarity is modest between yeast and human subunits (Boube et al., 2002; Bourbon, 2008; Levine & Tjian, 2003). Moreover, human Mediator contains subunits with no identifiable counterpart in yeast (Table 1).
Figure 1.Evolutionary timeline. Note large intervals for evolution of microbial to eukaryotic life, and for single-celled eukaryotes to metazoans. (see colour version of this figure online at www.informahealthcare.com/bmg).
Table 1.Basic comparison of Mediator subunits in humans (Hs), yeast (Sc), fly (Dm), and mouse (Mm) by percent identity, percent similarity, and size.Percent Identity to Hs
No. of Residues
Predicted MW (kDa)
Percent Similarity to Hs

Hs MED subunit

Sc

Dm

Mm

Hs

Sc

Dm

Mm

Hs

Sc

Dm

Mm

Sc

Dm

Mm
MED17%24%94%158156614751575168.564.3149.5167.112%38%97%MED421%40%94%27028425827029.732.228.329.838%58%97%MED619%45%95%24629524924628.432.828.428.431%63%98%MED7 (2)23%50%97%23322222023327.225.625.727.240%66%98%MED815%44%97%26822325226829.125.327.929.228%60%97%MED916%26%78%14614914414216.417.416.815.730%46%86%MED1024%57%99%13515713313515.717.915.715.741%75%100%MED1119%27%97%11711517611713.113.319.613.136%42%98%MED1413%37%96%1454108215531459160.6123.4172.3161.023%53%98%MED15 (2)18%30%89%788108174979286.8120.380.587.127%41%92%MED16 (4)15%27%87%87797481882896.8111.390.091.827%42%90%MED1714%43%96%65168764264972.978.571.672.627%62%98%MED1812%47%99%20830721720823.734.324.723.624%65%100%MED1913%29%94%24422033724426.324.935.426.322%37%96%MED2011%36%97%21221025221223.222.927.923.219%54%99%MED2125%54%98%14414014214415.616.116.115.642%66%99%MED2211%43%95%20012114320022.213.916.622.322%55%96%MED23 (4)n/a45%96%1368n/a14391367156.5n/a167.1156.1n/a63%99%MED24a
15%28%94%9891132993987110.3128.8111.7110.029%49%97%MED25 (4)n/a27%93%747n/a86374578.2n/a96.678.1n/a39%94%MED26n/a10%85%600n/a148358865.4n/a165.564.7n/a16%89%MED27a
13%37%98%31139729331135.443.133.935.325%60%100%MED28n/a24%94%178n/a18917819.5n/a21.119.5n/a37%98%MED29a
8%32%92%20043118819921.147.721.121.015%45%95%MED30n/a20%95%178n/a31817820.3n/a35.320.4n/a31%98%MED3128%43%98%13112720413115.814.723.515.840%55%100%CDK829%71%99%46455545446453.362.853.753.238%77%99%CDK19n/an/a96%502n/an/a50156.8n/an/a56.6n/an/a97%CCNC25%68%99%28332326728333.237.831.333.240%80%100%MED1213%34%97%2177142725312190243.1166.9279.5244.624%46%98%MED12L (2)n/an/a94%2145n/an/a2157240.1n/an/a241.4n/an/a96%MED13 (3)13%27%94%2174142026182171239.3160.0280.0238.622%40%97%MED13Ln/an/a92%2210n/an/a2207242.6n/an/a241.8n/an/a96%Numbers in parentheses next to Mediator subunits represent the number of isoforms documented in Universal Protein Resource (UniProt). The superscript (a) denotes that Hs subunits MED24, MED27 and MED29 were identified as orthologous to Sc MED5, MED3 and MED2 (Bourbon, 2008). All protein sequences were retrieved from UniProt, and percent identity and similarity were calculated using the EMBOSS Needle pairwise alignment tool on the EBI-EMBL server.


Evolutionary timeline. Note large intervals for evolution of microbial to eukaryotic life, and for single-celled eukaryotes to metazoans. (see colour version of this figure online at www.informahealthcare.com/bmg).

Basic comparison of Mediator subunits in humans (Hs), yeast (Sc), fly (Dm), and mouse (Mm) by percent identity, percent similarity, and size.

Numbers in parentheses next to Mediator subunits represent the number of isoforms documented in Universal Protein Resource (UniProt). The superscript (a) denotes that Hs subunits MED24, MED27 and MED29 were identified as orthologous to Sc MED5, MED3 and MED2 (Bourbon, 2008). All protein sequences were retrieved from UniProt, and percent identity and similarity were calculated using the EMBOSS Needle pairwise alignment tool on the EBI-EMBL server.

The Mediator complex is a global regulator of gene expression and as such, is considered a general transcription factor (Ansari et al., 2009; Takagi & Kornberg, 2006). However, what distinguishes Mediator from other general transcription factors (with the possible exception of TFIID) is its high degree of structural flexibility, its variable subunit composition, and its general requirement for activated (e.g. enhancer driven) transcription (Malik & Roeder, 2010). Consistent with its ability to stimulate activated transcription, Mediator appears to be the main binding interface for DNA-binding TFs within the PIC (Borggrefe & Yue, 2011). These features are important for both general and context-specific functions, such that this “general transcription factor” may operate in mechanistically distinct ways at different genes or in different cell types.

In this review, we summarize much of the published work on the Mediator complex, focusing mostly on the yeast and human complexes, in part because the majority of studies have been completed with these organisms. Indicative of the many ways that Mediator governs gene expression, this review is expansive and covers many aspects of Mediator function, including some that have emerged only recently. Periodically, we provide some of our own hypotheses or highlight future directions that arise from a particular set of findings. We start with the basic biochemical and biophysical features of the Mediator complex, then describe its diverse roles in regulating gene expression, from PIC structure to chromatin architecture. Throughout, we try to emphasize structure and mechanism, and to point out areas in which current understanding is limited.

In this section we outline basic information about Mediator subunit composition, known roles for specific subunits and modules, and how subunit composition might be regulated. We start with an overview of mass spectrometry (MS) studies of Mediator, as these have been instrumental in determining its subunit composition.

Mass spectrometry-based studies have defined the subunit composition of Mediator and uncovered new insights about its function. One of the first studies to characterize the components of yeast Mediator complexes with mass spectrometry identified two forms of the isolated complex, with and without the CDK8 module (Liu et al., 2001). In the following years, human orthologs of yeast (Sato et al., 2003b; Tomomori-Sato et al., 2004) and Drosophila (Sato et al., 2003a) Mediator subunits were identified using MS. Given the many subunits associated with Mediator and the fact that subunits appeared to be variably associated, the precise composition of the Mediator complex remained murky for some time. In a landmark study, the Conaway and Washburn labs used the shotgun proteomics MS-based method multidimensional protein identification technology (MudPIT) to define the set of consensus Mediator subunits (Sato et al., 2004). The subunit composition of human Mediator, purified from six different FLAG-tagged subunits, was systematically examined and compared. A follow-up study characterizing the abundance of subunits in isolated Mediator complexes found that complexes containing MED26 also contained the most pol II and were largely – but not completely – devoid of CDK8 module subunits (Paoletti et al., 2006). Another proteomics-based study from the Conaway group identified components of the super elongation complex and the general transcription factor TFIID as factors stably associated with Mediator via its MED26 subunit (Takahashi et al., 2011). Thus, MS-based proteomics enabled discovery of a role for MED26 in regulating the pol II initiation-elongation transition. The subunit composition of the Mediator complex has been independently confirmed by large scale immunoprecipitation mass spectrometry (IP-MS) studies of endogenous human complexes (Malovannaya et al., 2011) that also suggest novel interactions that may be functionally significant.

The Carey group, working in collaboration with the Wohlschlegel lab, has combined mass spectrometry with immobilized DNA template assays to assemble and characterize PIC composition under precisely controlled conditions. Their work has revealed new insights about Mediator and PIC assembly and function. For example, CHD1 was identified as a PIC factor whose recruitment was Mediator dependent (Lin et al., 2011). Another study by these investigators highlights the sensitivity of the proteomics technology. It was found that both HeLa and murine ES cells had very similar PIC compositions, with the Mediator and SAGA complexes as the two major activator-recruited factors (Chen et al., 2012b). Experiments in vitro suggested Mediator may assemble the PIC whereas SAGA was important for chromatin remodeling. Each of the above studies were coupled with genomic profiling of the relevant factors to provide in vivo data together with the proteomics.

The MudPIT-mass spectrometry methodology was also applied to address whether TF-induced structural rearrangements in Mediator (Taatjes et al., 2002) could accommodate distinct Mediator-cofactor interactions. Mediator complexes purified bound to different TF activation domains (SREBP or VP16) were compared with Mediator complexes purified by immunoprecipitation. Different sets of transcription cofactors were identified that were specific to each TF-bound Mediator complex, suggesting that different cofactors associate with Mediator in different structural states (Ebmeier & Taatjes, 2010). Furthermore, cofactors specific to CDK8-Mediator included P-TEFb and AFF4, both components of the super elongation complex (Luo et al., 2012b). These findings generated new and testable hypotheses that illustrate the value of MS-based proteomics as a discovery tool.

Compositionally distinct forms of Mediator can be isolated as stable entities (Belakavadi & Fondell, 2010; Elmlund et al., 2006; Spahr et al., 2003; Taatjes et al., 2002; Wang et al., 2001), with the most common being a 26 subunit “core” complex (21 subunit in Saccharomyces cerevisiae) and a 29 subunit “CDK8-Mediator” complex (25 subunit in S. cerevisiae). The subunit composition of the human core Mediator (hereafter called Mediator) and CDK8-Mediator complexes are shown in Table 2. What distinguishes each complex is a four-subunit CDK8 module consisting of the MED12, MED13, CDK8 and CCNC proteins; also, the MED26 subunit appears to dissociate upon CDK8 module binding (Taatjes et al., 2002), although a fraction of Mediator complexes might contain the CDK8 module and MED26 (Paoletti et al., 2006; Sato et al., 2004).
Table 2.List of human Mediator subunits, along with their approximate molecular weight.Mediator subunitMolecular weightMED1220 kDaMED436 kDaMED633 kDaMED734 kDaMED832 kDaMED916 kDaMED1016 kDaMED1113 kDaMED14150 kDaMED15105 kDaMED1695 kDaMED1778 kDaMED1828 kDaMED1926 kDaMED2023 kDaMED2119 kDaMED2216 kDaMED23130 kDaMED24100 kDaMED2592 kDaMED2670 kDaMED2737 kDaMED2820 kDaMED2924 kDaMED3025 kDaMED3118 kDaCDK855 kDaCCNC34 kDaMED12240 kDaMED13250 kDaCDK8 module subunits are shown at the bottom.


List of human Mediator subunits, along with their approximate molecular weight.

CDK8 module subunits are shown at the bottom.

Many studies have now documented the reversible “on/off” binding of the CDK8 module to Mediator, both in vitro and in cells (Davis et al., 2013; Drogat et al., 2012; Kim et al., 2006b; Knuesel et al., 2009a; Mo et al., 2004; Pavri et al., 2005; Tsai et al., 2013). The Holstege and Gustafsson labs used ChIP-chip assays to show co-localization of CDK8 module components with Mediator across the yeast genome, and the Holstege group completed ChIP-reChIP assays that suggested transient CDK8 module association (Andrau et al., 2006; Zhu et al., 2006). Similar genomic co-localization of Mediator and CDK8 module components was later observed in mammalian cells (Kagey et al., 2010).

Recombinant expression and purification has allowed multi-subunit head and middle modules of yeast Mediator to be purified (Koschubs et al., 2010; Takagi et al., 2006). Whereas this has been extremely valuable for structural and functional understanding of these domains (Cai et al., 2012; Imasaki et al., 2011; Lariviere et al., 2012; Robinson et al., 2012), it is not clear whether these sub-assemblies have significant biological roles on their own. Exceptions include the head module in trypanosomes (Lee et al., 2010a) and the four subunit CDK8 module, which has been isolated as a stable assembly in both yeast and human cells (Borggrefe et al., 2002; Elmlund et al., 2006; Knuesel et al., 2009b; Tsai et al., 2013). The regulatory roles for the CDK8 module are discussed in depth later in this review.

The different subunits of Mediator, to a degree, are involved in regulating different sets of genes. This theme first emerged with yeast genetic studies (Holstege et al., 1998; van de Peppel et al., 2005). Knockout of yeast Mediator subunits revealed that many are required for viability and play general roles in gene expression. The Med17 and Med21 subunits, in particular, are required for expression of virtually all protein-coding genes in yeast (Holstege et al., 1998; Thompson & Young, 1995). By comparison, other non-essential Mediator subunits have specialized, gene-selective roles in transcription (Uwamahoro et al., 2012). The combination of genetic and biochemical experiments in yeast led to a model in which select Mediator subunits help activate specific sets of genes (Linder et al., 2008; van de Peppel et al., 2005). This model is consistent with genetic studies of Mediator in flies and worms (Kim et al., 2004; Park et al., 2001a, 2000; Taubert et al., 2006).

Although every Mediator subunit knockout reported in mammals has been embryonic lethal (Ito et al., 2002, 2000; Stevens et al., 2002; Tudor et al., 1999; Westerling et al., 2007), cell lines have been derived from knockout embryos in some cases, allowing evaluation of Mediator activity in cellular and in vitro assays. Mouse knockout experiments from the Roeder (Med24 knockout) and Berk labs (Med23 knockout) have revealed that MED23, MED16, and MED24 might form a stable sub-assembly, as loss of either Med23 or Med24 resulted in Mediator complexes with reduced levels of these three subunits (Ito et al., 2002; Stevens et al., 2002). The Roeder group also noted sub-stoichiometric levels of Cdk8 upon loss of Med24 in murine embryonic fibroblasts (MEFs). MED1 represents another Mediator subunit whose absence does not seem to affect complex integrity. Mediator isolated from Med1 knockout MEFs is stable and transcriptionally active (Ito et al., 2000; Malik et al., 2004). It also appears that MED1-deficient Mediator complexes are present endogenously, as shown by the Tjian and Roeder labs (Malik et al., 2004; Taatjes & Tjian, 2004). Notably, endogenous Mediator complexes that lacked MED1 also lacked MED26, suggesting these subunits might form a sub-assembly in Mediator. EM analysis of this complex revealed regions with missing density (Figure 2) compared with the Mediator complex that contained MED1 and MED26 (Taatjes & Tjian, 2004).
Figure 2.EM structure of human Mediator compared with human Mediator lacking the MED1 and MED26 subunits. Both complexes are bound to the activation domain of VP16, and each is rendered at their predicted molecular weight (1.2 MDa or 0.9 MDa, respectively). The circled region indicates one area of missing protein density in the complex lacking MED1 and MED26. Note, however, that a pol II interaction surface (dashed yellow line; see text) is maintained in both structures, consistent with a general ability of each complex to activate transcription by VP16 (Taatjes & Tjian, 2004). (see colour version of this figure online at www.informahealthcare.com/bmg).


EM structure of human Mediator compared with human Mediator lacking the MED1 and MED26 subunits. Both complexes are bound to the activation domain of VP16, and each is rendered at their predicted molecular weight (1.2 MDa or 0.9 MDa, respectively). The circled region indicates one area of missing protein density in the complex lacking MED1 and MED26. Note, however, that a pol II interaction surface (dashed yellow line; see text) is maintained in both structures, consistent with a general ability of each complex to activate transcription by VP16 (Taatjes & Tjian, 2004). (see colour version of this figure online at www.informahealthcare.com/bmg).

The links between Mediator subunits and regulation of sets of genes derives, at least in part, from the fact that different TFs bind different Mediator subunits (Table 3). This is observed in both human and yeast cells, although a greater number of subunits appear to be bound by TFs in humans. Because TF-Mediator binding is essential for target gene activation, loss of a specific Mediator subunit can, to varying degrees, prevent expression of genes regulated by a given TF. This has been widely demonstrated with genetic studies in yeast and lower metazoans, with similar findings in mammals (van Essen et al., 2009). For example, the MED1 subunit is a common target for nuclear receptors. The Roeder group observed defects in nuclear receptor-dependent gene expression in Med1 knockout MEFs, whereas activation by other TFs that interact with different Mediator subunits was not negatively impacted (Ito et al., 2000). Similarly, mouse Med23 knockout cells were unable to support activation by the ELK-1 or E1A TFs, whereas activation by TFs such as VP16 and p53 were unaffected (Stevens et al., 2002). ELK-1 and E1A bind Mediator through Med23, whereas VP16 or p53 do not (Table 3). In follow-up work, the Berk lab examined the effect of Med23 knockout in different cell types. They observed that whereas Egr1 expression (induced in part by the ELK-1 TF) was ablated in mES cells, Egr1 expression recovered to a degree in Med23 knockout murine embryonic fibroblast (MEF) cells (Balamotis et al., 2009). This was due to differential TF requirements (i.e. less dependence on ELK-1 compared with other TFs) for Egr1 expression in MEFs. These data do not suggest the basic function of Med23 is distinct in MEFs, but rather that different TFs regulate Egr1 expression in MEFs compared with mES cells. This agrees with recent findings that demonstrate the same TF, especially those that respond to signaling cascades, can regulate different sets of genes in different cell types (Mullen et al., 2011; Trompouki et al., 2011).
Table 3.DNA-binding TFs and their identified Mediator subunit target(s)*.GeneTFReferenceOrganismGeneTFReferenceOrganismMED1TRαFondell et al., 1996
MED15Smad2/4Kato et al., 2002
Yuan et al., 1998
Smad3/4Kato et al., 2002
Malik et al., 2004
NHR-49Taubert et al., 2006

C. elegans
TRβYuan et al., 1998
Oaf1Thakur et al., 2009

Yeast
Zhu et al., 1997
Pdr1Thakur et al., 2008

Yeast
VDRYuan et al., 1998
Pdr3Thakur et al., 2008

Yeast
Rachez et al., 1999
VP16Park et al., 2000

Yeast
RARαYuan et al., 1998
Gal4Park et al., 2000

Yeast
Zhu et al., 1997
Gcn4Park et al., 2000

Yeast
Shao et al., 2000
Swanson et al., 2003

Yeast
Lee et al., 2007
Zhang et al., 2004
RXRαYuan et al., 1998
SREBP-1aYang et al., 2006
Zhu et al., 1997
MED16DifKim et al., 2004
PPARαYuan et al., 1998
Gcn4Swanson et al., 2003

Yeast
Zhu et al., 1997
MED17VP16Ito et al., 1999
PPARγYuan et al., 1998
Park et al., 2003

Drosophila
Ge et al., 2002
p53Ito et al., 1999
Ge et al., 2008
Meyer et al., 2010
ERKang et al., 2002
ERBurakov et al., 2000
Zhang et al., 2005
HsfPark et al., 2001b
Drosophila
Burakov et al., 2000
Park et al., 2003

Drosophila
Kim et al., 2008
Kim et al., 2004
Warnmark et al., 2001
DifPark et al., 2003

Drosophila
ARWang et al., 2002
Kim et al., 2004
GRHittelman et al., 1999
STAT2Lau et al., 2003
Chen et al., 2006
Gal4Koh et al., 1998

Yeast
Chen & Roeder, 2007
RXRPark et al., 2003

Drosophila
Kim et al., 2008
p65van Essen et al., 2009
HNF4Malik et al., 2002
MED19RESTDing et al., 2009
p53Ito et al., 1999
MED21TRαNevado et al., 2004

Yeast
Frade et al., 2000
Tup1Hallberg et al., 2006

Yeast
Drane et al., 1997
MED22GCN4Swanson et al., 2003

Yeast
Meyer et al., 2010
MED23C/EBPβMo et al., 2004
BRCA1Wada et al., 2004
ESXAsada et al., 2002
NR4AWansa & Muscat, 2005
Shimogawa et al., 2004
FXRPineda Torra et al., 2004
E1aStevens et al., 2002
RORαAtkins et al., 1999
Wang & Berk, 2002
AHRWang et al., 2004bBoyer et al., 1999
GATA-1Stumpf et al., 2006
ELK1Asada et al., 2002
Crawford et al., 2002
Stevens et al., 2002
Pit-1Gordon et al., 2006
Cantin et al., 2003
GATA-2Gordon et al., 2006
DifKim et al., 2004

Drosophila
GABPαUdayakumar et al., 2006
HSFKim et al., 2004

Drosophila
MYCLiu et al., 2008
IRF7Griffiths et al., 2013
POU1F1Gordon et al., 2006
MED24TRYuan et al., 1998
14-3-3Zilliacus et al., 2001
MED25VP16Mittler et al., 2003
PGC-1αWallberg et al., 2003
Yang et al., 2004
C/EBPβLi et al., 2008
Park et al., 2000
MED2Gcn4Zhang et al., 2004

yeast
RARLee et al., 2007
Natarajan et al., 1999

yeast
HNF4αRana et al., 2011
MED3Gcn4Zhang et al., 2004

yeast
ERMVerger et al., 2013
MED8Ace2Mehta et al., 2009

yeast
SOX9Nakamura et al., 2011
MED12RTAGwack et al., 2003
ATF6αSela et al., 2013
SOX9Zhou et al., 2002
MED28GRB2Wiederhold et al., 2004
Rau et al., 2006

Zebrafish
MED29DSXSato et al., 2003aNANOGTutter et al., 2009
Garrett-Engele et al., 2002

Drosophila
CTNNB1Kim et al., 2006aMED31HSFKim et al., 2004

Drosophila
RESTDing et al., 2008
CDK8MYCEberhardy & Farnham, 2002
G9aDing et al., 2008
HsfPark et al., 2001
Drosophila
Gli3Zhou et al., 2006
β-cateninKim et al., 2006aAICDXu et al., 2011bMED14GRHittelman et al., 1999
Chen et al., 2006
STAT2Lau et al., 2003
HNF4Malik et al., 2002
ERαLee et al., 2005aPPARγGrontved et al., 2010
SREBP-1aToth et al., 2004
*All interactions were identified in mammals unless otherwise noted. References that validate the interaction are also listed.


DNA-binding TFs and their identified Mediator subunit target(s)*.

All interactions were identified in mammals unless otherwise noted. References that validate the interaction are also listed.

These biochemical and knockout studies could reflect a biologically relevant means to regulate the Mediator complex. Loss of select Mediator subunits could minimize or perhaps even prevent expression of sets of genes activated or repressed by specific TFs. Whether this represents a biologically relevant mechanism remains to be established; however, the means to implement such regulation are straightforward: expression of a specific Mediator subunit could be reduced or individual subunits could be targeted for degradation by the proteasome (Davis et al., 2013) and/or targeted by miRNAs. In each circumstance, sets of genes could be down-regulated (or up-regulated) because a TF binding site on Mediator was lost. A simple “on” versus “off” switch may not depend solely on a single Mediator subunit, however, as numerous studies have documented cooperative or redundant TF binding among Mediator subunits (Chen et al., 2006; Ding et al., 2009; Grontved et al., 2010; Hasegawa et al., 2012; Imberg-Kazdan et al., 2013; Kim & Gross, 2013).

Studies from the Tjian lab have suggested that in differentiated cells, the subunit composition of Mediator becomes more simplified. By tracking murine ES cells through different stages of differentiation, Deato et al. and D’Alessio et al. noted that protein and steady-state mRNA levels of many Mediator subunits declined, in some cases to nearly undetectable levels, in differentiated cells (D'Alessio et al., 2011; Deato et al., 2008). An implication from their work is that proliferating cells, such as cancer cells or stem cells, might generally express the full complement of Mediator subunits whereas differentiated cells express only a subset of Mediator subunits.

Initiation of a signaling cascade (e.g. an inflammatory response to a cytokine) can ultimately result in changes in gene expression; because Mediator directly controls pol II activity, and therefore, gene expression patterns, Mediator is considered an endpoint of signaling cascades (Jiang et al., 1998; Takagi & Kornberg, 2006). The fact that post-translational modifications (PTMs) help regulate Mediator function supports this notion (Fondell, 2013), as does the fact that many DNA-binding TFs (which are themselves subject to regulation by signaling cascades) ultimately function by interacting with Mediator at their target promoters (Borggrefe & Yue, 2011).

A growing number of studies have shown how Mediator activity can be governed by post-translational modification (PTM) of its subunits (Nagalingam et al., 2012). PTM sites have been uncovered with global proteomics approaches (Beausoleil et al., 2004; Olsen et al., 2006). In more detailed mechanistic studies, a number of Mediator PTM sites have been linked to functional outcomes. The Fondell lab has uncovered key roles for MED1 phosphorylation in the MAPK/ERK signaling pathway. Phosphorylation of MED1 (at T1032 and T1457) correlated with increased transcription and increased MED1 stability within Mediator (Belakavadi et al., 2008; Pandey et al., 2005). Increased transcription was noted in response to nuclear receptor target genes, consistent with MED1 binding by nuclear receptors (Table 3). In agreement with these findings, the Wang group has shown that expression of the androgen receptor oncogene target UBE2C was sensitive to MED1 phosphorylation at T1032 (Chen et al., 2011). MED1 phosphorylation was linked to more stable and active PICs; furthermore, UBE2C expression correlated with chromatin loop formation (linking the enhancer and promoter), and this architectural change was dependent on MED1 phosphorylation by the PI3K/AKT pathway. Using a combination of in vitro and MS-based methods, the O’Malley lab has demonstrated that several Mediator subunits, including MED1, are phosphorylated upon formation of active transcription complexes (Foulds et al., 2013).

Yeast Mediator complexes are also extensively phosphorylated, suggesting that PTMs represent a conserved means to regulate Mediator function. The Cramer and Mann laboratories completed a SILAC-based phospho-proteomic analysis of Mediator in S. cerevisiae (Miller et al., 2012). In all, this analysis identified 125 modification sites within 17 Mediator subunits. This same study also confirmed a role for Med15 phosphorylation (a common target of stress-induced TFs) in maintaining repression of stress-response genes under normal conditions (Miller et al., 2012). Earlier work also implicated PTMs in regulating Mediator activity. Two sites within S. cerevisiae Med13 (Srb9) were shown to be targeted by PKA (Chang et al., 2004), whereas phosphorylation of Med2 (by CDK8/Srb10) was able to block gene activation by a single TF responsive to low iron conditions (Hallberg et al., 2004; van de Peppel et al., 2005). Although the phosphorylation sites identified in Med2 (S208) or Med13/Srb9 (S608 and S1236) are not conserved in human Mediator, this pair of studies was among the first to confirm PTM-dependent regulation of Mediator function (Chang et al., 2004; van de Peppel et al., 2005).

Of course, many different PTMs are observed in eukaryotes, and it is certain that modifications other than phosphorylation will be discovered that control Mediator function. Ubiquitylation is a well-established regulator of protein degradation and signals proteins for recruitment to the proteasome. The Clurman lab demonstrated that MED13 and its paralog MED13L are ubiquitylated by the ubiquitin ligase FBW7, and this modification regulates MED13 and MED13L abundance and stability (Davis et al., 2013). FBW7-dependent ubiquitylation relies upon substrate phosphorylation (Welcker & Clurman, 2008); the Clurman group identified canonical phospho-degron motifs in MED13 and MED13L (at T326) that controlled MED13/MED13L ubiquitylation and turnover in vitro and in cells (Davis et al., 2013). Significantly, FBW7-dependent degradation of MED13 helps regulate CDK8 module interaction with Mediator, which has important regulatory consequences (described later). In a related study in Schizosaccharomyces pombe, Cdk11 phosphorylation of Med27 (Pmc3) and Med4 (Pmc4) was shown to regulate CDK8 module–Mediator association (Drogat et al., 2012).

Despite its large size and many subunits, Mediator is largely devoid of known enzymatic functions. Yeast Med5 was shown to harbor acetyltransferase activity toward a nucleosomal substrate (Lorch et al., 2000), whereas murine Med8 is capable of nucleating assembly of a ubiquitin ligase consisting of Elongin B and C, CUL2 and RBX1 (Brower et al., 2002). The kinase CDK8, part of the CDK8 module, represents a well studied, evolutionarily conserved enzymatic activity that can associate with Mediator (Xu & Ji, 2011). Mediator does not appear to have sequence-specific DNA binding capability, and seemingly relies upon DNA-binding TFs for recruitment. It is interesting to note, however, that Mediator has been linked to promoter-selective regulatory functions in both human cells and yeast (Ansari et al., 2012; Xu et al., 2011a). These functions involve Mediator interactions with auxiliary factors (e.g. HMGA1, SAGA) and do not appear to represent a Mediator DNA-binding activity. The lack of predicted or known DNA-binding or enzymatic functions, however, does not preclude such activities from existing within Mediator. Many examples have been reported of DNA-binding or enzymatic functions in proteins lacking predicted sequence motifs (Hu et al., 2009, Linares et al., 2007).

Finally, it is possible that Mediator subunits could have biological function as individual entities. It is noteworthy that, in an exhaustive immunoprecipitation-mass spectrometry study, the interaction network of MED15 was distinct relative to other Mediator subunits, suggesting it may function independently of Mediator (Malovannaya et al., 2011). The MED12 subunit might function independently as a regulator of TGFβ signaling. The Benards group showed evidence that MED12 could function in the cytoplasm to directly block TGFβ signaling by interacting with TGFβR2. This unexpected activity for MED12 provides rationale for reduced MED12 expression as a drug-resistance mechanism, as observed in a subset of drug-resistant tumors (Huang et al., 2012a). MED12 represents an interesting case as it has been identified as a signaling pathway “hub” gene in a Caenorhabditis elegans RNAi screen (Lehner et al., 2006), supporting distinct functions relative to other Mediator subunits.

As we describe later in this review, the large size and variable subunit composition of Mediator is required for its numerous regulatory functions, ranging from chromatin organization to TF binding. Precisely why Mediator is so large remains an open question, however, and much remains to be discovered about how subunits work collectively and what each subunit contributes to Mediator function.

Mediator subunits contain an unusually high number of intrinsically disordered regions, and many of these intrinsically disordered regions contain known or predicted protein-protein interaction domains (Toth-Petroczy et al., 2008). Although the yeast and human Mediator sequences are only weakly conserved (Table 1), the placement of disordered regions within subunits is similar. As a general trend, the size and number of intrinsically disordered regions has increased from yeast to humans (Toth-Petroczy et al., 2008).

The flexibility predicted by the sequences has been verified with structural studies. Early structural studies with yeast Mediator immediately revealed its flexibility. In 1999, pioneering electron microscopy (EM) work in the Kornberg lab indicated the general structure of yeast Mediator and provided the first evidence of its conformational variability (Asturias et al., 1999). Particularly striking were the structural changes that occurred with Mediator-pol II interaction. Subsequent work by the Asturias group has shown evidence for conformational flexibility among different Mediator domains in the absence of pol II binding (Cai et al., 2009). This flexibility can even be inferred from the yeast Mediator structure (Figure 3).
Figure 3.Cryo-EM structure of yeast Mediator. The EM data reveal structural flexibility (Cai et al., 2009) that can even be inferred from the 3D reconstruction, with its large domains connected by narrow linkers. Note also the extensive surface area, due to channels and cavities in the structure. (see colour version of this figure online at www.informahealthcare.com/bmg).


Cryo-EM structure of yeast Mediator. The EM data reveal structural flexibility (Cai et al., 2009) that can even be inferred from the 3D reconstruction, with its large domains connected by narrow linkers. Note also the extensive surface area, due to channels and cavities in the structure. (see colour version of this figure online at www.informahealthcare.com/bmg).

Structural studies with portions of the yeast Mediator complex, mostly involving head and middle module subunits, have shown conformational flexibility as well. The Cramer lab has crystallized several sub-assemblies within the Mediator head module and middle module, and these data have suggested molecular mechanisms that underlie Mediator conformational dynamics (Koschubs et al., 2009; Seizl et al., 2011). A Med7–Med21 dimer was shown to possess a flexible hinge that adopted two different crystal forms (Baumli et al., 2005). Conformational flexibility of this “middle” domain may be important for coordinating structural shifts that propagate throughout the Mediator complex. Flexibility was also observed with crystal structures of head module subunits, including Med20 within a Med8–Med18–Med20 complex (Lariviere et al., 2006).

In a remarkable set of papers, crystal structures representing a majority of the seven subunit yeast Mediator head module were reported. The Takagi lab was first to report a structure for the S. cerevisiae head module (Imasaki et al., 2011), followed by a head module crystal structure from the Kornberg group (Robinson et al., 2012); the Cramer lab reported the first S. pombe head module structure (Lariviere et al., 2012). Comparison of these structures showed conformational differences, even among both crystal structures from S. cerevisiae. The S. pombe head module crystals further supported a dynamic structure (Lariviere et al., 2012); for example, Med6 adopted different conformational states in different crystals, and various domain movements and rotations were noted throughout the assembly. Evidence for structural variability was also seen in Mediator head module crystal structures in S. cerevisiae (Imasaki et al., 2011). Prior to the crystal structure data, EM studies of the S. cerevisiae Mediator head module indicated the movable and fixed jaw domains were highly flexible (Cai et al., 2010), likely due to the flexibility of linkers (e.g. the “joint” consisting of portions of Med17, Med11, and Med22 and a flexible region within Med8) that connect these domains (Lariviere et al., 2012).

The studies described above highlight the inherent flexibility of the Mediator complex; that is, conformational variation that occurs apart from binding any external factors. Below, we summarize structural data that indicate larger scale conformational changes in Mediator. At a basic level, each case describes structural shifts that are triggered by distinct “ligands” that, upon binding Mediator, induce structural changes. The ligands include pol II, the CDK8 module and DNA-binding TFs.

Perhaps the most functionally significant biological similarity between yeast and human Mediator is pol II binding. Genetic and biochemical experiments that focused on the C-terminal domain (CTD) of the large subunit of pol II were instrumental in identification of Mediator in yeast (Kim et al., 1994; Thompson et al., 1993). Yeast Mediator subunits physically and functionally interacted with the pol II CTD (Myers et al., 1998), leading to initial models of a stable Mediator–pol II holoenzyme. Later, it was shown that human Mediator could also bind with high affinity to the pol II CTD; interestingly, the CDK8-Mediator complex is incapable of binding the pol II CTD (Myers et al., 1998; Naar et al., 2002). This biochemical difference between Mediator and CDK8-Mediator reflects basic differences in how these distinct forms of Mediator regulate transcription, and is described later.

Upon binding the pol II CTD, human Mediator undergoes a major structural shift, as shown in Figure 4 (Naar et al., 2002). Interestingly, the structural state induced by pol II CTD binding appears to be identical to the structural state induced by VP16 binding (within the limits of the low-resolution EM reconstructions). VP16 is a potent transcriptional activator, and these structural similarities suggested that the structural state of Mediator could regulate its biological activity (Naar et al., 2002; Taatjes et al., 2002). Whereas the CTD binding site on human Mediator was roughly estimated based upon antibody labeling, it must be emphasized that the human pol II CTD is over 350 residues in length and may adopt an extended or disordered structure (Meinhart et al., 2005).
Figure 4.Human Mediator undergoes a structural shift upon binding the pol II CTD. EM structures of unliganded Mediator (left) and CTD-bound Mediator (right) are shown. Note the CTD-Mediator sample is bound to native, full-length (52 YSPTSPS heptad repeat) mammalian CTD (Naar et al., 2002). (see colour version of this figure online at www.informahealthcare.com/bmg).


Human Mediator undergoes a structural shift upon binding the pol II CTD. EM structures of unliganded Mediator (left) and CTD-bound Mediator (right) are shown. Note the CTD-Mediator sample is bound to native, full-length (52 YSPTSPS heptad repeat) mammalian CTD (Naar et al., 2002). (see colour version of this figure online at www.informahealthcare.com/bmg).

In a breakthrough finding with yeast Mediator, the Kornberg lab was able to map at least a portion of the pol II CTD-Mediator interaction. By soaking a five-repeat CTD peptide into crystals of the seven-subunit Mediator head module, the Kornberg group was able to co-crystallize the pol II CTD bound to a portion of the Mediator complex for the first time (Robinson et al., 2012). The structure reveals that the CTD adopts an extended conformation (at least for the five-repeat domain used) and interacts with the Med6, Med8, and Med17 subunits. Also, the structure of the Mediator head module, which itself is conformationally flexible and dynamic (Cai et al., 2010), did not undergo significant re-organization upon pol II CTD binding, at least under these conditions (Robinson et al., 2012). This contrasts with pol II CTD binding to the human Mediator complex, which has been shown to trigger structural shifts upon binding (Figure 4) (Naar et al., 2002). Whereas the length of the CTD was different in these studies (five CTD repeats versus the 52 repeat sequence for human), this suggests a potential distinction in the binding interface or the activation mechanism. Another possible distinction is the recent observation by the Asturias group that, in S. cerevisiae, the pol II CTD interacts with a Mediator region distal from its assembly site in the PIC (Tsai et al., 2013).

Mediator not only binds the pol II CTD, but interacts extensively with the rest of the 12-subunit pol II complex as well (Soutourina et al., 2011). The pol II enzyme can bind the same general region – the head domain of Mediator – in human and yeast, and large structural shifts accompany pol II binding. This was first documented in yeast upon examination of 2D projections of EM data. The yeast Mediator structure appeared to unfold and extend upon pol II binding, and similar transitions were observed with murine Mediator complexes (Asturias et al., 1999). Also interesting were observations made with yeast pol II enzymes lacking the CTD. Yeast Mediator did not appear capable of stably binding pol II without the CTD; however, a CTD peptide was not able to induce structural unfolding (Asturias et al., 1999). Subsequent EM studies with yeast Mediator-pol II complexes have expanded upon these observations (Cai et al., 2009, 2010; Davis et al., 2002) and have suggested that the head domain of Mediator regulates movement of the pol II clamp during initiation, perhaps via interactions with the Rpb4/7 subunits (Cai et al., 2012).

Sweeping structural changes also accompany pol II binding to human Mediator, as shown in Figure 5 (Bernecky et al., 2011). Pol II binding induces structural reorganization throughout the complex, not simply at the pol II interaction site. Of interest is the structural shift in the leg/tail domain, as this represents a site of interaction for the CDK8 module (Knuesel et al., 2009a). Although speculative, it appears that pol II binding allosterically blocks CDK8 module binding at this distant site (Figure 5). This agrees with biochemical and functional studies that indicate mutually exclusive binding of CDK8 module or pol II to Mediator (Ebmeier & Taatjes, 2010; Knuesel et al., 2009a; Naar et al., 2002). Of course, the structural shift induced by pol II binding also implies a Mediator structural shift back upon pol II dissociation from Mediator, which presumably occurs during the pol II transition from an initiating or paused state to productive elongation (Core & Lis, 2008; Gilmour, 2009; Nechaev & Adelman, 2011).
Figure 5.Schematic outlining human Mediator structural changes induced by pol II-TFIIF binding. Two different views (front and side 1) are shown. Three Mediator domains (labeled 1, 2 and 3) are highlighted in the side 1 view and their putative locations are indicated following pol II-TFIIF binding. Note that structural re-organization occurs throughout the Mediator complex upon pol II-TFIIF binding, including the distal “leg/tail” domain (boxed), which represents a site for CDK8 module binding (Bernecky et al., 2011). The “Mediator only” and the “Mediator–pol II–TFIIF” structures each are bound to the activation domain of VP16. Pol II is shown in red (PDB 1Y1V). (see colour version of this figure online at www.informahealthcare.com/bmg).


Schematic outlining human Mediator structural changes induced by pol II-TFIIF binding. Two different views (front and side 1) are shown. Three Mediator domains (labeled 1, 2 and 3) are highlighted in the side 1 view and their putative locations are indicated following pol II-TFIIF binding. Note that structural re-organization occurs throughout the Mediator complex upon pol II-TFIIF binding, including the distal “leg/tail” domain (boxed), which represents a site for CDK8 module binding (Bernecky et al., 2011). The “Mediator only” and the “Mediator–pol II–TFIIF” structures each are bound to the activation domain of VP16. Pol II is shown in red (PDB 1Y1V). (see colour version of this figure online at www.informahealthcare.com/bmg).

A role for TFIIF in stabilizing the Mediator-pol II interaction was an unexpected finding from the cryo-EM studies with human Mediator-pol II assemblies (Bernecky et al., 2011). In the absence of TFIIF, pol II interacted with Mediator at the same head/body region, but did not stably orient itself. The inclusion of TFIIF in the human studies was based upon earlier work with the yeast head module of Mediator, in which a pol II-TFIIF complex was found to associate, whereas the head module did not interact with pol II alone (Takagi et al., 2006). It is not clear whether TFIIF serves a similar role in yeast, however (Rani et al., 2004).

Despite these structural data, it remains unclear what molecular contacts (i.e. among amino acid residues) are made between Mediator and pol II upon binding. The inherent flexibility of Mediator (Toth-Petroczy et al., 2008) and pol II (Kostek et al., 2006) has thus far limited the resolution of EM reconstructions. Apart from the Kornberg group’s crystal structure of the pol II CTD bound to the head module (Robinson et al., 2012), high-resolution structural details of the Mediator-pol II interaction are lacking. It is also not known how these interactions might change upon TF binding, which can induce global structural shifts in Mediator, in particular, at its pol II binding domain (see below).

As shown in Figure 6, the human Mediator complex undergoes substantial structural shifts upon interaction with the CDK8 module (CDK8, CCNC, MED12, MED13). Although the CDK8 module binds at the “leg” region of the complex, structural shifts occur throughout, including major re-organization in the head/middle region. As noted above, the head/middle region of the Mediator complex represents the pol II interaction site within the PIC. Biochemical experiments and MS data have confirmed that when bound to Mediator, the CDK8 module blocks pol II binding (Ebmeier & Taatjes, 2010; Knuesel et al., 2009a), including binding to the pol II CTD (Naar et al., 2002). Thus, a mutual allosteric block appears to contribute to pol II–CDK8 module antagonism. Although definitive confirmation in cells is practically and technically difficult, correlations have emerged that suggest mutually exclusive CDK8 module versus pol II occupancy at certain well-tested, inducible genes (Kim et al., 2006b; Mo et al., 2004; Pavri et al., 2005). As described later, this CDK8 module–pol II antagonism for binding Mediator may represent a key regulatory checkpoint.
Figure 6.CDK8 module–Mediator binding appears to occlude pol II–Mediator binding by an allosteric mechanism. EM structures of Mediator and CDK8-Mediator (both bound to the activation domain of VP16) are shown (Taatjes et al., 2002). The lower panel shows “bottom” views of each complex, with the dashed line on Mediator representing the surface that appears to make direct contacts with pol II (Bernecky et al., 2011). The bracket shows the general region occupied by pol II upon binding human Mediator, and the corresponding position in the CDK8-Mediator complex. The structural difference in this bracketed region may reflect a structural change important to prevent pol II (and pol II CTD) binding to CDK8-Mediator. (see colour version of this figure online at www.informahealthcare.com/bmg).


CDK8 module–Mediator binding appears to occlude pol II–Mediator binding by an allosteric mechanism. EM structures of Mediator and CDK8-Mediator (both bound to the activation domain of VP16) are shown (Taatjes et al., 2002). The lower panel shows “bottom” views of each complex, with the dashed line on Mediator representing the surface that appears to make direct contacts with pol II (Bernecky et al., 2011). The bracket shows the general region occupied by pol II upon binding human Mediator, and the corresponding position in the CDK8-Mediator complex. The structural difference in this bracketed region may reflect a structural change important to prevent pol II (and pol II CTD) binding to CDK8-Mediator. (see colour version of this figure online at www.informahealthcare.com/bmg).

The structural shifts that propagate through the human Mediator complex upon CDK8 module binding are not evident with yeast Mediator. A functional outcome, however, is shared in that yeast CDK8-Mediator does not bind the pol II enzyme (Myers et al., 1998; Spahr et al., 2003). In yeast, pol II binding is physically blocked by the Cdk8 module due to direct competition for Mediator surfaces involved in pol II binding. In S. cerevisiae, the Cdk8 module binds via its Med13 subunit, as observed with human CDK8 module (Knuesel et al., 2009a; Tsai et al., 2013). However, in S. cerevisiae, Cdk8 itself plays an auxiliary role by binding the middle module of Mediator. This interaction occludes an alternate site of pol II CTD binding, thus preventing Mediator-pol II association (Tsai et al., 2013). Examples of the distinct binding modes for the yeast Mediator Cdk8 module are shown in Figure 7. An interesting implication of these structural and biochemical studies is they suggest the presence of alternate modes of pol II–Mediator interaction (i.e. pol II binding at the middle module instead of the head module) in yeast. This could provide a means to sequester pol II in an inactive state, which can occur under conditions of limiting nutrients (Andrau et al., 2006). The Cdk8 module is actually degraded under these conditions, which could promote formation of such structural intermediates (Holstege et al., 1998). In S. pombe, the Cdk8 module directly blocks pol II binding, evidently by competing for similar sites on the Mediator complex (Elmlund et al., 2006). In contrast to budding yeast S pombe lack subunits that comprise the “tail” domain of yeast Mediator (Boube et al., 2002; Spahr et al., 2001), suggesting a requirement for a distinct mode of interaction.
Figure 7.Distinct modes of CDK8 module (CKM) binding to yeast Mediator. EM structure at left shows a single CKM-Mediator interaction via Med13, whereas the structure on the right shows a more extensive interface that also involves Cdk8 (Tsai et al., 2013). Scale bar: 100 Å. (see colour version of this figure online at www.informahealthcare.com/bmg).


Distinct modes of CDK8 module (CKM) binding to yeast Mediator. EM structure at left shows a single CKM-Mediator interaction via Med13, whereas the structure on the right shows a more extensive interface that also involves Cdk8 (Tsai et al., 2013). Scale bar: 100 Å. (see colour version of this figure online at www.informahealthcare.com/bmg).

In S. cerevisiae, the Cdk8 module subunits (srb8, srb9, srb10, srb11) were identified genetically as suppressors of growth phenotypes associated with truncations of the pol II CTD (Carlson, 1997). The ability of the S. cerevisiae Cdk8 module to physically block a newly discovered pol II CTD interaction site on Mediator provides an explanation (Tsai et al., 2013). Although pol II CTD truncations would negatively affect Mediator binding, mutations within Cdk8 module subunits (srb8-11) would promote pol II CTD-Mediator binding, thus suppressing the transcriptional defect of CTD truncation.

Gene expression patterns are regulated in large part by DNA-binding TFs (Lee & Young, 2013). It is widely understood that TFs activate or repress transcription by somehow affecting pol II activity. Yet, in eukaryotic cells, TFs do not bind pol II; instead, they bind factors that control pol II activity directly (e.g. Mediator) or indirectly (e.g. chromatin remodeling complexes). Because Mediator interacts extensively with pol II, it represents perhaps the most functionally important factor through which TFs regulate transcription.

EM studies with human Mediator complexes revealed a surprising discovery: the structure of the complex changed markedly upon TF binding. This was first observed by structural comparison of Mediator itself (purified with epitope-tagged MED26) with Mediator complexes bound to the activation domain of SREBP or VP16 (purified using GST-SREBP or GST-VP16). As shown in Figure 8, the structural differences are substantial and propagate throughout the entire complex, despite localization of TF binding to a single site (Taatjes et al., 2002). That binding of a single TF activation domain (typically ∼50 residues in length) could trigger such sweeping conformational changes was difficult to comprehend. However, follow-up experiments confirmed that the TF activation domain alone was sufficient: the structural state of the “activator free” Mediator sample could be controlled by simply adding the VP16 or SREBP activation domain. Incubation of activator-free Mediator with GST-VP16 induced the VP16-Mediator structural state, whereas incubation with GST-SREBP induced the SREBP-Mediator structural state (Taatjes et al., 2002). Subsequent experiments extended these observations with other TFs (Meyer et al., 2010; Taatjes et al., 2004) and confirmed that Mediator subunit composition did not change with these structural transitions (Ebmeier & Taatjes, 2010). A general conclusion from these studies was that TFs that interacted with different subunits or surfaces on Mediator could induce different structural shifts upon binding.
Figure 8.TF binding induces structural shifts throughout the human Mediator complex. Similar views of EM structures of Mediator without a TF bound (top), or bound to the activation domain of SREBP or VP16 are shown (Taatjes et al., 2002). Note that structural changes appear to propagate throughout the complex, and that structural changes are distinct for each TF. Localization of the VP16 (X) and SREBP (+) binding sites are shown. Scale bar: 100 Å. (see colour version of this figure online at www.informahealthcare.com/bmg).


TF binding induces structural shifts throughout the human Mediator complex. Similar views of EM structures of Mediator without a TF bound (top), or bound to the activation domain of SREBP or VP16 are shown (Taatjes et al., 2002). Note that structural changes appear to propagate throughout the complex, and that structural changes are distinct for each TF. Localization of the VP16 (X) and SREBP (+) binding sites are shown. Scale bar: 100 Å. (see colour version of this figure online at www.informahealthcare.com/bmg).

Much remains to be uncovered with respect to how TF-induced structural changes affect Mediator function. Currently, it appears that TF-directed structural shifts may regulate gene expression by (1) altering Mediator–pol II interactions to activate the pol II enzyme within the PIC, and (2) regulating the timing and genomic location of key Mediator-cofactor interactions. TF-Mediator binding was shown to stabilize pol II orientation, based upon comparative cryo-EM structural studies with Mediator-pol II-TFIIF complexes in the presence or absence of the VP16 activation domain (Bernecky & Taatjes, 2012). Specific TF-induced structural shifts also correlate with activation of pol II within the PIC, at least in the case of p53 (Meyer et al., 2010). By examining PIC formation, gene expression, and Mediator structure in the presence of wild-type or mutant p53, Meyer et al. linked not only factor recruitment, but also Mediator structural shifts, as essential for activated transcription (Meyer et al., 2010). Similar observations were made by the Berk lab, in which activation of pol II bound at the Egr1 promoter was mechanistically linked to a phosphorylation-dependent switch in the ELK1-MED23 interaction (Balamotis et al., 2009). These findings imply that Mediator can adopt an “active” structural state upon TF binding that can trigger changes in pol II function (Wang et al., 2012). This model fits well with “post-recruitment” mechanisms of gene activation (e.g. activation of paused pol II complexes) that predominate in higher organisms (Core & Lis, 2008).

TF-induced structural shifts may also enable Mediator – a general transcription factor – to adopt gene-specific functionality. Because different TFs induce different structural shifts upon binding Mediator, different protein surfaces are likely exposed that could mediate distinct protein-protein interactions. This concept was supported by proteomics studies of Mediator in different TF-bound structural states (Ebmeier & Taatjes, 2010), in which different co-regulatory factors were found to associate with Mediator in its different structural states.

The scope of the structural changes imply a coordinated set of movements among numerous (perhaps a majority) Mediator subunits. Such coordination has been described with a multiple allosteric network model, in which a structural shift at one site propagates throughout a network of protein subunits (Lewis, 2010). This model also suggests how an interconnected protein network such as Mediator could enable such dramatic structural transitions in the absence of ATP hydrolysis (Bray & Duke, 2004). Structural changes induced by TF binding are substantial, as they can be clearly detected from even low-resolution data. The scope of the structural changes could also result from coordinated movement of large domains – perhaps comprised of multiple subunits – by dissociation at one site and re-association at another, analogous to the structural re-arrangement observed with human TFIID (Cianfrocco et al., 2013).

The Mediator structural changes outlined above involve what appear to be coordinated and robust structural shifts throughout the complex. Moreover, the conformational shifts are distinct based upon whether pol II, CDK8 module, or TFs bind the Mediator complex. This suggests a straightforward mechanism to regulate Mediator activity, summarized schematically in Figure 9. Note that in some circumstances, Mediator is rendered incapable of specific interactions (e.g. the CDK8 module does not interact with Mediator in its pol II-bound structural state). This could be important to ensure appropriate timing of events during various stages of transcription.
Figure 9.A working model for Mediator and CDK8-Mediator regulation of transcription initiation and elongation. This model depicts four functionally distinct structural states (I–IV) for Mediator. We hypothesize that different Mediator surfaces will be exposed in each state, which may help coordinate timing of factor recruitment to the promoter, in accordance with requirements for various stages of transcription. According to this model, state I and state II are compatible with pre-initiation events, state III represents transcription initiation (possibly including paused pol II), and state IV represents an elongation-competent structure. In state I, Mediator is not bound to a TF; Mediator is capable of binding pol II in this structural state, but pol II will be inactive or minimally active (i.e. basal transcription). TF binding (e.g. VP16) causes a structural shift to state II. Mediator is also capable of binding pol II in this conformational state, with the potential to direct high levels of “activated” transcription. This structural state might also coordinate timing of other Mediator-cofactor interactions at the promoter that could regulate subsequent stages of transcription (Ebmeier & Taatjes, 2010). If pol II binds the TF-Mediator complex, this leads to structural state III. This structural state may be compatible with activated transcription, perhaps by promoting synergy among PIC factors (e.g. TFIIH, TFIID and TFIIB) that assemble around the Mediator–pol II complex. Note that in this structural state, the CDK8 module is incapable of binding Mediator. Upon transcription initiation and pol II transition to productive elongation, pol II breaks contacts with Mediator; Mediator structure transitions back to state II (TF bound, but no pol II). The CDK8 module is able to bind Mediator in this structural state. If the CDK8 module binds Mediator, Mediator adopts structural state IV. This structural state (i.e. CDK8-Mediator) does not allow pol II binding. Thus, the CDK8-Mediator complex prevents a second pol II enzyme from immediately re-engaging the promoter, which might otherwise cause defects in mRNA processing or defects during initiation by this second pol II. Furthermore, the CDK8-Mediator complex could help assemble and/or regulate elongation factors, thereby influencing ongoing elongation events. The ability of CDK8-Mediator or core Mediator (i.e. Mediator containing MED26) to positively influence pol II elongation has been documented by several groups (Donner et al., 2010; Galbraith et al., 2013; Takahashi et al., 2011). Yet Mediator and other PIC components remain at the promoter following pol II promoter escape, leaving a “scaffold” complex (Yudkovsky et al., 2000). These apparently contradictory findings are reconciled by growing evidence that elongating pol II complexes are likely stationary, and that rather than moving directionally along DNA, pol II instead “reels in” the DNA template (Papantonis et al., 2010). This has already been demonstrated for bacterial polymerases (Kapanidis et al., 2006; Revyakin et al., 2006), and DNA polymerases work in much the same way (Anachkova et al., 2005). Stationary, elongating pol II complexes could be juxtaposed with promoter-bound factors, facilitating Mediator- or CDK8-Mediator-dependent regulation of pol II elongation. We emphasize that this is a model, and that many aspects remain to be rigorously tested. (see colour version of this figure online at www.informahealthcare.com/bmg).


A working model for Mediator and CDK8-Mediator regulation of transcription initiation and elongation. This model depicts four functionally distinct structural states (I–IV) for Mediator. We hypothesize that different Mediator surfaces will be exposed in each state, which may help coordinate timing of factor recruitment to the promoter, in accordance with requirements for various stages of transcription. According to this model, state I and state II are compatible with pre-initiation events, state III represents transcription initiation (possibly including paused pol II), and state IV represents an elongation-competent structure. In state I, Mediator is not bound to a TF; Mediator is capable of binding pol II in this structural state, but pol II will be inactive or minimally active (i.e. basal transcription). TF binding (e.g. VP16) causes a structural shift to state II. Mediator is also capable of binding pol II in this conformational state, with the potential to direct high levels of “activated” transcription. This structural state might also coordinate timing of other Mediator-cofactor interactions at the promoter that could regulate subsequent stages of transcription (Ebmeier & Taatjes, 2010). If pol II binds the TF-Mediator complex, this leads to structural state III. This structural state may be compatible with activated transcription, perhaps by promoting synergy among PIC factors (e.g. TFIIH, TFIID and TFIIB) that assemble around the Mediator–pol II complex. Note that in this structural state, the CDK8 module is incapable of binding Mediator. Upon transcription initiation and pol II transition to productive elongation, pol II breaks contacts with Mediator; Mediator structure transitions back to state II (TF bound, but no pol II). The CDK8 module is able to bind Mediator in this structural state. If the CDK8 module binds Mediator, Mediator adopts structural state IV. This structural state (i.e. CDK8-Mediator) does not allow pol II binding. Thus, the CDK8-Mediator complex prevents a second pol II enzyme from immediately re-engaging the promoter, which might otherwise cause defects in mRNA processing or defects during initiation by this second pol II. Furthermore, the CDK8-Mediator complex could help assemble and/or regulate elongation factors, thereby influencing ongoing elongation events. The ability of CDK8-Mediator or core Mediator (i.e. Mediator containing MED26) to positively influence pol II elongation has been documented by several groups (Donner et al., 2010; Galbraith et al., 2013; Takahashi et al., 2011). Yet Mediator and other PIC components remain at the promoter following pol II promoter escape, leaving a “scaffold” complex (Yudkovsky et al., 2000). These apparently contradictory findings are reconciled by growing evidence that elongating pol II complexes are likely stationary, and that rather than moving directionally along DNA, pol II instead “reels in” the DNA template (Papantonis et al., 2010). This has already been demonstrated for bacterial polymerases (Kapanidis et al., 2006; Revyakin et al., 2006), and DNA polymerases work in much the same way (Anachkova et al., 2005). Stationary, elongating pol II complexes could be juxtaposed with promoter-bound factors, facilitating Mediator- or CDK8-Mediator-dependent regulation of pol II elongation. We emphasize that this is a model, and that many aspects remain to be rigorously tested. (see colour version of this figure online at www.informahealthcare.com/bmg).

Early studies of Mediator in both yeast and human cells zeroed in on one its most basic functions: an ability to stabilize or facilitate PIC formation (Cantin et al., 2003; Koleske et al., 1992; Ranish et al., 1999; Wu et al., 2003). In fact, simply tethering a Mediator subunit to a DNA-binding domain could promote PIC formation and activate transcription in yeast (Balciunas et al., 1999; Cheng et al., 2002; Young et al., 2008). The central role of Mediator in PIC structure and function is best reflected by the fact that every PIC factor (TFIIA, TFIIB, TFIID, TFIIE, TFIIF, TFIIH and pol II itself) has been physically and/or functionally linked to Mediator, often in studies in both yeast and human cells. Additional transcription regulators that could be considered auxiliary PIC factors have been physically and/or functionally linked to Mediator. These include TFIIS/TCEA1, Gdown1/POLR2M, NC2/DR1, BRD4, cohesin, DSIF, P-TEFb, p300, and PC4/SUB1. We discuss Mediator–PIC interactions and focus on several auxiliary factors in the following sections.

The TATA-binding protein (TBP) is sometimes considered a surrogate for the 15+ subunit TFIID complex. TFIIA and TFIIB each interact with TBP – a DNA-binding subunit within TFIID – in the PIC (Geiger et al., 1996; Nikolov et al., 1995; Tan et al., 1996). Therefore, these three factors are considered together in this section.

The Carey group has been instrumental in demonstrating functional coordination between Mediator and TFIID. Using immobilized template assays and extracts depleted or supplemented with purified factors, Mediator was shown to coordinate TFIID binding to promoter DNA (Johnson et al., 2002) and to promote synergistic PIC assembly on chromatin templates modified by the global co-activator p300 (Black et al., 2006). The Carey lab also demonstrated synergy in DNA binding of TFIID-TFIIA assemblies with Mediator (Johnson & Carey, 2003) that appear to support recent structural data that indicate TFIIA-directed structural re-arrangement of TFIID upon DNA binding (Cianfrocco et al., 2013).

The Roeder lab has uncovered numerous examples of functional synergy between Mediator and TFIID (Guermah et al., 1998, 2001). In a pair of detailed studies, Baek et al. demonstrated that Mediator contributed to stable recruitment of TFIIB, TFIID and TFIIE to gene promoters and also regulated the activities of these factors during transcription initiation (Baek et al., 2002, 2006). Interestingly, these activities were shown to be largely independent of an activator, revealing a role for Mediator even in basal transcription; a role for Mediator in basal transcription was uncovered by several other labs as well (Mittler et al., 2001; Takagi & Kornberg, 2006; Wang et al., 2013), and likely results from its general role as a structural scaffold for PIC assembly (described below).

Another study highlighting Mediator-TFIID functional interdependence was completed by the Tjian group. Using in vitro and knockdown analyses (S2 cells) for basal and activated transcription, Marr et al. discovered that TFIID and Mediator functioned interdependently. In fact, at inducible genes responsive to the MTF-1 transcription factor, Mediator acted as a checkpoint for gene activation and TFIID activity (Marr et al., 2006). This study also revealed an elaborate functional relationship among different Mediator subunits at genes regulated by the same TF; this led the authors to suggest that loss of specific Mediator subunits could influence potential promoter-selective activities or differentially impact transduction of the TF activation signal to the PIC (Marr et al., 2006). Clearly, much more needs to be resolved about the mechanisms driving functional cooperativity or antagonism among select Mediator and TFIID subunits. Adding to the complexity, cooperative or antagonistic functions likely involve additional factors. The Martinez lab, for example, has shown that negative regulation by NC2/DR1 and Topoisomerase I (TOP1MT) is countered by Mediator and TFIID (Xu et al., 2011a).

Taken together, these findings suggest a direct interaction between Mediator and TFIID. This was convincingly demonstrated by the Conaway lab in 2011. Using a combination of biochemical and proteomics experiments, Takahashi et al. identified a direct interaction between TFIID and MED26; interestingly, the MED26–TFIID interaction was not essential for TFIID recruitment, but rather appeared to regulate timing of MED26 interaction with elongation factors (Takahashi et al., 2011).

Cooperativity between Mediator and TFIID has also been observed in yeast (Koleske et al., 1992). Genetic experiments have demonstrated that Mediator subunit mutations can result in defective TFIID recruitment (Lim et al., 2007; Takahashi et al., 2009). Also, the Green lab demonstrated synergy between TFs, Mediator, TBP and TFIIB that occurred in part by a TF-induced structural change attributed to TFIIB (Li et al., 1999).

Finally, the SAGA complex, which is structurally related to TFIID (Wu et al., 2004), has been shown to functionally cooperate with Mediator (Larschan & Winston, 2005). The Martinez lab characterized a Mediator interaction surface within SAGA (SUPT7L) that facilitated MYC-dependent gene activation (Liu et al., 2008). A genetic study in yeast, completed by the Morse lab, indicated an intriguing link between Mediator tail module subunits and regulation of SAGA-dependent genes (Ansari et al., 2012). Because promoters of SAGA-dependent genes typically contain the TATA sequence (whereas TFIID-dependent genes do not) (Basehoar et al., 2004), this study suggests mechanisms by which Mediator might adopt promoter-specific functions.

TFIIE and TFIIH directly interact (Maxon et al., 1994), and TFIIE helps regulate TFIIH activity and assembly into the PIC (Ohkuma & Roeder, 1994; Serizawa et al., 1994). TFIIH is a 10-subunit complex that possesses ATPase, helicase and kinase activities that are important for pol II transcription (Compe & Egly, 2012). The kinase within TFIIH, CDK7, is conserved from yeast to humans and phosphorylates the pol II CTD during transcription initiation. Among other things, phosphorylation of the pol II CTD disrupts CTD-Mediator binding, likely facilitating the transition from initiation to elongation (Max et al., 2007; Svejstrup et al., 1997). Many genetic links between Mediator, TFIIE, and/or TFIIH have been made in model organisms (Sakurai & Fukasawa, 1998, 2000; Sakurai et al., 1996). Biochemical and genetic studies in yeast have linked the tail module subunit Med15 (Gal11) to stable binding of TFIIE and TFIIH (Badi & Barberis, 2001; Sakurai & Fukasawa, 1997, 2003). As this subunit is separated from putative TFIIE/TFIIH assembly sites within the yeast PIC (Imasaki et al., 2011), these findings suggest a potential allosteric mechanism.

Because Mediator binds the unphosphorylated pol II CTD, this likely contributes to the Mediator-dependent stimulation of TFIIH kinase activity toward the CTD within the PIC. Mediator was first shown to enhance TFIIH phosphorylation of the Pol II CTD 12-fold in a yeast reconstituted transcription system consisting of pol II and basal factors (Kim et al., 1994). This activity was later demonstrated in mammals (Jiang et al., 1998). Consistent with its role as an architectural factor, Mediator stabilizes TFIIH assembly into the PIC (Guidi et al., 2004; Nair et al., 2005). A direct interaction between Mediator subunit Med11 and TFIIH has been documented by both the Cramer and Werner labs. The Cramer group performed structural and functional mutagenesis studies, whereas the Werner group examined global gene expression and global recruitment of TFIIH in yeast expressing Med11 mutants (Esnault et al., 2008; Seizl et al., 2011). Work by the Myers group determined a key role for the Med19 subunit (middle module subunit of yeast Mediator) in transducing activation by TFs and promoting TFIIH phosphorylation of the pol II CTD (Baidoobonso et al., 2007). These findings have been supported by in vitro studies with p53 and human Mediator (Meyer et al., 2010). A potential role for DNA-binding TFs in regulating pol II CTD phosphorylation by Mediator–TFIIH is intriguing, in part because it is consistent with an early observation that enhancer-dependent transcription appears especially sensitive to pol II CTD truncations (Gerber et al., 1995).

A host of genetic and biochemical studies demonstrated Mediator interaction with pol II; such studies were among the first to identify the Mediator complex in yeast (Kim et al., 1994; Nonet & Young, 1989; Thompson et al., 1993). Many of these reports focused on the pol II CTD, which binds yeast or human Mediator with apparent high affinity (Myers et al., 1998; Naar et al., 2002). Genetic interactions were observed between Mediator and other pol II subunits, however, suggesting a more extensive interaction between Mediator and pol II (Reeves & Hahn, 2003; Soutourina et al., 2011). This was confirmed with EM studies of Mediator-pol II complexes (Bernecky et al., 2011; Davis et al., 2002).

A functionally distinct module within pol II, consisting of the RPB4 and RPB7 subunits, forms a “stalk” that guides nascent RNA from the transcribing pol II enzyme. Interestingly, the Rpb4/7 subunits are essential in S. pombe, but not in the budding yeast S. cerevisiae (Choder & Young, 1993; Sakurai et al., 1999). In S. pombe, genetic interactions have been identified between the pol II Rpb4 subunit and the Med31 and Med8 subunits. In fact, Rpb4 knockdown shows similar phenotypes to Med8 or Med31 mutants, suggesting cooperative functions (Sharma et al., 2006). These phenotypes also mimic Cdk7 (Kin28) or Mat1 mutant yeast, which represent TFIIH subunits (Lee et al., 2005b). Structural data with the yeast Mediator (S. cerevisiae) head module support a physical interaction with Rpb4/7 (Cai et al., 2010) and suggest a means by which Mediator could facilitate transcription initiation (Cai et al., 2012).

TFIIF forms a stable complex with the pol II enzyme (Bushnell et al., 1996; Tan et al., 1994), and both complexes appear to assemble into the PIC as a unit (Rani et al., 2004). Whereas direct Mediator-TFIIF binding has not been convincingly demonstrated, it is notable that TFIIF stabilizes pol II orientation within a TF-bound Mediator–pol II–TFIIF assembly (Bernecky et al., 2011). Furthermore, a pol II-TFIIF complex, but not pol II alone, was shown to stably associate with the head module of yeast Mediator (Takagi et al., 2006). These results suggest that TFIIF might make additional contacts with Mediator when bound to pol II, or that TFIIF induces a pol II conformation that allows a different and more stable interaction with Mediator.

Structural studies with yeast and human Mediator–pol II complexes have indicated that pol II binds at a similar site at the head region of Mediator (Asturias et al., 1999; Bernecky et al., 2011; Davis et al., 2002). The orientation of pol II, however, has been different with yeast Mediator compared with human. This discrepancy could reflect true biological differences in PIC structure. Yeast and humans are separated by perhaps 2 billion years on the evolutionary timescale (Figure 1) and Mediator sequences are poorly conserved (Table 1); therefore, its interactions with pol II and its activation mechanism may be different in yeast compared with humans. Also, various transient interaction intermediates have been observed with yeast Mediator-pol II complexes (Tsai et al., 2013), suggesting an association that is distinct from humans.

We hypothesize, however, that the current discrepancies in yeast and human Mediator–pol II structures could simply reflect the fact that the composition of the Mediator-pol II assemblies have been different (Bernecky et al., 2011). Cryo-EM analyses of human Mediator–pol II complexes were completed in the presence and absence of a TF activation domain (VP16) and in the presence and absence of TFIIF (Bernecky et al., 2011; Bernecky & Taatjes, 2012). In the absence of TFIIF, pol II binds Mediator, but it does not stably orient itself; similarly, in the absence of a TF (VP16), a pol II-TFIIF complex binds Mediator, but does not adopt a stable orientation. Required for pol II to stably orient was (1) TF-Mediator binding and (2) the presence of TFIIF. These observations implicate structural differences – stable versus variable pol II orientation – in the ability of TF-Mediator binding to direct high levels of “activated” transcription (TF-dependent) versus low level “basal” transcription (TF-independent). Structural studies with yeast Mediator–pol II complexes have been completed in the absence of a TF and TFIIF and have examined partial assemblies of Mediator or pol II. Further structural studies of yeast Mediator with pol II-TFIIF and/or a TF activation domain should determine whether TFs and TFIIF serve similar structural roles in yeast. Ultimately, however, it will be important to evaluate how TFIIF and TF-Mediator binding affect pol II orientation within the entire PIC. Such experiments appear feasible only with cryo-EM.

Recently, the Nogales lab completed a cryo-EM analysis of a partial PIC containing TBP, TFIIA, TFIIB, TFIIE, TFIIF, TFIIH and pol II bound to promoter DNA (He et al., 2013). Docking existing crystal structure data within this large cryo-EM structural map revealed much about the overall architecture of the human PIC at pseudo-atomic level resolution. In Figure 10, we have merged this partial PIC structure with the human Mediator–pol II–TFIIF structure, which was also generated using cryo-EM and single particle reconstruction techniques (Bernecky et al., 2011). Although speculative, the two models appear complementary and suggest how a fully assembled, active PIC might be organized. A major component lacking from the model in Figure 10 is the TFIID complex. Given the large size of TFIID and its well-documented structural dynamics (Cianfrocco et al., 2013; Grob et al., 2006), several possibilities can be envisioned for how TFIID might assemble.
Figure 10.A structural model of the human PIC. The cryo-EM structure of human pol II, TBP, TFIIA, TFIIB, TFIIE and TFIIF bound to promoter DNA (closed complex (He et al., 2013)) was docked into the cryo-EM map of human Mediator–pol II–TFIIF (Bernecky et al., 2011). In the docked structures, the Mediator-pol II-TFIIF cryo-EM map is shown in blue mesh, whereas the color-coding for the other PIC factors is indicated. For reference, the same orientation of the Mediator–pol II–TFIIF structure alone is shown in solid blue. Addition of TFIIH (pink) to the model blocks details of the structure, therefore, we show the model with and without TFIIH (below). The view without TFIIH also indicates an open region for its assembly into the PIC. Note that some structural reorganization occurs within the PIC upon TFIIH binding (He et al., 2013). To generate the model, the docked pol II crystal structure was used as a reference to align both cryo-EM maps in Chimera. (see colour version of this figure online at www.informahealthcare.com/bmg).


A structural model of the human PIC. The cryo-EM structure of human pol II, TBP, TFIIA, TFIIB, TFIIE and TFIIF bound to promoter DNA (closed complex (He et al., 2013)) was docked into the cryo-EM map of human Mediator–pol II–TFIIF (Bernecky et al., 2011). In the docked structures, the Mediator-pol II-TFIIF cryo-EM map is shown in blue mesh, whereas the color-coding for the other PIC factors is indicated. For reference, the same orientation of the Mediator–pol II–TFIIF structure alone is shown in solid blue. Addition of TFIIH (pink) to the model blocks details of the structure, therefore, we show the model with and without TFIIH (below). The view without TFIIH also indicates an open region for its assembly into the PIC. Note that some structural reorganization occurs within the PIC upon TFIIH binding (He et al., 2013). To generate the model, the docked pol II crystal structure was used as a reference to align both cryo-EM maps in Chimera. (see colour version of this figure online at www.informahealthcare.com/bmg).

Although the PIC model shown in Figure 10 is speculative and will likely be revised once additional data with larger PIC assemblies are obtained, it illustrates several important points. One is the physical size of the PIC and the extended surface area for protein-protein and protein-nucleic acid interactions. A second point is the central role for Mediator as a scaffold about which the rest of the PIC assembles. Third, within the fully assembled PIC, a majority of Mediator (and TFIID, incidentally) remains exposed, ostensibly to mediate interactions with other architectural or regulatory factors. Finally, the PIC model emphasizes the tightly packed nature of the PIC. Within such a tightly packed assembly, structural shifts of the scale that occur upon TF-Mediator binding (Figure 8) could be expected to trigger substantial re-organization of Mediator-PIC contacts. We postulate that such structural re-organization is a fundamental mechanism by which DNA-binding TFs activate transcription. Many genes appear to have Mediator, pol II, TFIID, and other GTFs pre-loaded at transcription start sites, yet high level or “activated” transcription does not occur until a key TF binds the promoter (typically in response to activation of a signaling pathway). In other words, the PIC appears to adopt an inactive, latent state that is poised to become activated by pathway-specific TFs.

Among the TF-Mediator complexes examined thus far using EM, each has induced large-scale conformational changes upon binding, and the structural shift has been linked to activation of transcription (Meyer et al., 2010). Whereas the TF-induced structural states can be distinct, a common structural shift occurs at the Mediator–pol II interaction site (Figure 11). This shared structural feature among distinct TF-bound Mediator complexes suggests a common activation mechanism. Unfortunately, the low structural resolution cannot delineate whether similar Mediator surfaces are exposed for pol II binding in each case, and future work will be needed to address this key question.
Figure 11.A common structural feature among TF-bound human Mediator complexes. EM structures are shown for Mediator bound to different TF activation domains and compared with Mediator that is not TF-bound (inset). A shared structural feature is a “pocket” (green arrow) and the surfaces corresponding to probable sites of pol II interaction are highlighted with the dashed yellow line. Note these features are absent from the unliganded Mediator structure. (see colour version of this figure online at www.informahealthcare.com/bmg).


A common structural feature among TF-bound human Mediator complexes. EM structures are shown for Mediator bound to different TF activation domains and compared with Mediator that is not TF-bound (inset). A shared structural feature is a “pocket” (green arrow) and the surfaces corresponding to probable sites of pol II interaction are highlighted with the dashed yellow line. Note these features are absent from the unliganded Mediator structure. (see colour version of this figure online at www.informahealthcare.com/bmg).

Early models of gene regulation by yeast Mediator centered on the importance of pol II recruitment (Keaveney & Struhl, 1998; Ptashne & Gann, 1997). Mediator occupancy correlated with pol II occupancy and assembly of stable pre-initiation complexes. Moreover, tethering select Mediator subunits to DNA-binding domains was often sufficient for PIC assembly and activation of transcription (Balciunas et al., 1999; Cheng et al., 2002; Young et al., 2008). Because a vast array of TFs bind (i.e. recruit) Mediator, it is clear that a basic function of TFs is to help recruit Mediator (and other PIC components) to gene promoters or enhancers. Further mechanistic studies supported this model, but have revealed additional aspects that appear equally important for regulating transcription, at least in metazoans. This includes the prevalence of paused pol II complexes as regulatory intermediates (Core et al., 2008; Guenther et al., 2007; Muse et al., 2007; Seila et al., 2008; Zeitlinger et al., 2007). Whereas paused pol II complexes are a major regulatory intermediate in human cells, this does not appear to be the case in yeast or worms, which lack NELF (Peterlin & Price, 2006). Mediator appears to regulate paused pol II complexes, although the molecular mechanisms remain incompletely understood (Balamotis et al., 2009; Galbraith et al., 2013; Knuesel & Taatjes, 2011; Meyer et al., 2010; Takahashi et al., 2011; Wang et al., 2005a).

Regulation of promoter-bound, paused pol II complexes represents a divergence in Mediator function in higher organisms, with perhaps a few exceptions (Lee et al., 2010b). Several differences between yeast and mammalian transcription appear to contribute. A role for MED26 in activating paused pol II fits with its emergence in metazoan organisms (Takahashi et al., 2011). Pausing/pause release factors such as DSIF and Gdown1/POL2RM display strong functional synergy with mammalian Mediator (Cheng et al., 2012; Hu et al., 2006; Jishage et al., 2012; Malik et al., 2007), whereas similar roles are not evident in yeast (yeast lack a Gdown1 ortholog). Cohesin has emerged as a regulator of pol II pausing/pause release (Fay et al., 2011; Schaaf et al., 2013), and functional coordination between Mediator and cohesin appears specific to metazoans (Kagey et al., 2010; Phillips-Cremins et al., 2013). Finally, the mechanistic links between TF-induced structural changes and activation of paused pol II may also represent a divergent activation mechanism for human versus yeast Mediator. Whereas yeast Mediator is conformationally flexible, it remains to be determined whether TFs induce structural changes in yeast Mediator. The ability of Mediator to activate transcription beyond pol II recruitment and PIC assembly – that is, to activate pol II after it has been recruited to the PIC – has been directly tied to TF binding (Balamotis et al., 2009; Malik et al., 2002; Meyer et al., 2010; Park et al., 2001b; Wang et al., 2005a). We hypothesize that the factors emerging as regulators of pol II pausing and pause release (e.g. cohesin, Gdown1, MED26, P-TEFb) are, at least in part, regulated via structural shifts in Mediator that are triggered upon binding an external factor, such as a TF. A scheme summarizing this working model is shown in Figure 12.
Figure 12.A model for TF-dependent “post-recruitment” activation of a fully assembled but latent PIC. In the absence of a key TF, a PIC may occupy the promoter, but pol II remains largely inactive or paused. Upon TF binding to the promoter, it interacts with Mediator and triggers a structural shift in the complex, which activates the PIC and allows pol II to escape the promoter region and transition to a productively elongating state. Part of this process could involve functional synergy between Mediator and pausing/elongation factors such as DSIF, Gdown1/POLR2M or the SEC. (see colour version of this figure online at www.informahealthcare.com/bmg).


A model for TF-dependent “post-recruitment” activation of a fully assembled but latent PIC. In the absence of a key TF, a PIC may occupy the promoter, but pol II remains largely inactive or paused. Upon TF binding to the promoter, it interacts with Mediator and triggers a structural shift in the complex, which activates the PIC and allows pol II to escape the promoter region and transition to a productively elongating state. Part of this process could involve functional synergy between Mediator and pausing/elongation factors such as DSIF, Gdown1/POLR2M or the SEC. (see colour version of this figure online at www.informahealthcare.com/bmg).

Emerging evidence for Mediator involvement in transcription elongation suggests a broader regulatory role in gene expression (Conaway & Conaway, 2013). An indication that metazoan Mediator activity extended beyond transcription initiation came from studies of Drosophila heat shock genes, in which paused pol II engaged in active elongation upon heat shock-induced recruitment of HSF and Mediator (Park et al., 2001b). A direct interaction between the HSF transcription factor and Mediator was demonstrated, and both HSF and Mediator recruitment to HSF target genes occurred in a rapid and coordinated fashion upon heat shock (independently of other PIC factors). The authors concluded that the HSF-Mediator interaction triggered activation of paused pol II (Park et al., 2001b). In vitro studies by the Roeder lab and studies in murine embryonic stem cells by the Berk group showed further evidence for Mediator in “post-recruitment” or elongation events (Malik et al., 2002; Wang et al., 2005a). In each study, Mediator recruitment by a TF (HNF4 or ELK1) correlated mainly with activation of transcription rather than pol II recruitment per se. Mediator is also detected by ChIP in the body of genes (in addition to gene promoters) in human cells (Donner et al., 2007, 2010; Takahashi et al., 2011), suggesting some type of interaction (direct or indirect) with the coding region during transcription elongation.

Whereas yeast appear to lack paused pol II complexes as a regulatory intermediate, links between Mediator and elongation in yeast have been uncovered with genetic and biochemical experiments (Gaillard et al., 2009; Kremer et al., 2012; Rodriguez-Gil et al., 2010). And, as in human cells, ChIP experiments have detected Mediator in the coding region of yeast genes (Andrau et al., 2006; Zhu et al., 2006), although these findings are not always observed (Kim & Gross, 2013) and could be considered controversial (Fan & Struhl, 2009). Taken together, these findings support a conserved role for Mediator in regulating elongating pol II. The mechanisms by which Mediator regulates elongation events, however, appear to have expanded in metazoans. Below, we summarize Mediator interactions with elongation factors and focus on TFIIS, Gdown1/POLR2M and the super elongation complex (SEC).

TFIIS is an elongation factor that interacts with pol II (Kettenberger et al., 2004) and stimulates RNA cleavage for stalled or terminated transcripts. In yeast, TFIIS (Dst1) has been shown to interact with Mediator via an N-terminal domain not involved in pol II binding (Wery et al., 2004). TFIIS genetically interacts with Med18 (Srb5), Med16 (Sin4), and Med31 (Soh1) (Malagon et al., 2004); studies by the Ranish and Werner labs established a role for Mediator in TFIIS recruitment to the PIC, with further genetic links to Med31 (Guglielmi et al., 2007; Kim et al., 2007). In mammalian cells, the Malik lab established a role for Mediator and TFIIS in pol II transcription through the +1 nucleosome (Nock et al., 2012). The Lis lab, studying the role of TFIIS in Drosophila, observed that TFIIS was required for release of paused pol II, but not for generating the paused complex (Adelman et al., 2005). This is consistent with the findings of Nock et al. in that Mediator-TFIIS function promoted pol II transition from an initiating to a productively elongating state (Nock et al., 2012).

Gdown1 appears to regulate pol II pausing; like NELF, a well-established regulator of paused pol II complexes, Gdown1 orthologs are absent in yeast and worms, but are present in Drosophila and mammals (Cheng et al., 2012). A role for Mediator in regulating Gdown1 was originally uncovered by the Gnatt lab, who observed that Gdown1 negatively regulated pol II activity, but this negative effect could be overcome by Mediator (Hu et al., 2006). The Price and Roeder labs completed a pair of studies that have further defined the functional interplay between Mediator and the pol II-associated factor Gdown1 (Cheng et al., 2012; Jishage et al., 2012). Both groups confirmed that Gdown1 blocks the binding of TFIIF to pol II, and that Mediator was critical to overcome this block. Based upon in vitro and genome-wide ChIP-Seq analyses, Gdown1 appears important for stabilizing paused pol II and preventing premature termination (Cheng et al., 2012). A model in which Mediator remodels or modifies Gdown1 to allow TFIIF-pol II binding was proposed, based in part upon the fact that Gdown1 does not dissociate from elongating pol II complexes.

Cryo-EM studies have provided a structural understanding for this TFIIF-Gdown1 antagonism and are consistent with a central role for Mediator. Analysis of the Gdown1-pol II complex revealed Gdown1 binding centered over the pol II cleft, between RPB5 and RPB1 (Wu et al., 2012). Notably, these surfaces partially overlap with TFIIF binding sites on pol II (Chen et al., 2010b; Eichner et al., 2010; He et al., 2013), in agreement with the cellular and biochemical data that showed mutually exclusive Gdown1 or TFIIF binding to pol II (Cheng et al., 2012; Jishage et al., 2012; Wu et al., 2012). As shown schematically in Figure 13, a cryo-EM structure of the human Mediator–pol II–TFIIF assembly reveals an extensive Mediator–pol II interface along the pol II-Gdown1 docking site (Bernecky et al., 2011). Thus, Mediator–pol II interactions are centered on the Gdown1 and TFIIF binding surfaces, in support of the Mediator requirement for alleviating Gdown1-TFIIF antagonism.
Figure 13.Gdown1/POLR2M, TFIIF and pol II each converge on the same structural interface of Mediator. At left is shown a “bottom” view of the VP16-Mediator complex (Taatjes et al., 2002). A pol II interaction surface is highlighted by the yellow dashed line. This distinctive interaction surface forms upon TF binding (see Figure 11). At center is a “front” view of the Mediator complex, with pol II (red ribbon; PDB 1Y1V) oriented consistent with its bound state orientation in the VP16-Mediator-pol II-TFIIF assembly (Bernecky et al., 2011), shown at right. Highlighted at right is a general location for Gdown1 binding to pol II, based upon cryo-EM data (Wu et al., 2012), as well as the approximate location of TFIIF, based upon crosslinking-MS data and cryo-EM data (Chen et al., 2010b; Eichner et al., 2010; He et al., 2013). The cryo-EM map for the VP16-Mediator-pol II-TFIIF assembly is shown in blue mesh, with pol II docked as described (Bernecky et al., 2011). (see colour version of this figure online at www.informahealthcare.com/bmg).


Gdown1/POLR2M, TFIIF and pol II each converge on the same structural interface of Mediator. At left is shown a “bottom” view of the VP16-Mediator complex (Taatjes et al., 2002). A pol II interaction surface is highlighted by the yellow dashed line. This distinctive interaction surface forms upon TF binding (see Figure 11). At center is a “front” view of the Mediator complex, with pol II (red ribbon; PDB 1Y1V) oriented consistent with its bound state orientation in the VP16-Mediator-pol II-TFIIF assembly (Bernecky et al., 2011), shown at right. Highlighted at right is a general location for Gdown1 binding to pol II, based upon cryo-EM data (Wu et al., 2012), as well as the approximate location of TFIIF, based upon crosslinking-MS data and cryo-EM data (Chen et al., 2010b; Eichner et al., 2010; He et al., 2013). The cryo-EM map for the VP16-Mediator-pol II-TFIIF assembly is shown in blue mesh, with pol II docked as described (Bernecky et al., 2011). (see colour version of this figure online at www.informahealthcare.com/bmg).

The functional studies of Gdown1, TFIIF, paused pol II, and Mediator implicated a role for Mediator in “remodeling” or “modifying” Gdown1 to enable TFIIF function (Cheng et al., 2012; Jishage et al., 2012). Although numerous mechanisms can be envisioned, we hypothesize that Mediator structural shifts, perhaps triggered by TF binding, could play a role in coordinating pol II pause release involving Gdown1 and TFIIF. TF binding can cause major structural re-organization within Mediator, in particular, at a region which pol II, Gdown1, and TFIIF would converge (Figure 13).

The SEC consists of a set of factors broadly implicated in regulation of pol II transcription elongation (Lin et al., 2010). Various forms of the complex appear to regulate different sets of genes in metazoans (Luo et al., 2012a), and core components include P-TEFb (CDK9 and CCNT1/2) and AFF4. Mediator interactions with SEC components have been emerging (Galbraith et al., 2013; Vijayalingam & Chinnadurai, 2013; Wang et al., 2013) and seem to involve both Mediator and CDK8-Mediator complexes. Using proteomics and biochemistry, MED26 was found to associate with the SEC, and MED26 depletion affected a subset of elongation-regulated genes; SEC occupancy at c-MYC and HSP70 correlated with MED26 levels, as did pol II CTD phosphorylation (Takahashi et al., 2011). A physical and functional association between CDK8-Mediator and SEC components has also been characterized. Proteomics and biochemical experiments identified SEC components P-TEFb and AFF4 associated with CDK8-Mediator complexes (Ebmeier & Taatjes, 2010), and evidence for functional coordination between CDK8 and P-TEFb was observed upon analysis of serum response gene expression in HCT116 cells (Donner et al., 2010) and Dio1 gene expression in α2 cells (Belakavadi & Fondell, 2010). The SEC in general and P-TEFb in particular have been shown to be important for pol II pause release, allowing productive elongation (Zhou et al., 2012b). Interestingly, both MED26 and the CDK8 module can dissociate from the Mediator complex (Taatjes et al., 2002).

The Espinosa lab has further established the importance of CDK8 in transcription elongation and/or pol II pause release at HIF1A target genes (Galbraith et al., 2013). During hypoxia, CDK8 was important for recruitment of SEC components AFF4 and CDK9 (the kinase within P-TEFb) to HIF1A-bound promoters, and CDK8 occupancy correlated with pol II pause release. The fact that functionally distinct human Mediator complexes (core Mediator, via MED26, and CDK8-Mediator) each appear to act in conjunction with SEC factors in elongation may reflect differing roles in establishing or releasing paused pol II complexes, or may result from gene-selective requirements. The Espinosa group has also postulated that a variant form of Mediator that contains both MED26 (typically associated with core Mediator only) and CDK8 module components might be functioning in response to HIF1A activation (Galbraith et al., 2013).

Although most non-coding RNA (ncRNA) genes are transcribed by pol II, it was only relatively recently that a definitive role for Mediator in ncRNA expression was confirmed. A role for Mediator in the transcription of ncRNAs appears to be conserved in yeast, plants, and mammals. Working with mouse embryonic stem cells, the Tora lab isolated a complex that included Mediator and the histone acetyltransferase complex Ada-Two-A-containing (ATAC) that was involved in the expression of ncRNA genes (Krebs et al., 2010). In Arabidopsis thaliana it was shown Mediator is required for microRNA (miRNA) transcription and for transcription of long ncRNAs that serve as scaffolds for recruitment of RNA pol V. In each case, Mediator function was linked to pol II recruitment to the ncRNA genes (Kim et al., 2011). In the yeast S. pombe, a Med8-Med18-Med20 subcomplex (Mediator head module subunits) was required for ncRNA transcription and siRNA processing involved in silencing transcription at centromeres (Thorsen et al., 2012).

Non-coding RNAs have emerged as major players in the control of gene expression patterns throughout human development and disease (Guttman & Rinn, 2012; Hu et al., 2012; Wilusz et al., 2009). A prevalent mechanism of action for ncRNAs is interaction with protein complexes that regulate transcription (Wang & Chang, 2011). Recently, the Shiekhattar group discovered that ncRNAs can govern gene expression by directly binding Mediator and controlling its activity (Lai et al., 2013). Whereas most ncRNAs thus far characterized function in trans (Guttman & Rinn, 2012), the Shiekhattar group identified a class of ncRNA called ncRNA-activating (ncRNA-a) that are transcribed from gene enhancers and appear to activate neighboring genes in cis (Orom et al., 2010). Following-up on this discovery, it was demonstrated that at least a subset of these ncRNAs can interact with Mediator to help direct enhancer-dependent transcription activation. Significantly, the ncRNA–Mediator interaction appears to function by regulating CDK8 kinase activity and coordinating enhancer–promoter gene loop formation (Lai et al., 2013).

A single miRNA can regulate many genes at the post-transcriptional stage due to the ability to target transcripts based upon perfect or imperfect sequence complementarity. MiRNAs are metazoan-specific, which fits with their tissue-specific regulatory mechanisms and their ability to discriminate alternately spliced transcripts (Ebert & Sharp, 2012). A consequence of miRNA action is to down-regulate specific mRNA translation by either degrading the RNA directly or preventing its translation at the ribosome. In a previous section, we outlined how Mediator complexes that lack specific subunits are generally stable; because they lack specific subunits, however, such complexes are more limited or specialized in their ability to activate transcription. MicroRNAs, by their ability to target select Mediator subunits, could represent a biologically relevant means to regulate Mediator subunit composition, thereby impacting its regulatory potential genome-wide. Because most protein-coding transcripts are predicted to be regulated by miRNA targeting (Friedman et al., 2009), it is likely that miRNAs could play a role in regulating Mediator subunit composition (and therefore, its activity) in metazoans.

In support of this hypothesis, a miRNA screen with human placental trophoblasts under hypoxic conditions identified MED1 as a target of miR-205. Reporter assays confirmed a specific target sequence in the 3′-UTR of MED1 that could be important for regulation of transcriptional response to hypoxia (Mouillet et al., 2010). The MED13 subunit, a component of the CDK8 module, is targeted by miRNAs important for metabolic homeostasis (Carrer et al., 2012; Grueter et al., 2012). The Olson lab showed that the heart-specific miR-208a targets MED13; overexpression of MED13 or inhibition of miR-208a expression caused increased insulin sensitivity and glucose tolerance in mice. By contrast, MED13 depletion caused metabolic syndrome. Metabolic defects were further linked to MED13-specific repression of genes regulated by nuclear receptors, including thyroid hormone receptor (Grueter et al., 2012).

Proper RNA processing requires capping, removal of introns via splicing complexes, transcript termination and polyadenylation of the cleaved pre-mRNA. Processing of pol II transcripts occurs both concurrently and after transcription and the molecular mechanisms involved remain an area of active research (Darnell, 2013; Kornblihtt et al., 2013; Perales & Bentley, 2009). The Wang lab reported an association between MED23 and the RNA processing factor hnRNP L using affinity purification mass spectrometry (Huang et al., 2012b). Partial genomic colocalization of MED23 and hnRNP L was also demonstrated along with splicing factors related to U1/U2 snRNPs. This study demonstrated a direct association of the Mediator complex with the mRNA splicing machinery and indicated roles for MED23 in alternate splicing, cleavage and poly-adenylation. A role for the Mediator subunit Med18 (Srb5) in RNA cleavage and polyadenylation in budding yeast has also been described (Mukundan & Ansari, 2011). Med18 was shown to occupy the 5′- and 3′-ends of the selected genes and recruitment of RNA cleavage-polyadenylation factors was impaired in Med18 null cells. A novel role for CDK8 and CCNC in the 3′-processing of small nuclear RNAs (snRNA) was also characterized in Drosophila and human cells (Chen et al., 2012a). CDK8 and CCNC along with subunits of the Integrator complex (Baillat et al., 2005) were identified in a genome-wide RNAi screen and found to be biochemically associated. Interestingly, expression of a kinase-dead CDK8 mutant resulted in misprocessing of snRNAs, suggesting a role for the CDK8 kinase in snRNA maturation.

As a central component of the PIC, Mediator is mechanistically situated to regulate the recruitment and activity of factors that can remodel or modify chromatin. Moreover, Mediator is targeted by a vast array of DNA-binding TFs, which bind at enhancers and promoters and recruit Mediator to specific genomic loci. Mediator, in turn, interacts directly and extensively with the pol II enzyme. Thus, Mediator appears to function as a “molecular bridge” that communicates regulatory signals from DNA-binding TFs to the pol II enzyme (Figure 14). This simple model approximates what has been observed in a growing number of studies that suggest Mediator can function as a chromatin architectural factor to help enforce gene expression patterns in cells.
Figure 14.A simple schematic illustrating enhancer-promoter communication via Mediator. Mediator can bind simultaneously to enhancer-bound TFs and the PIC, including pol II. (see colour version of this figure online at www.informahealthcare.com/bmg).


A simple schematic illustrating enhancer-promoter communication via Mediator. Mediator can bind simultaneously to enhancer-bound TFs and the PIC, including pol II. (see colour version of this figure online at www.informahealthcare.com/bmg).

In this section, we consider gene loops to involve juxtaposition of the 5′ and 3′ end of genes or enhancer–promoter contacts. A role for Mediator in gene loop formation was suggested from studies of enhancer-promoter communication during activation in response to nuclear receptors. Because MED1 is a common target for nuclear receptors (Table 3), it was demonstrated that MED1 knockdown negatively regulated NR-dependent activation; it was also noted, however, that loss of expression coincided with loss of a gene loop connecting the enhancer and promoter of select genes (Park et al., 2005; Wang et al., 2005b). The Young lab discovered that Mediator and cohesin work cooperatively to form enhancer-promoter gene loops; moreover, they demonstrated that this basic function was important to maintain robust expression of cell type-specific genes (Kagey et al., 2010). An important role for the cohesin loading factor Nipbl was noted with Mediator-cohesin complexes, distinguishing from cohesin-CTCF interactions throughout the genome of murine ES cells. Significantly, Mediator and cohesin occupancy – and the corresponding gene loops – changed upon ES cell differentiation into MEFs. Mediator and cohesin occupancy and loops were lost at pluripotency genes and newly established at genes specifically up-regulated in MEFs (Kagey et al., 2010). These findings suggested a key role for Mediator in expression of lineage-specific genes. The Corces group, working in collaboration with the Dekker and Taylor labs, has expanded upon this theme. By tracking six different developmentally regulated loci from the mES cell stage to neural progenitor cells, the authors concluded that Mediator and cohesin were essential for formation of enhancer-promoter loops that helped specify expression of key developmentally regulated genes (Phillips-Cremins et al., 2013). This role has been further verified with the characterization of super-enhancers that depend upon Mediator (and numerous other well-known transcription regulators) for their maintenance of lineage-specific gene expression patterns (Loven et al., 2013; Whyte et al., 2013).

In addition to cohesin, Mediator can interact with ncRNAs, at least at a subset of genes, to facilitate gene activation and gene loop formation, as shown by the Shiekhattar group (Lai et al., 2013). Also, in yeast, the Med18 (Srb5) subunit was shown to govern gene loop formation between gene 5′ and 3′ ends; loop formation in this instance was shown to affect mRNA processing events at the 3′ end of the gene (Mukundan & Ansari, 2013).

Evidence for Mediator-nucleosome interactions has been obtained in both yeast and humans in vitro (Lorch et al., 2000; Nock et al., 2012), as has Mediator interaction with the chromatin remodeling protein CHD1 (Khorosjutina et al., 2010; Lin et al., 2011). Genetic and biochemical experiments have shown yeast Mediator can broadly influence chromatin structure (Kremer et al., 2012; Macatee et al., 1997), and can interact with histone tails (Liu & Myers, 2012; Zhu et al., 2011b). Additional work in yeast has demonstrated a Mediator dependence for maintaining heterochromatin regions and also roles for telomere and centromere maintenance (Carlsten et al., 2012; Mozdy et al., 2008; Peng & Zhou, 2012; Zhu et al., 2011a). The Gustafsson lab linked association of Mediator with heterochromatin to the establishment of boundaries between active and inactive genes at sub-telomeric loci (Zhu et al., 2011a). These roles were attributed to Med5 and Med7, and it was noted that Med5 loss also affected yeast replicative life span (Zhu et al., 2011a). Med5 loss affected histone H4K16Ac levels at sub-telomeric regions, suggesting a molecular mechanism underlying the changes in life span (Dang et al., 2009).

In human cells, the RE1 silencing TF (REST) can interact with the CDK8 module subunit MED12, which helps form a ternary complex with the EHMT2/G9a H3K9 methyltransferase (Ding et al., 2008). Notably, EHMT2-mediated H3K9 methylation can initiate a cascade of events to establish a repressive chromatin state. The CDK8 module, which can associate with Mediator to form a stable CDK8-Mediator complex, can phosphorylate nucleosomes at histone H3S10 (Knuesel et al., 2009b). The GCN5L acetyltransferase can also stably associate with human CDK8-Mediator (but not core Mediator) and this complex has been shown to cooperatively phospho-acetylate histone H3 to generate the tandem S10-phospho/K14-acetyl mark (Meyer et al., 2008). This mark has been implicated in activation at serum response genes (Clayton et al., 2000; Lo et al., 2000), and work by the Espinosa lab has shown a positive role for CDK8 in activation of these genes (Donner et al., 2010). However, the MSK2/1 kinases have been implicated in H3S10 phosphorylation at serum response genes in mice (Soloaga et al., 2003), and the regulatory importance of CDK8 for H3S10 phosphorylation per se remains uncertain.

The histone acetyltransferase p300 also functionally cooperates with Mediator (Acevedo & Kraus, 2003; Black et al., 2006; Huang et al., 2003), and Mediator has been shown to co-localize with p300 in ChIP-Seq studies (Wang et al., 2011). In fact, like p300, Mediator subunits are now considered a reliable surrogate for identification of enhancers using ChIP-Seq (Loven et al., 2013; Whyte et al., 2013). Also, in a pair of studies, the Carey lab identified structural and functional antagonism between Mediator and the heterochromatin-associated PRC1 complex (Lehmann et al., 2012) and the heterochromatin protein HP1γ/CBX3 (Smallwood et al., 2008).

In each of the cases described above (CDK8 modification of histone H3, functional cooperativity/antagonism with p300, PRC1, or HP1γ), similar roles for yeast Mediator have not been observed. Whereas yeast express a structural ortholog of p300 (Rtt109) (Tang et al., 2008), yeast orthologs for the PRC1 complex or HP1γ/CBX3 do not exist (although S. pombe express Swi6, an HP1-like protein), and yeast CDK8 (Srb10) is unable to phosphorylate histones (Hengartner et al., 1998).

CDK8 was originally identified with Cyclin C (CCNC) in yeast as a protein that when mutated suppressed growth defects associated with pol II CTD truncation mutations (Liao et al., 1995). The CDK8-CCNC dimer (Srb10-Srb11 in yeast) co-migrated with other yeast Mediator proteins in early pol II holoenzyme purifications and also appeared to phosphorylate the pol II CTD in vitro (Liao et al., 1995), which contributed to its characterization as a transcriptional CDK (Loyer et al., 2005). In yeast, CDK8-CCNC (Srb10/11) were linked genetically (Carlson, 1997; Holstege et al., 1998) and biochemically (Borggrefe et al., 2002) to MED12 and MED13 (Srb8/9). Similarly, genetic and biochemical experiments in metazoans identified a CDK8, CCNC, MED12 and MED13 complex that could associate with Mediator (Loncle et al., 2007; Taatjes et al., 2002; Wang et al., 2001).

Genetic experiments in C. elegans and Drosophila have linked CDK8 module subunits to transcription repression or activation (Carrera et al., 2008; Gaytan de Ayala Alonso et al., 2007; Janody & Treisman, 2011; Wang et al., 2004a). In yeast, early experiments suggested that CDK8 (Srb10) was a negative regulator of transcription in vivo because it appeared to phosphorylate the pol II CTD prior to PIC assembly (Hengartner et al., 1998), and microarray analysis with a kinase dead CDK8 (Srb10) mutant revealed de-repression of approximately 3% of protein-coding genes (Holstege et al., 1998). A later study in yeast also reported that CDK8 module components have a generally repressive role in transcription (van de Peppel et al., 2005). However, using analog-sensitive mutants, the Hahn group revealed positive roles for the CDK8 (Srb10) kinase that were only observed upon inhibition of another transcription-relevant kinase, CDK7 (Kin28) (Liu et al., 2004).

Whereas the sequences of CDK8 module components (CDK8, CCNC, MED12, MED13) have diverged considerably across evolution (Table 1), CDK8 kinase activity has been retained. This conserved activity is reflected in the known substrates of CDK8, which include DNA-binding TFs in both yeast and human cells (Table 4). Structures of the yeast and human CDK8 modules are shown in Figure 15. It is noteworthy that the CDK8 module reversibly associates with Mediator in both yeast and humans; this allows recruitment of the CDK8 kinase to regulatory loci on a genome-wide scale (Andrau et al., 2006; Zhu et al., 2006). Genome-wide targeting implies widespread roles for the CDK8 module and the CDK8 kinase in transcription. Further highlighting the basic role for CDK8 in gene expression, knockout of this subunit in flies or mice is embryonic lethal (Loncle et al., 2007; Westerling et al., 2007). Much remains to be discovered, but current understanding makes clear that the CDK8 module regulates transcription through varied mechanisms and in context-specific ways (Galbraith et al., 2010).
Figure 15.EM structures of the human (A) and yeast (B) CDK8 module. The structures are distinct and not shown in similar orientations. Structural distinctions may derive from sequence differences, in particular, the much larger sizes for human MED12 and MED13. MED13 forms an extended hook-like structure in each, and this subunit has been shown to contact core Mediator (Knuesel et al., 2009a; Tsai et al., 2013). Whereas the general location of MED13 was determined for the human CDK8 module, localization of each subunit was determined for the yeast structure with antibody labeling (C). (see colour version of this figure online at www.informahealthcare.com/bmg).
Table 4.Current known CDK8 kinase substrates.ProteinReferenceProteinReference
Notch ICD
Fryer et al., 2004

Msn2
Chi et al., 2001

SMAD1
Alarcon et al., 2009

Gcn4
Chi et al., 2001

SMAD3
Alarcon et al., 2009

Gal4
Hirst et al., 1999

SREBP
Zhao et al., 2012

Ste12
Nelson et al., 2003

E2F1
Morris et al., 2008

Phd1
Raithatha et al., 2012

STAT1
Bancerek et al., 2013

Sip4
Vincent et al., 2001
histone H3Meyer et al., 2008

Fkh2
Szilagyi et al., 2012
CCNHAkoulitchev et al., 2000
Med2Liu et al., 2004
PC4Gu et al., 1999
Taf2Liu et al., 2004
CDK8Knuesel et al., 2009bBdf1Liu et al., 2004
MED13Knuesel et al., 2009bNote that each substrate can be considered chromatin associated. DNA-binding TFs are shown in bold font. Human CDK8 kinase substrates are shown on the left and yeast Cdk8 (srb10) substrates are shown on the right.


EM structures of the human (A) and yeast (B) CDK8 module. The structures are distinct and not shown in similar orientations. Structural distinctions may derive from sequence differences, in particular, the much larger sizes for human MED12 and MED13. MED13 forms an extended hook-like structure in each, and this subunit has been shown to contact core Mediator (Knuesel et al., 2009a; Tsai et al., 2013). Whereas the general location of MED13 was determined for the human CDK8 module, localization of each subunit was determined for the yeast structure with antibody labeling (C). (see colour version of this figure online at www.informahealthcare.com/bmg).

Current known CDK8 kinase substrates.

Note that each substrate can be considered chromatin associated. DNA-binding TFs are shown in bold font. Human CDK8 kinase substrates are shown on the left and yeast Cdk8 (srb10) substrates are shown on the right.

The kinase activity of CDK8 can function to activate or repress (by various mechanisms) transcription by DNA-binding TFs. These roles, described further below, are fairly straightforward and not controversial. It is less clear how the CDK8 module functions within the context of the PIC. Some initial purifications of the human Mediator complex used classic biochemical techniques in which activator-dependent transcription activity was tracked over a series of chromatography columns. The Mediator complexes, called CRSP and PC2 at the time, were isolated in the Tjian, Meisterernst, and Roeder labs, respectively (Kretzschmar et al., 1994; Malik et al., 2000; Ryu et al., 1999). Notably absent from these transcriptionally active Mediator fractions were components of the CDK8 module, suggesting no direct role in PIC activation. Later work has shown the CDK8 module blocks pol II-Mediator binding in both yeast and human systems (Elmlund et al., 2006; Ebmeier & Taatjes, 2010; Knuesel et al., 2009a; Naar et al., 2002; Tsai et al., 2013); however, the CDK8 module has also been shown to positively affect transcription elongation in human cells (Donner et al., 2010) and may function as a pol II pause release factor (Galbraith et al., 2013). Whereas such roles could be categorized as positive or negative, when considered in the context of all stages of transcription (pre- and post-initiation, elongation, termination), a more consistent and less contradictory model emerges. Activities that could be considered negative are likely essential to ensure the integrity and timing of transcriptional events at a particular gene locus. An illustration of this model is shown in Figure 9.

The current known substrates for the CDK8 kinase are listed in Table 4. Many of these substrates are DNA-binding TFs (Ansari et al., 2005), and all substrates can be considered chromatin-associated. Although a functional role for all known CDK8 targets is not established, we highlight several for which a regulatory role has been uncovered.

In yeast, Cdk8 (Srb10) positively regulates the activity of the Gal4 and Sip4 TFs by phosphorylation (Hirst et al., 1999; Vincent et al., 2001). By contrast, Cdk8 (Srb10) phosphorylation of the TFs Gcn4, Ste12, or Phd1 promotes their degradation by the proteasome (Chi et al., 2001; Nelson et al., 2003; Raithatha et al., 2012). Each of these Cdk8-regulated TFs is involved in nutrient response, with Gal4 and Sip4 being active in the fed state. In the presence of limited nutrients, the entire Cdk8 module is degraded in yeast (Holstege et al., 1998). Thus, in a nutrient-starved state, Gal4 and Sip4 will be repressed (no Cdk8-dependent activation) and Gcn4, Ste12 and Phd1 will be stabilized (no Cdk8-dependent phosphorylation and degradation). This fits very well with the physiological roles of these TFs, as activation of Gcn4, Ste12 and Phd1 target genes is critical to reprogram yeast metabolic pathways to enable survival when nutrients are scarce.

As with yeast Cdk8 (Srb10), mammalian CDK8 has been linked to both repression and activation of TF activity by phosphorylation. CDK8 represses SREBP-1c and E2F1 activity by phosphorylation (Morris et al., 2008; Zhao et al., 2012, 2013). By contrast, CDK8 has been shown to help activate the bone morphogenetic pathway (BMP) and transforming growth factor beta (TGF-β) pathways through phosphorylation of SMAD 1/5 and SMAD 2/3, respectively (Alarcon et al., 2009; Gao et al., 2009). Likewise, activation of the interferon or Notch signaling pathway can occur through CDK8 phosphorylation of STAT1 or the Notch receptor intracellular domain (Bancerek et al., 2013; Fryer et al., 2004). Interestingly, CDK8-dependent TF phosphorylation is often coupled with increased protein turnover (Alarcon et al., 2009; Fryer et al., 2004), which appears essential for a robust response to signaling inputs (Metivier et al., 2003; Reid et al., 2003).

Many CDKs auto-phosphorylate at a conserved threonine (T) residue in their activation loops (a.k.a. T-loop), which activates the kinase (Johnson et al., 1996). Rather than a threonine in its T-loop, CDK8 and its paralog CDK19 contain an aspartate (D) at this position (Leclerc et al., 1996). As a consequence, it has been postulated that Glu99 of Cyclin C adjusts the orientation of three important arginines (rather than a phosphoresidue) within CDK8 to activate the kinase. Although the crystal structure of the human CDK8-CCNC complex was unable to definitively address this question (Schneider et al., 2011), the yeast CDK8-CCNC dimer appears to represent a constitutively active kinase, in agreement with the Glu99 structural model. By contrast, human CDK8 appears to be regulated differently. The human CDK8-CCNC dimer is largely inactive, but recombinant protein complexes containing CDK8, CCNC and MED12 exhibit far greater kinase activity than CDK8-CCNC alone (Knuesel et al., 2009b). Thus, MED12 appears to activate CDK8 kinase activity within the human CDK8 module. Biochemical purification and proteomics has revealed auxiliary proteins that co-purify with the human CDK8 module that might also play roles in regulating its kinase activity (Knuesel et al., 2009b), and recent studies have implicated ncRNAs and p21 in regulation of CDK8 kinase activity (Lai et al., 2013; Porter et al., 2012).

Another means to regulate CDK8 kinase activity is to control its access to substrates. The yeast and human CDK8 modules bind Mediator through their MED13 subunit (Knuesel et al., 2009a; Tsai et al., 2013). The direct association with Mediator (which is recruited to genomic loci by TFs) will localize CDK8 with the PIC and DNA-binding TFs. This co-localization represents a simple means to control substrate access, and is consistent with current known CDK8 kinase substrates (Table 4).

The Espinosa lab has led in the identification of context-specific roles for CDK8 and has also revealed clues regarding CDK8 mechanism in transcription elongation. By analyzing cellular responses to p53 activating agents, Donner et al. observed differential activation of p53 target genes that was dependent on the CDK8 module (Donner et al., 2007). In fact, at genes activated in response to Nutlin-3, the occupancy of most PIC factors, including Mediator, remained similar before and after stimulus. Recruitment of the CDK8 module, however, increased and correlated with mRNA levels. This CDK8-dependent up-regulation of select genes within the p53 network established a stimulus-specific role for CDK8, which is now a common theme (Donner et al., 2007). Later, the Espinosa group studied the role of CDK8 during serum response and noted that CDK8 was required for strong activation of canonical serum-induced genes such as EGR1 and FOS (Donner et al., 2010). Importantly, they determined that CDK8 knockdown affected pol II elongation, including the elongation rate and the phosphorylation status of the pol II CTD. This was linked further to defects in CDK9 and CDK7 occupancy at affected genes upon CDK8 knockdown. More recently, Galbraith et al. (2013) observed pathway-specific roles for CDK8 in transcription elongation of HIF1A target genes during hypoxia. Among other things, this work provided an additional context in which CDK8 occupancy was linked to CDK9 occupancy.

Collectively, these results suggest a role for the CDK8 module in transcription elongation and also imply a physical and functional connection between CDK8 and CDK9 (P-TEFb). Related to this, the Fisher laboratory determined that CDK7 (kinase within TFIIH) activates CDK9 kinase activity by phosphorylating its T-loop (at CDK9 residue T186) on chromatin in human cells. Blocking CDK7 activity (using a Shokat analog-sensitive mutant CDK7) indirectly affected pol II CTD phosphorylation at Ser2, a CDK9 (P-TEFb) substrate (Larochelle et al., 2012). In yeast, a functional interplay between Cdk7 and Cdk8 has been characterized by the Hahn lab, which studied pol II CTD phosphorylation using analog-sensitive Cdk7 (Kin28) and Cdk8 (srb10) mutants. The Hahn group determined that although Cdk8 did not appear to directly phosphorylate the pol II CTD, Cdk8 activity affected the ability of Cdk7 to phosphorylate the CTD (Liu et al., 2004). These interesting findings suggest a co-regulatory network among transcription-associated kinases CDK7, CDK8 and CDK9.

Other stimulus-specific roles for CDK8 have been observed in response to Wnt/β-catenin signaling (via E2F1) and interferon response (via STAT1). CDK8 has been identified as a colon cancer oncogene and oncogenesis requires CDK8 kinase activity (Firestein et al., 2008). One substrate linked to oncogenesis was E2F1, a TF that normally represses β-catenin. Upon phosphorylation by CDK8, however, this E2F1 repression is lost, enabling β-catenin to drive tumorigenesis (Morris et al., 2008). CDK8 was also shown to play a key role in STAT1 antiviral response (Bancerek et al., 2013). The STAT1 TF is activated by various extracellular signals; STAT1 activation involves phosphorylation within its activation domain at residue S727. The Kovarik group demonstrated that CDK8 phosphorylated STAT1 at S727, but in a stimulus-specific manner. In particular, CDK8 phosphorylated and activated STAT1 in response to interferon-gamma, whereas other “non-cytokine” STAT1-activating signals were not dependent upon CDK8. In agreement with work from the Espinosa lab, the regulatory role of CDK8 at STAT1 target genes appeared to involve pol II elongation (Bancerek et al., 2013).

CDK8-dependent phosphorylation of sterol regulatory element-binding protein (SREBP)-1c was shown to negatively regulate this TF, thereby down-regulating genes in the lipogenic pathway (Zhao et al., 2012). Phosphorylation of SREBP-1c led to its ubiquitination and degradation. This work showed that CDK8 kinase activity was an important regulator of lipid metabolism, whose dysregulation is associated with diabetes and insulin resistance. CDK8 has also been shown to play a role nutrient signaling through glucose metabolism and the mTOR pathway, which also appears to be connected to CDK8 kinase activity (Kuchin et al., 1995, 2000; Mousley et al., 2012; Song et al., 1996). It will be interesting to further dissect the physiological roles for the CDK8 kinase; given that many of its current known targets are TFs, it could play major roles in regulating metabolism and disease.

CDK8 module components have been linked to developmental pathways in humans, worms, zebrafish and flies (Malik & Roeder, 2010). CDK8 and CCNC also appear to be involved in development of the amoeba Dictyostelium discoideum (Lin et al., 2004; Takeda et al., 2002). A study of the D. discoideum kinome revealed that CDK8 was part of a set of core kinases that were conserved in D. discoideum, yeast, and throughout metazoa, highlighting the potential evolutionary importance of CDK8 in development (Goldberg et al., 2006).

In addition to CDK8 and CCNC, MED12 and MED13 are also critical regulators of developmental gene expression programs (Hong et al., 2005; Kennison & Tamkun, 1988; Rau et al., 2006; Wang et al., 2006). In a C. elegans RNAi screen, Med12 (dpy-22) was identified by the Fraser lab as a highly connected “hub” gene that regulated numerous signaling pathways (Lehner et al., 2006). Others have connected MED12 to Ras and Wnt signaling pathways (Kim et al., 2006a) implicated in vulval development and Hox gene expression, respectively (Moghai & Sternberg, 2003; Yoda et al., 2005). Point mutants in MED12 have been associated with X-linked intellectual disability (XLID) in humans (Ding et al., 2008), namely FG and Lujan syndromes (Risheg et al., 2007), through disruption of CDK8 association and hyperactivated Sonic Hedgehog (SHH) signaling (Zhou et al., 2012a). Work in Drosophila showed that each genetic component of the Cdk8 module was required for organismal development, but not cell viability (Loncle et al., 2007), and indicated a functional separation between Cdk8:Ccnc and Med12:Med13 in regulating target genes and development in the eye, leg and wing (Janody et al., 2003; Loncle et al., 2007; Treisman, 2001). CDK8 module subunits were also found to be transcriptional endpoints of the Wnt (Carrera et al., 2008) and Notch signaling pathways (Janody & Treisman, 2011). The CDK8 module also plays key roles in cell fate choice and differentiation in the hematopoietic system through induction of RUNX and GATA family transcription factors (Gobert et al., 2010). This study further established a role for MED12 and MED13 independent of CDK8 and CCNC in promoting blood cell differentiation.

Vertebrates have genomes that contain duplications of CDK8 (now referred to as CDK19 (Malumbres et al., 2009)), MED12 (MED12L), and MED13 (MED13L). The fact that three of the four CDK8 module components have paralogs raises interesting questions about their biological functions. Although CDK19, MED12L and MED13L are largely unstudied, existing data indicate their biological roles are not redundant. CDK19 cannot compensate for CDK8 knockout in mice (Westerling et al., 2007) and differential interactions and activities have been noted in protein interaction screens and transcription assays (Fukasawa et al., 2012; Tsutsui et al., 2008). Moreover, a recent study from the Espinosa lab has shown that CDK8, but not CDK19, is required for induction of hypoxia inducible factor 1A (HIF1A) target genes in response to hypoxia (Galbraith et al., 2013).

Distinct physiological roles for CDK19, MED12L and MED13L may manifest in cell- and tissue-specific ways. A Northern blot analysis across various human tissues suggested that CDK8 is ubiquitous, whereas CDK19 shows tissue specific expression (Tsutsui et al., 2011). Also, the CDK19 gene was found to be disrupted in a patient with microcephaly, mental retardation, and congenital retinal folds (Mukhopadhyay et al., 2010). MED13L mutations are associated with the congenital heart defect transposition of the great arteries, and it appears that MED13L is involved in both brain and heart development (Muncke et al., 2003).

The plant Mediator complex (Kidd et al., 2011) was purified in 2007 from the model organism A. thaliana, via ion exchange chromatography and immunoprecipitation (IP) with a Med6 antibody (Backstrom et al., 2007). Although the purification identified 21 conserved Mediator subunits, some human orthologs appeared to be missing, including MED1 and the CDK8 module. Sequence analysis, however, predicts the presence of the Cdk8 module in A. thaliana (Gillmor et al., 2010; Ito et al., 2011; Wang & Chen, 2004). Primary sequences of plant Mediator subunits are quite different from those of other eukaryotes (Backstrom et al., 2007). Nevertheless, conserved motifs imply there may be more similarity at the structural level than the sequence level (Bourbon, 2008; Mathur et al., 2011). Plant-specific paralogs of Mediator subunits are also evident (Mathur et al., 2011), and a thorough investigation into their functions will be required for a complete understanding of transcriptional regulation in plants.

Plant Mediator subunits have been implicated in stress and immune responses and development (Anderson et al., 2004; Autran et al., 2002; Bonawitz et al., 2012; Cerdan & Chory, 2003; Cevik et al., 2012; Elfving et al., 2011; Kidd et al., 2009; Zhang et al., 2013; Zheng et al., 2013). A functional diversification of Mediator in plants, however, is suggested by reports implicating plant Mediator in ncRNA biogenesis (Kim et al., 2011), genome stability (Kobbe et al., 2008) and rRNA processing (Barneche et al., 2000). Plants contain unique transcription factors (Backstrom et al., 2007), additional RNA polymerases (Huang et al., 2009), polyploid genomes, and distinct biological requirements compared to other eukaryotes. Therefore, dissection of Mediator’s roles in plants will likely continue to reveal both shared and plant-specific biological functions.

A growing number of studies implicate Mediator in human disease, and several excellent reviews have been written on this topic (Napoli et al., 2012; Spaeth et al., 2011). An attractive aspect of Mediator as a therapeutic target is that, generally speaking, its different subunits control different sets of genes. Therefore, targeting a single Mediator subunit might block a specific pathway, but allow a majority of cellular transcription to function normally. At least some Mediator subunits appear to function in a cell-type specific manner (Chen et al., 2010a; Ge et al., 2002; Grueter et al., 2012; Jiang et al., 2010; Pope & Bresnick, 2013; Stumpf et al., 2010; Yin et al., 2012), probably due to cell-type specific transcription factors or cofactors that interact with these subunits. Such biological characteristics suggest that targeting select Mediator subunits, perhaps by blocking a specific TF binding site, could have both gene- and cell-type specific effects. Given the well-documented challenges with targeting protein-protein interfaces that control transcription (Darnell, 2002), however, these putative advantages may be difficult to realize (Phillips & Taatjes, 2013). Progress with structural analysis using NMR has revealed high-resolution information about a few TF–Mediator subunit interactions (Brzovic et al., 2011; Thakur et al., 2008), including SREBP–MED15 and VP16–MED25 (Milbradt et al., 2011; Vojnic et al., 2011; Yang et al., 2006). These structural data will be useful for generating small molecules that could bind key control points within Mediator.

A particularly promising therapeutic target is CDK8 (Xu & Ji, 2011), which is a potent oncogene (Firestein et al., 2008; Kapoor et al., 2010; Morris et al., 2008) whose expression is associated with poor clinical outcomes (Firestein et al., 2010; Nagalingam et al., 2012; Porter et al., 2012). In mammals, CDK8 function can maintain tumors and stem cells in an undifferentiated state (Adler et al., 2012) and can promote cell growth via the serum response pathway (Donner et al., 2010). Most of the established biological roles for CDK8 appear to depend on its kinase activity, which provides an opportunity for small molecule inhibitors (Cee et al., 2009).

Another intriguing and significant biological function for Mediator is its essential role in the expression of genes that drive and maintain an oncogenic state. The Wang lab has outlined a role for MED23 in driving lung cancers with hyperactive Ras signaling; they also noted that elevated MED23 expression levels correlate with poor clinical outcomes (Yang et al., 2012). The Young lab identified Mediator as one of several factors critical for maintaining the function of “super-enhancers” that direct high-level expression of oncogenic genes in cancer cells (Loven et al., 2013). Notably, super-enhancers appear especially sensitive to disruption of Mediator function, suggesting a therapeutic opportunity. However, super-enhancers also exist in normal cells and appear to drive robust expression of lineage-specific genes (Whyte et al., 2013). The identification of super-enhancers was facilitated by ChIP-Seq analyses that allowed assessment of factor occupancy across different cell types. Whereas basic features of super-enhancers are not distinct from canonical enhancer elements (Carey, 1998), their selective association with loci that regulate lineage-specific (or disease-specific) gene expression programs is an important distinguishing feature.

Mediator subunits are targeted by viral activator proteins (e.g. E1a, RTA and VP16) to transcribe the viral genome during infection (Boyer et al., 1999; Fang et al., 2004; Gwack et al., 2003; Mittler et al., 2003; Yang et al., 2004). The Mediator subunit targets of viral transcription activator proteins (e.g. MED25 or MED23) therefore represent a potential means to block viral infection or propagation. Several studies that used RNAi screens have shown Mediator is required for HIV infection and replication (Bushman et al., 2009; Fahey et al., 2011). MED7 was shown to be important for early HIV reverse transcription (Konig et al., 2008) and MED4, MED6, MED7, MED14 and MED28 were required for HIV infection (Brass et al., 2008). In addition, a group of Mediator subunits were linked to HIV replication and Tat activated transcription (Zhou et al., 2008). Mediator (and TFIIH) was also shown to be critical to re-activate latent HIV-1 transcription; in this case, activation by NFκB corresponded with loss of the CDK8 module and increased occupancy of core Mediator (Kim et al., 2006b).

In the past few years, expanded roles for Mediator have been discovered that have solidified its essential and central role in regulating pol II transcription. Among many recent noteworthy advances have been structural insights provided by X-ray crystallography and experiments that have established Mediator as a regulator of chromatin architecture. As our understanding of basic mechanisms that control gene expression have progressed, so has our understanding about how Mediator regulates different stages of transcription, from pre-initiation to elongation and RNA processing. At any given locus, the same Mediator complex is probably mediating these distinct regulatory events. For example, it is not likely that Mediator dissociates upon transcription initiation and a different Mediator complex re-associates to help regulate pol II elongation. Similarly, we hypothesize that the same Mediator complex can at once control chromatin architecture (e.g. via promoter-enhancer looping) and PIC assembly and activity. How these different activities are controlled temporally remains an interesting but challenging mechanistic question.

Throughout this review, we have emphasized Mediator size and structural dynamics in part because it provides a plausible mechanism by which the same Mediator complex could perform different functions during different transcriptional stages (such as pre-initiation and elongation), while providing a means to respond to different promoter contexts (e.g. TF or CDK8 module binding). Its large size and extensive surface area allow Mediator to process multiple regulatory inputs (whether from proteins or nucleic acids) at the same time. Although the complexity of Mediator and its role in global regulation of pol II transcription make it challenging to study in vitro and in vivo, its fundamental importance in all aspects of biology should continue to expand the number of scientists that study its function. Practical improvements in structural and chemical biology, combined with genetic and genomic techniques and established biochemical and analytical methods should continue to yield important and transformative insights about Mediator function. Ideally, this will include identification of strategies that will be effective for therapeutic purposes.