---
reference_id: "PMID:36245280"
title: Integrative Functional Genomics Identifies Systemic Lupus Erythematosus Causal Genetic Variant in the IRF5 Risk Locus.
authors:
- Hou G
- Zhou T
- Xu N
- Yin Z
- Zhu X
- Zhang Y
- Cui Y
- Ma J
- Tang Y
- Cheng Z
- Shen Y
- Chen Y
- Zou LH
- Wang YF
- Yin Z
- Guo Y
- Ding H
- Ye Z
- Shen N
journal: Arthritis Rheumatol
year: '2023'
doi: 10.1002/art.42390
content_type: abstract_only
---

# Integrative Functional Genomics Identifies Systemic Lupus Erythematosus Causal Genetic Variant in the IRF5 Risk Locus.
**Authors:** Hou G, Zhou T, Xu N, Yin Z, Zhu X, Zhang Y, Cui Y, Ma J, Tang Y, Cheng Z, Shen Y, Chen Y, Zou LH, Wang YF, Yin Z, Guo Y, Ding H, Ye Z, Shen N
**Journal:** Arthritis Rheumatol (2023)
**DOI:** [10.1002/art.42390](https://doi.org/10.1002/art.42390)

## Content

1. Arthritis Rheumatol. 2023 Apr;75(4):574-585. doi: 10.1002/art.42390. Epub 2023
 Feb 8.

Integrative Functional Genomics Identifies Systemic Lupus Erythematosus Causal 
Genetic Variant in the IRF5 Risk Locus.

Hou G(1), Zhou T(2), Xu N(2), Yin Z(3), Zhu X(2), Zhang Y(2), Cui Y(4), Ma J(2), 
Tang Y(2), Cheng Z(2), Shen Y(2), Chen Y(3), Zou LH(3), Wang YF(5), Yin Z(6), 
Guo Y(6), Ding H(2), Ye Z(3), Shen N(7).

Author information:
(1)Shanghai Institute of Rheumatology, Renji Hospital, Shanghai Jiao Tong 
University School of Medicine, Shanghai, China, Shenzhen Futian Hospital for 
Rheumatic Diseases, Shenzhen, China, and State Key Laboratory of Oncogenes and 
Related Genes, Shanghai Cancer Institute, Renji Hospital, Shanghai Jiao Tong 
University School of Medicine, Shanghai, China.
(2)Shanghai Institute of Rheumatology, Renji Hospital, Shanghai Jiao Tong 
University School of Medicine, Shanghai, China.
(3)Shenzhen Futian Hospital for Rheumatic Diseases, and Joint Research 
Laboratory for Rheumatology of Shenzhen University Health Science Center and 
Shenzhen Futian Hospital for Rheumatic Diseases, Shenzhen, China.
(4)Shanghai Institute of Nutrition and Health, Shanghai Institutes for 
Biological Sciences, University of Chinese Academy of Sciences, Chinese Academy 
of Sciences, Shanghai, China.
(5)School of Life and Health Sciences, School of Medicine, and Warshel Institute 
for Computational Biology, The Chinese University of Hong Kong, Shenzhen, 
Guangdong, China.
(6)Sheng Yushou Center of Cell Biology and Immunology, Joint International 
Research Laboratory of Metabolic & Developmental Sciences, School of Life 
Sciences and Biotechnology, Shanghai Jiao Tong University, Shanghai, China.
(7)Shanghai Institute of Rheumatology, Renji Hospital, Shanghai Jiao Tong 
University School of Medicine, Shanghai, China, Shenzhen Futian Hospital for 
Rheumatic Diseases, Shenzhen, China, State Key Laboratory of Oncogenes and 
Related Genes, Shanghai Cancer Institute, Renji Hospital, Shanghai Jiao Tong 
University School of Medicine, Shanghai, China, Center for Autoimmune Genomics 
and Etiology, Division of Human Genetics, Cincinnati Children's Hospital Medical 
Center, Cincinnati, Ohio, and Department of Pediatrics, University of 
Cincinnati, Cincinnati, Ohio.

OBJECTIVE: IRF5 plays a crucial role in the development of lupus. Genome-wide 
association studies have identified several systemic lupus erythematosus (SLE) 
risk single-nucleotide polymorphisms (SNPs) enriched in the IRF5 locus. However, 
no comprehensive genome editing-based functional analysis exists to establish a 
direct link between these variants and altered IRF5 expression, particularly for 
enhancer variants. This study was undertaken to dissect the regulatory function 
and mechanisms of SLE IRF5 enhancer risk variants and to explore the utilization 
of clustered regularly interspaced short palindromic repeat interference 
(CRISPRi) to regulate the expression of disease risk gene to intervene in the 
disease.
METHODS: Epigenomic profiles and expression quantitative trait locus analysis 
were applied to prioritize putative functional variants in the IRF5 locus. 
CRISPR-mediated deletion, activation, and interference were performed to 
investigate the genetic function of rs4728142. Allele-specific chromatin 
immunoprecipitation-quantitative polymerase chain reaction and allele-specific 
formaldehyde-assisted isolation of regulatory element-quantitative polymerase 
chain reaction were used to decipher the mechanism of alleles differentially 
regulating IRF5 expression. The CRISPRi approach was used to evaluate the 
intervention effect in monocytes from SLE patients.
RESULTS: SLE risk SNP rs4728142 was located in an enhancer region, indicating a 
disease-related regulatory function, and risk allele rs4728142-A was closely 
associated with increased IRF5 expression. We demonstrated that an 
rs4728142-containing region could act as an enhancer to regulate the expression 
of IRF5. Moreover, rs4728142 affected the binding affinity of zinc finger and 
BTB domain-containing protein 3 (ZBTB3), a transcription factor involved in 
regulation. Furthermore, in monocytes from SLE patients, CRISPR-based 
interference with the regulation of this enhancer attenuated the production of 
disease-associated cytokines.
CONCLUSION: These results demonstrate that the rs4728142-A allele increases the 
SLE risk by affecting ZBTB3 binding, chromatin status, and regulating IRF5 
expression, establishing a biologic link between genetic variation and lupus 
pathogenesis.

Â© 2022 American College of Rheumatology.

DOI: 10.1002/art.42390
PMID: 36245280 [Indexed for MEDLINE]