---
reference_id: "PMID:27698206"
title: Association between ATG16L1 gene polymorphism and the risk of Crohn's disease.
authors:
- Zhang BB
- Liang Y
- Yang B
- Tan YJ
journal: J Int Med Res
year: '2017'
doi: 10.1177/0300060516662404
content_type: full_text_xml
---

# Association between ATG16L1 gene polymorphism and the risk of Crohn's disease.
**Authors:** Zhang BB, Liang Y, Yang B, Tan YJ
**Journal:** J Int Med Res (2017)
**DOI:** [10.1177/0300060516662404](https://doi.org/10.1177/0300060516662404)

## Content

1. J Int Med Res. 2017 Dec;45(6):1636-1650. doi: 10.1177/0300060516662404. Epub 
2016 Oct 2.

Association between ATG16L1 gene polymorphism and the risk of Crohn's disease.

Zhang BB(1), Liang Y(2), Yang B(1), Tan YJ(1).

Author information:
(1)1 Department of Medical Affairs, General Hospital of PLA Chengdu Military 
Area Command, Chengdu, China.
(2)2 Department of Thoracic Surgery, General Hospital of PLA Chengdu Military 
Area Command, Chengdu, China.

Objective To perform a meta-analysis to evaluate studies investigating the 
association between ATG16L1 gene polymorphism and Crohn's disease. Methods 
PubMed, Embase and Web of Science databases were searched for all studies 
focusing on the association of ATG16L1 and Crohn's disease. Combined odds ratios 
with 95% confidence intervals were calculated for four genetic models (allelic 
model: G allele versus A allele; additive model: GG versus AA; dominant model: 
GA + GG versus AA; recessive model: GG versus GA + AA) using either a random 
effects or fixed effects model. Results A total of 47 case-control studies 
involving 18 638 cases and 30 181 controls were included in the final 
meta-analysis. There was a significant association between ATG16L1 and Crohn's 
disease for all four genetic models. Significant associations were also shown in 
subgroup analyses when stratified by study design (population- or 
hospital-based). Conclusion In this meta-analysis, the ATG16L1 genotype was 
significantly associated with the risk of developing Crohn's disease.

DOI: 10.1177/0300060516662404
PMCID: PMC5805181
PMID: 27698206 [Indexed for MEDLINE]

Crohn’s disease is a type of inflammatory bowel disease associated with chronic
relapsing inflammation of the digestive tract anywhere from the mouth to the anus.1 Although its aetiopathogenesis remains unclear, it is well established that
Crohn’s disease is a complex disorder resulting from the interactions of genetic,
environmental and microbial factors. Among these, genetic factors may be responsible
for a major component of disease susceptibility.2

The role of autophagy processes in the development of inflammatory bowel disease is
attracting increasing attention.3 It is possible that genes involved in the autophagy pathway may contribute to
the pathogenesis of Crohn’s disease. The autophagy-related 16-like 1
(ATG16L1) gene encodes an important protein involved in the
formation of autophagosomes during autophagy.4 Genome-wide association studies have shown an association between
ATG16L1 polymorphism involving an amino acid change at position
300 and increased susceptibility to Crohn’s disease.5,6 This substitution of threonine
with alanine is the result of a single nucleotide polymorphism in which adenine (A)
is replaced with guanine (G). This association has been examined in numerous
studies, but the results have been inconsistent. The present meta-analysis was
designed to evaluate the association between ATG16L1 and Crohn’s
disease using the Preferred Reporting Items for Systematic Reviews and Meta-analyses
(PRISMA) criteria.7

Two investigators (B.B.Z and B.Y.) systematically searched the databases PubMed
(up to June 2016), Embase (1966 to June 2016) and Web of Science (2003 to June
2016), and also references from articles, reviews and abstracts presented at
meetings of related scientific societies. The following search terms were used:
(“ATG16L1”) AND (“Crohn’s disease” OR “inflammatory bowel
diseases”) AND (“polymorphism” OR “mutation” OR “variant” OR “genotype”).
Studies were limited to those published in English.

The same two investigators independently screened each of the titles, abstracts
and full texts to determine whether the studies met the following criteria: (i)
evaluation of the association of Crohn’s disease and ATG16L1
polymorphism; (ii) case–control design; (iii) sufficient data for the estimation
of odds ratios (ORs) and 95% confidence intervals (CIs). In addition, a quality
assessment was performed on all included studies using the Newcastle–Ottawa
Scale (NOS) as described elsewhere.8

The following data were collected from each study included in the meta-analysis:
first author’s name, publication date, country, total numbers of cases and
controls, and frequency of ATG16L1 genotypes in cases and
controls.

Strength of agreement between the investigators regarding study selection was
evaluated using the Kappa statistic. The combined ORs and 95% CIs were
calculated for the allelic model (G allele versus A allele), the additive model
(GG versus AA), the dominant model (GA + GG versus AA) and the recessive model
(GG versus GA + AA) using either the random effects model9 or the fixed effects model.10 Galbraith plots were created to graphically assess the source of any
heterogeneity. Publication bias was analyzed using Begg’s funnel plots and
Egger’s test, with a P-value < 0.05 being considered
representative of statistically significant publication bias.11 Conformity with the Hardy–Weinberg equilibrium amongst the controls was
determined using the χ2-square test and was considered to be in
agreement when the P-value is ≥ 0.05. All statistical analyses
were performed using Stata statistical software version 11.0 (StataCorp, College
Station, TX, USA).

A total of 843 potentially relevant articles were initially identified. After
exclusion of duplicate studies and application of the inclusion criteria, a
total of 44 articles12–55 were included in the
qualitative synthesis (Figure
1). Büning et al.13 contained three separate case–control studies and Fowler et al.19 contained two separate case–control studies; therefore, a total of 47
case–control studies involving 18 638 cases and 30 181 controls were included in
the final meta-analysis. The main characteristics of these studies are given in
Table 1. Figure 1.Flow diagram of the study selection process. CD, Crohn’s disease.
Table 1.Main characteristics of studies included in the meta-analysis.ReferenceSource of subjectsGenotype and allele distribution
(case/control)HWENOS scoreGGGAAAGConformsStatistical significanceBaldassano et al., 200712Population-based58/7865/13619/67181/292YesNS6Büning et al., 2007:13 study 1Population-based98/68149/14363/74345/279YesNS6Büning et al., 2007:13 study 2Population-based38/4986/10923/49162/207YesNS6Büning et al., 2007:13 study 3Population-based60/6678/10219/47198/234YesNS6Cummings et al., 200714Hospital-based209/196282/33081/157700/722YesNS6Prescott et al., 200715Population-based435/321565/626236/2881435/1268YesNS6Roberts et al., 200716Population-based166/130243/28587/134575/545YesNS7Yamazaki et al., 200717Population-based23/32184/167274/238230/231YesNS6Baptista et al., 200818Population-based46/4294/9040/57186/174YesNS8Fowler et al., 2008:19 study 1Population-based243/339315/601111/304801/1279YesNS6Fowler et al., 2008:19 study 2Population-based59/11073/18922/121191/409NoP = 0.046Gaj et al., 200820Population-based24/3225/7011/3773/134YesNS8Glas et al., 200821Population-based———906/1673N/AN/A8Hancock et al., 200822Population-based216/321288/56982/266720/1211YesNS7Lakatos et al., 200823Population-based92/33125/8349/33309/149YesNS7Lappalainen et al., 200824Population-based———232/179N/AN/A6Latiano et al., 200825Population-based227/214335/376105/159789/804YesNS7Okazaki et al., 200826Population-based77/88103/15028/76257/326YesNS8Perricone et al. 200827Population-based33/3073/7657/54139/136YesNS7Peterson et al., 200828Population-based———655/505N/AN/A6Van Limbergen et al., 200829Population-based217/98294/176118/71728/372YesNS6Weersma et al., 200830Population-based121/280125/42840/163367/988YesNS7Amre et al., 200931Population-based102/64137/13547/91341/263YesNS8Dema et al., 200932Population-based178/246314/407115/206670/899YesNS7Dusatkova et al., 200933Population-based107/132158/23968/128372/503YesNS7Lacher et al., 200934Population-based60/5673/12819/69193/240YesNS7Márquez et al., 200935Population-based125/221156/34763/177406/789YesNS7Newman et al., 200936Population-based159/253204/41572/227522/921NoP = 0.039Palomino-Morales et al., 200937Hospital-based216/183253/31675/167685/682YesNS7Cotterill et al., 201038Population-based———317/840N/AN/A7Csöngei et al., 201039Population-based108/79151/16356/72367/321YesNS7Gazouli et al., 201040Population-based189/161222/27463/104600/596YesNS6Sventoraityte et al., 201041Population-based16/4428/8911/5360/177YesNS8Fabio et al., 201142Population-based94/50134/9751/43322/197YesNS6Frank et al., 201143Hospital-based25/1722/1914/2372/53NoP = 0.0075Lauriola et al., 201144Population-based6/69/113/321/23YesNS6Jung et al., 201245Population-based———638/864N/AN/A6Wang et al., 201246Population-based44/33164/140141/179252/206YesNS6Hirano et al., 201347Population-based———1993/10141N/AN/A6Dalton et al., 201448Population-based22/849/3312/1493/49YesNS6Jakobsen et al., 201449Population-based———293/566N/AN/A7Scolaro et al., 201450Population-based25/4853/10628/84103/202YesNS8Serbati et al., 201451Population-based10/943/7616/3063/94NoP < 0.0016Zhang et al., 201452Population-based77/62134/166209/272288/290NoP < 0.0017Na et al., 201553Population-based———54/51N/AN/A7Salem et al., 201554Hospital-based108/2978/1350/15294/71NoP < 0.0016Yang et al., 201555Population-based226/211838/1033745/11921290/1455YesNS7HWE, Hardy–Weinberg equilibrium; N/A, not available; NOS,
Newcastle–Ottawa scale.NS, not statistically significant
(P ≥ 0.05).

Flow diagram of the study selection process. CD, Crohn’s disease.

Main characteristics of studies included in the meta-analysis.

HWE, Hardy–Weinberg equilibrium; N/A, not available; NOS,
Newcastle–Ottawa scale.

NS, not statistically significant
(P ≥ 0.05).

When all the studies were pooled in the meta-analysis, a significant association
was seen between ATG16L1 and Crohn’s disease in all four
genetic models (allelic model: OR = 1.29, 95% CI = 1.22, 1.37, Figure 2; additive model:
OR = 1.80, 95% CI = 1.68, 1.92, Figure 3; dominant model: OR = 1.47, 95% CI = 1.39, 1.55, Figure 4; recessive model:
OR = 1.46, 95% CI = 1.39, 1.54, Figure 5). When stratified by study design (population- or
hospital-based), a significant association between ATG16L1 and
Crohn’s disease was still seen in all four genetic models (Table 2). Figure 2.Forest plot of the association between ATG16L1 and
Crohn’s disease using the allelic model (G allele versus A allele).
The pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
random effects model.
Figure 3.Forest plot of the association between ATG16L1 and
Crohn’s disease using the additive model (GG versus AA). The pooled
odds ratio (OR) and 95% confidence intervals (CI) are indicated by
the diamond. Percentage weights were calculated using a fixed
effects model.
Figure 4.Forest plot of the association between ATG16L1 and
Crohn’s disease using the dominant model (GG + GA versus AA). The
pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
fixed effects model.
Figure 5.Forest plot of the association between ATG16L1 and
Crohn’s disease using the recessive model (GG versus GA + AA). The
pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
fixed effects model.
Table 2.Results of meta-analysis and subgroup analysis for the association
between ATG16L1 and Crohn’s disease according to
the allelic, additive, dominant and recessive models.Group analysedAllelic model (G vs A)Additive model (GG vs AA)Dominant model (GG + GA vs AA)Recessive model (GG vs GA + AA)OR (95% CI)Analysis modelOR (95% CI)Analysis modelOR (95% CI)Analysis modelOR (95% CI)Analysis modelAll1.29 (1.22, 1.37)Random effects1.80 (1.68, 1.92)Fixed effects1.47 (1.39, 1.55)Fixed effects1.46 (1.39, 1.54)Fixed effectsPopulation-based1.28 (1.20, 1.37)Random effects1.76 (1.64, 1.89)Fixed effects1.44 (1.36, 1.52)Fixed effects1.46 (1.38, 1.54)Fixed effectsHospital-based1.46 (1.31, 1.62)Fixed effects2.18 (1.76, 2.70)Fixed effects1.86 (1.54, 2.26)Fixed effects1.51 (1.29, 1.76)Fixed effectsNOS score ≥ 71.33 (1.24, 1.43)Random effects1.83 (1.68, 1.99)Fixed effects1.49 (1.39, 1.59)Fixed effects1.47 (1.37, 1.57)Fixed effectsConform to HWE1.32 (1.24, 1.40)Random effects1.79 (1.67, 1.92)Fixed effects1.46 (1.38, 1.54)Fixed effects1.46 (1.38, 1.54)Fixed effectsOR, odds ratio; CI, confidence intervals; NOS, Newcastle–Ottawa
Scale; HWE, Hardy–Weinberg equilibrium.

Forest plot of the association between ATG16L1 and
Crohn’s disease using the allelic model (G allele versus A allele).
The pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
random effects model.

Forest plot of the association between ATG16L1 and
Crohn’s disease using the additive model (GG versus AA). The pooled
odds ratio (OR) and 95% confidence intervals (CI) are indicated by
the diamond. Percentage weights were calculated using a fixed
effects model.

Forest plot of the association between ATG16L1 and
Crohn’s disease using the dominant model (GG + GA versus AA). The
pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
fixed effects model.

Forest plot of the association between ATG16L1 and
Crohn’s disease using the recessive model (GG versus GA + AA). The
pooled odds ratio (OR) and 95% confidence intervals (CI) are
indicated by the diamond. Percentage weights were calculated using a
fixed effects model.

Results of meta-analysis and subgroup analysis for the association
between ATG16L1 and Crohn’s disease according to
the allelic, additive, dominant and recessive models.

OR, odds ratio; CI, confidence intervals; NOS, Newcastle–Ottawa
Scale; HWE, Hardy–Weinberg equilibrium.

Sensitivity analyses were conducted to determine whether modification of the
inclusion criteria of the meta-analysis affected the final results. When the
included studies were limited to those conforming to the Hardy–Weinberg
equilibrium (P ≥ 0.05), the pooled ORs of these 33 studies were
not materially different from those of the full meta-analysis (Table 2). Likewise,
when the included studies were limited to those with a high NOS score (≥7), the
pooled ORs of these 22 studies were not materially different from those of the
full meta-analysis (Table
2).

Significant heterogeneity existed in the allelic model (I2 = 75.4%). A
Galbraith plot was created to graphically assess the source of heterogeneity
(Figure 6). The
studies by Yamazaki et al.,17 Fowler et al.19 (study 1), Latiano et al.,25 Amre et al.,31 Lacher et al.,34 Palomino-Morales et al.,37 Jung et al.45 and Hirano et al.47 were identified as contributors to the heterogeneity. When these eight
studies were excluded, the I2 was 0.0% and the OR (95% CI) was 1.33
(1.28, 1.37). Figure 6.Galbraith plot of the allelic model. The outliers were the studies by
Yamazaki et al.,17 Fowler et al.19 (study 1), Latiano et al.,25 Amre et al.,31 Lacher et al.,34 Palomino-Morales et al.,37 Jung et al.45 and Hirano et al.47 b, effect estimate; se, standard error.

Galbraith plot of the allelic model. The outliers were the studies by
Yamazaki et al.,17 Fowler et al.19 (study 1), Latiano et al.,25 Amre et al.,31 Lacher et al.,34 Palomino-Morales et al.,37 Jung et al.45 and Hirano et al.47 b, effect estimate; se, standard error.

The shapes of the Begg’s funnel plots did not reveal any evidence of obvious
asymmetry (Figure 7). No
statistical evidence of publication bias was found using Egger’s regression test
(P = 0.09 for the allelic model; P = 0.62
for the additive model; P = 0.08 for the dominant model; and
P = 0.83 for the recessive model). Figure 7.Begg’s funnel plots with pseudo 95% confidence limits of all studies
in the meta-analysis using the four model types: (a) allelic model
(G allele versus A allele); (b) additive model (GG versus AA); (c)
dominant model (GG + GA versus AA); (d) recessive model (GG versus
GA + AA). SE, standard error; OR, odds ratio.

Begg’s funnel plots with pseudo 95% confidence limits of all studies
in the meta-analysis using the four model types: (a) allelic model
(G allele versus A allele); (b) additive model (GG versus AA); (c)
dominant model (GG + GA versus AA); (d) recessive model (GG versus
GA + AA). SE, standard error; OR, odds ratio.

Since Hampe et al.5 reported in 2007 that ATG16L1 gene polymorphism was
associated with Crohn’s disease, many studies have evaluated the relationship
between ATG16L1 and the risk of Crohn’s disease.56 However, the results are inconsistent. As the strength of results from a
single case–control study is weak due to small sample sizes, the combination of many
studies in a meta-analysis has the benefit of overcoming this limitation by
increasing the sample size and generating more robust results. Meta-analysis has
been widely used in genetic association studies.57,58 The present meta-analysis was
performed to assess whether the combined evidence supports an association between
ATG16L1 and Crohn’s disease.

The present meta-analysis examined ATG16L1 gene polymorphism and its
relationship with the risk of Crohn’s disease based on data from 47 case–control
studies involving 18 638 cases and 30 181 controls. Most of these studies reported
that ATG16L1 was associated with the risk of Crohn’s disease, but
not all. The results of the meta-analyses demonstrated that overall there was
evidence of a significant association between ATG16L1 gene
polymorphism and Crohn’s disease. This significant association remained in all four
genetic models when subgroup analyses were performed based on study design
(population-based or hospital-based).

When considering the potential mechanisms linking ATG16L1
polymorphism with an increased risk of Crohn’s disease, it has been shown that
ATG16L1 polymorphism impairs the autophagy processing of
pathogenic bacteria and the function of intestinal Paneth cells.59,60 In addition,
it has been shown that ATG16L1 polymorphism is associated with
increased susceptibility to Helicobacter pylori infection.61 In patients with Crohn’s disease, it has been reported that homozygosity of
the ATG16L1 risk allele (GG) was associated with a reduced ability
to clear pathosymbionts.62 Paneth cells in ATG16L1-deficient mice have been shown to be
dysfunctional and to demonstrate increased expression of pro-inflammatory
cytokines.63,64

When interpreting the results of this meta-analysis, a number of limitations should
be acknowledged. First, it is well known that both environmental factors and
individual genetic predisposition contribute to the development of Crohn’s disease.
Due to the lack of original data, however, potential interactions between these two
types of influence has not been evaluated. Secondly, ATG16L1 seems
to exert a close functional correlation with other genes in regulating autophagy.
For example, the interaction of ATG16L1 and NOD2
has been implicated in the pathogenesis of Crohn’s disease.63 Potential gene–gene interactions require further evaluation. Thirdly, the
ATG16L1 genotype has been reported to be associated with
disease phenotype,65 which has clinical significance. Further combined analyses are needed to
clarify the association between the ATG16L1 genotype and Crohn’s
disease phenotype.

In conclusion, the present meta-analysis of robust data and unbiased results
demonstrated an association between ATG16L1 genotype and the
development of Crohn’s disease. These findings will be helpful in understanding the
aetiology of Crohn’s disease and indicate that the ATG16L1 gene
might have potential as a therapeutic or diagnostic target.