---
reference_id: "PMID:26729329"
title: Enhanced diagnostic yield in Meckel-Gruber and Joubert syndrome through exome sequencing supplemented with split-read mapping.
authors:
- Watson CM
- Crinnion LA
- Berry IR
- Harrison SM
- Lascelles C
- Antanaviciute A
- Charlton RS
- Dobbie A
- Carr IM
- Bonthron DT
journal: BMC Med Genet
year: '2016'
doi: 10.1186/s12881-015-0265-z
content_type: abstract_only
---

# Enhanced diagnostic yield in Meckel-Gruber and Joubert syndrome through exome sequencing supplemented with split-read mapping.
**Authors:** Watson CM, Crinnion LA, Berry IR, Harrison SM, Lascelles C, Antanaviciute A, Charlton RS, Dobbie A, Carr IM, Bonthron DT
**Journal:** BMC Med Genet (2016)
**DOI:** [10.1186/s12881-015-0265-z](https://doi.org/10.1186/s12881-015-0265-z)

## Content

1. BMC Med Genet. 2016 Jan 4;17:1. doi: 10.1186/s12881-015-0265-z.

Enhanced diagnostic yield in Meckel-Gruber and Joubert syndrome through exome 
sequencing supplemented with split-read mapping.

Watson CM(1)(2), Crinnion LA(3)(4), Berry IR(5), Harrison SM(6), Lascelles C(7), 
Antanaviciute A(8), Charlton RS(9), Dobbie A(10), Carr IM(11), Bonthron 
DT(12)(13).

Author information:
(1)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. c.m.watson@leeds.ac.uk.
(2)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. c.m.watson@leeds.ac.uk.
(3)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. l.a.crinnion@leeds.ac.uk.
(4)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. l.a.crinnion@leeds.ac.uk.
(5)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. ianberry@nhs.net.
(6)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. s.harrison@leeds.ac.uk.
(7)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. c.lascelles@leeds.ac.uk.
(8)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. umaan@leeds.ac.uk.
(9)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. ruth.charlton1@nhs.net.
(10)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. angus.dobbie@nhs.net.
(11)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. i.m.carr@leeds.ac.uk.
(12)Yorkshire Regional Genetics Service, St. James's University Hospital, Leeds, 
LS9 7TF, UK. d.t.bonthron@leeds.ac.uk.
(13)School of Medicine, University of Leeds, St. James's University Hospital, 
Leeds, LS9 7TF, UK. d.t.bonthron@leeds.ac.uk.

BACKGROUND: The widespread adoption of high-throughput sequencing technologies 
by genetic diagnostic laboratories has enabled significant expansion of their 
testing portfolios. Rare autosomal recessive conditions have been a particular 
focus of many new services. Here we report a cohort of 26 patients referred for 
genetic analysis of Joubert (JBTS) and Meckel-Gruber (MKS) syndromes, two 
clinically and genetically heterogeneous neurodevelopmental conditions that 
define a phenotypic spectrum, with MKS at the severe end.
METHODS: Exome sequencing was performed for all cases, using Agilent SureSelect 
v5 reagents and Illumina paired-end sequencing. For two cases medium-coverage 
(9Ã—) whole genome sequencing was subsequently undertaken.
RESULTS: Using a standard analysis pipeline for the detection of single 
nucleotide and small insertion or deletion variants, molecular diagnoses were 
confirmed in 12 cases (4%). Seeking to determine whether our cohort harboured 
pathogenic copy number variants (CNV), in JBTS- or MKS-associated genes, 
targeted comparative read-depth analysis was performed using FishingCNV. These 
analyses identified a putative intragenic AHI1 deletion that included three 
exons spanning at least 3.4 kb and an intergenic MPP4 to TMEM237 deletion that 
included exons spanning at least 21.5 kb. Whole genome sequencing enabled 
confirmation of the deletion-containing alleles and precise characterisation of 
the mutation breakpoints at nucleotide resolution. These data were validated 
following development of PCR-based assays that could be subsequently used for 
"cascade" screening and/or prenatal diagnosis.
CONCLUSIONS: Our investigations expand the AHI1 and TMEM237 mutation spectrum 
and highlight the importance of performing CNV screening of disease-associated 
genes. We demonstrate a robust increasingly cost-effective CNV detection 
workflow that is applicable to all MKS/JBTS referrals.

DOI: 10.1186/s12881-015-0265-z
PMCID: PMC4700600
PMID: 26729329 [Indexed for MEDLINE]