---
reference_id: "PMID:35163737"
title: Clinical Utility of a Unique Genome-Wide DNA Methylation Signature for KMT2A-Related Syndrome.
authors:
- Foroutan A
- Haghshenas S
- Bhai P
- Levy MA
- Kerkhof J
- McConkey H
- Niceta M
- Ciolfi A
- Pedace L
- Miele E
- Genevieve D
- Heide S
- Alders M
- Zampino G
- Merla G
- Fradin M
- Bieth E
- Bonneau D
- Dieterich K
- Fergelot P
- Schaefer E
- Faivre L
- Vitobello A
- Maitz S
- Fischetto R
- Gervasini C
- Piccione M
- van de Laar I
- Tartaglia M
- Sadikovic B
- Lebre AS
journal: Int J Mol Sci
year: '2022'
doi: 10.3390/ijms23031815
content_type: abstract_only
---

# Clinical Utility of a Unique Genome-Wide DNA Methylation Signature for KMT2A-Related Syndrome.
**Authors:** Foroutan A, Haghshenas S, Bhai P, Levy MA, Kerkhof J, McConkey H, Niceta M, Ciolfi A, Pedace L, Miele E, Genevieve D, Heide S, Alders M, Zampino G, Merla G, Fradin M, Bieth E, Bonneau D, Dieterich K, Fergelot P, Schaefer E, Faivre L, Vitobello A, Maitz S, Fischetto R, Gervasini C, Piccione M, van de Laar I, Tartaglia M, Sadikovic B, Lebre AS
**Journal:** Int J Mol Sci (2022)
**DOI:** [10.3390/ijms23031815](https://doi.org/10.3390/ijms23031815)

## Content

1. Int J Mol Sci. 2022 Feb 5;23(3):1815. doi: 10.3390/ijms23031815.

Clinical Utility of a Unique Genome-Wide DNA Methylation Signature for 
KMT2A-Related Syndrome.

Foroutan A(1)(2), Haghshenas S(1)(2), Bhai P(2), Levy MA(2), Kerkhof J(2), 
McConkey H(2), Niceta M(3), Ciolfi A(3), Pedace L(4), Miele E(4), Genevieve 
D(5), Heide S(6), Alders M(7), Zampino G(8)(9), Merla G(10)(11), Fradin M(12), 
Bieth E(13), Bonneau D(14), Dieterich K(15), Fergelot P(16), Schaefer E(17), 
Faivre L(18)(19), Vitobello A(18)(19), Maitz S(20), Fischetto R(21), Gervasini 
C(22), Piccione M(23), van de Laar I(24), Tartaglia M(3), Sadikovic B(1)(2), 
Lebre AS(25)(26).

Author information:
(1)Department of Pathology and Laboratory Medicine, Western University, London, 
ON N6A 3K7, Canada.
(2)Verspeeten Clinical Genome Centre, London Health Sciences Centre, London, ON 
N6A 5W9, Canada.
(3)Genetics and Rare Diseases Research Division, Ospedale Pediatrico Bambino 
Gesù, IRCCS, 00146 Rome, Italy.
(4)Department of Pediatric Onco-Hematology and Cell and Gene Therapy, Ospedale 
Pediatrico Bambino Gesù, IRCCS, 00146 Rome, Italy.
(5)Medical Genetic Department for Rare Diseases and Personalized Medicine, 
Reference Center AD SOOR, AnDDI-RARE, Groupe DI, Inserm U1183-Institute for 
Regenerative Medicine and Biotherapy, Montpellier University, Centre Hospitalier 
Universitaire de Montpellier, 34090 Montpellier, France.
(6)Department of Genetics, Referral Center for Intellectual Disabilities, APHP 
Sorbonne University, Pitié Salpêtrière Hospital, 75013 Paris, France.
(7)Department of Clinical Genetics, Amsterdam UMC, University of Amsterdam, 1105 
AZ Amsterdam, The Netherlands.
(8)Center for Rare Diseases and Congenital Defects, Fondazione Policlinico 
Universitario A. Gemelli, IRCCS, 00168 Rome, Italy.
(9)Facoltà di Medicina e Chirurgia, Università Cattolica del S. Cuore, 20123 
Roma, Italy.
(10)Department of Molecular Medicine and Medical Biotechnology, Università di 
Napoli "Federico II", 80131 Naples, Italy.
(11)Laboratory of Regulatory and Functional Genomics, Fondazione Casa Sollievo 
della Sofferenza, 71013 San Giovanni Rotondo, Italy.
(12)Service de Génétique, CHU de Rennes, 35203 Rennes, France.
(13)Medical Genetics Department, University of Angers, CHU Angers, 49000 Angers, 
France.
(14)Department of genetics, CHU d'Angers, 49000 Angers, France and MitoVasc, UMR 
CNRS 6015-INSERM 1083, University of Angers, 49055 Angers, France.
(15)CHU Grenoble Alpes, Inserm, U1209, Institute of Advanced Biosciences, 
Université Grenoble Alpes, 38000 Grenoble, France.
(16)Medical Genetics Department, Inserm U1211, Reference Center AD SOOR, 
AnDDI-RARE, Bordeaux University, Centre Hospitalier Universitaire de Bordeaux, 
33076 Bordeaux, France.
(17)Service de Génétique Médicale-Institut de Génétique Médicale 
d'Alsace-Hôpitaux Universitaires de Strasbourg, 67091 Strasbourg, France.
(18)Inserm, UMR1231, Equipe GAD, Bâtiment B3, Université de Bourgogne Franche 
Comté, 15 boulevard du Maréchal de Lattre de Tassigny, 21000 Dijon, France.
(19)Unité Fonctionnelle Innovation en Diagnostic Génomique des Maladies Rares, 
FHU-TRANSLAD, Department of Medical Genetics, Dijon University Hospital, 21000 
Dijon, France.
(20)Clinical Pediatric Genetics Unit, Pediatrics Clinics, MBBM Foundation, S. 
Gerardo Hospital, 20900 Monza, Italy.
(21)Clinical Genetics Unit, Department of Pediatric Medicine, Giovanni XXIII 
Children's Hospital, 02115 Bari, Italy.
(22)Medical Genetics, Department of Health Sciences, Università degli Studi di 
Milano, 20142 Milan, Italy.
(23)Department of Sciences for Health Promotion and Mother and Child Care "G. 
D'Alessandro", University of Palermo, 90127 Palermo, Italy.
(24)Department of Clinical Genetics, Erasmus MC, University Medical Center 
Rotterdam, 3000 CA Rotterdam, The Netherlands.
(25)Team Physiopathologie des Maladies Psychiatriques, GDR3557-Institut de 
Psychiatrie, Institute of Psychiatry and Neuroscience of Paris (IPNP), INSERM 
U1266, Université de Paris, 75006 Paris, France.
(26)Centre Hospitalier Universitaire de Reims, Pôle de Biologie Médicale et 
Pathologie, Service de GénéTique, 51100 Reims, France.

Wiedemann-Steiner syndrome (WDSTS) is a Mendelian syndromic intellectual 
disability (ID) condition associated with hypertrichosis cubiti, short stature, 
and characteristic facies caused by pathogenic variants in the KMT2A gene. 
Clinical features can be inconclusive in mild and unusual WDSTS presentations 
with variable ID (mild to severe), facies (typical or not) and other associated 
malformations (bone, cerebral, renal, cardiac and ophthalmological anomalies). 
Interpretation and classification of rare KMT2A variants can be challenging. A 
genome-wide DNA methylation episignature for KMT2A-related syndrome could allow 
functional classification of variants and provide insights into the 
pathophysiology of WDSTS. Therefore, we assessed genome-wide DNA methylation 
profiles in a cohort of 60 patients with clinical diagnosis for WDSTS or Kabuki 
and identified a unique highly sensitive and specific DNA methylation 
episignature as a molecular biomarker of WDSTS. WDSTS episignature enabled 
classification of variants of uncertain significance in the KMT2A gene as well 
as confirmation of diagnosis in patients with clinical presentation of WDSTS 
without known genetic variants. The changes in the methylation profile resulting 
from KMT2A mutations involve global reduction in methylation in various genes, 
including homeobox gene promoters. These findings provide novel insights into 
the molecular etiology of WDSTS and explain the broad phenotypic spectrum of the 
disease.

DOI: 10.3390/ijms23031815
PMCID: PMC8836705
PMID: 35163737 [Indexed for MEDLINE]

Conflict of interest statement: The authors declare that they have no competing 
interests.