---
reference_id: "PMID:26204230"
title: "CTLA-4 as a genetic determinant in autoimmune Addison's disease."
authors:
- Wolff AS
- Mitchell AL
- Cordell HJ
- Short A
- Skinningsrud B
- Ollier W
- Badenhoop K
- Meyer G
- Falorni A
- Kampe O
- Undlien D
- Pearce SH
- Husebye ES
journal: Genes Immun
year: '2015'
doi: 10.1038/gene.2015.27
content_type: full_text_xml
---

# CTLA-4 as a genetic determinant in autoimmune Addison's disease.
**Authors:** Wolff AS, Mitchell AL, Cordell HJ, Short A, Skinningsrud B, Ollier W, Badenhoop K, Meyer G, Falorni A, Kampe O, Undlien D, Pearce SH, Husebye ES
**Journal:** Genes Immun (2015)
**DOI:** [10.1038/gene.2015.27](https://doi.org/10.1038/gene.2015.27)

## Content

1. Genes Immun. 2015 Sep;16(6):430-6. doi: 10.1038/gene.2015.27. Epub 2015 Jul
23.

CTLA-4 as a genetic determinant in autoimmune Addison's disease.

Wolff AS(1), Mitchell AL(2), Cordell HJ(2), Short A(3), Skinningsrud B(4)(5), 
Ollier W(3), Badenhoop K(6), Meyer G(6), Falorni A(7), Kampe O(8), Undlien 
D(4)(5), Pearce SH(2), Husebye ES(1)(9).

Author information:
(1)Department of Clinical Science, University of Bergen, Bergen, Norway.
(2)Institute of Genetic Medicine, Newcastle University, Newcastle upon Tyne, UK.
(3)Centre for Integrated Genomic Medical Research, Institute of Population 
Health, Manchester University, Manchester, UK.
(4)Institute of Medical Genetics, University of Oslo, Oslo, Norway.
(5)Department of Medical Genetics, Oslo University Hospital, Oslo, Norway.
(6)Department of Endocrinology and Diabetes, Internal Medicine 1, 
Johann-Wolfgang-Goethe-University's Hospital, Frankfurt, Germany.
(7)Department of Medicine, University of Perugia, Perugia, Italy.
(8)Department of Medicine, Solna, Karolinska University Hospital, Karolinska 
Institutet, Stockholm, Sweden.
(9)Department of Medicine, Haukeland University Hospital, Bergen, Norway.

In common with several other autoimmune diseases, autoimmune Addison's disease 
(AAD) is thought to be caused by a combination of deleterious susceptibility 
polymorphisms in several genes, together with undefined environmental factors 
and stochastic events. To date, the strongest genomic association with AAD has 
been with alleles at the HLA locus, DR3-DQ2 and DR4. The contribution of other 
genetic variants has been inconsistent. We have studied the association of 16 
single-nucleotide polymorphisms (SNPs) within the CD28-CTLA-4-ICOS genomic 
locus, in a cohort comprising 691 AAD patients of Norwegian and UK origin with 
matched controls. We have also performed a meta-analysis including 1002 patients 
from European countries. The G-allele of SNP rs231775 in CTLA-4 is associated 
with AAD in Norwegian patients (odds ratio (OR)=1.35 (confidence interval (CI) 
1.10-1.66), P=0.004), but not in UK patients. The same allele is associated with 
AAD in the total European population (OR=1.37 (CI 1.13-1.66), P=0.002). A 
three-marker haplotype, comprising PROMOTER_1661, rs231726 and rs1896286 was 
found to be associated with AAD in the Norwegian cohort only (OR 2.43 (CI 
1.68-3.51), P=0.00013). This study points to the CTLA-4 gene as a susceptibility 
locus for the development of AAD, and refines its mapping within the wider 
genomic locus.

DOI: 10.1038/gene.2015.27
PMCID: PMC4561510
PMID: 26204230 [Indexed for MEDLINE]

Conflict of interest statement: The authors declare no conflict of interest.

Primary adrenal insufficiency (Addison’s disease, AD) is most often caused by autoimmune destruction of the adrenal cortices, resulting in insufficient production of the steroid hormones, cortisol and aldosterone. Autoimmune diseases, including autoimmune AD (AAD), cluster in families and are believed to share some common etiological factors. The underlying mechanisms resulting in AAD are still obscure, but a contribution from genetic variants is clear.1, 2 The strongest genomic associations have been reported with the human leukocyte antigen (HLA) haplotypes DR3–DQ2 and DR4 (subtype DRB1*0404)-DQ8) in several populations.3, 4, 5, 6 Other associations include variants in genes involved in T-cell signalling and activation, especially the 1858T allele (rs2476601) of the tyrosine-protein phosphatase nonreceptor type 22 (PTPN22) gene.7 The product of this gene, lymphoid tyrosine phosphatase (LYP), acts as an essential mediator of T-cell receptor (TCR) signalling by controlling phosphorylation/dephosphorylation of key signalling pathway kinases.8 Polymorphisms in STAT4, linked to CD4+ cell fate and GATA3, involved in CD8+ cell homeostasis, have also been found to confer susceptibility to AAD.9 In addition, other genes that encode cell-surface receptors that regulate activation responses in T lymphocytes, for example, CD274 (programmed death ligand 1)10 and cytotoxic T-lymphocyte antigen-4, CTLA-4 (CD152)11, 12, 13 have also been linked to AAD susceptibility. CTLA-4, which is encoded by a gene located on chromosome 2q33, competes with the costimulator CD28 for binding to B7 on antigen-presenting cells, and downregulates T-cell responses to antigen-presentation/T-cell receptor engagement. Several studies have implicated CTLA-4 single-nucleotide polymorphisms (SNPs) in autoimmune diseases, including AAD. Recently, Schubert et al.14 described heterozygous mutations in CTLA-4 that resulted in a dominantly inherited complex immune dysregulation syndrome due to disruption of immune cell balance. Importantly, slightly different patterns of genetic association have been seen in different autoimmune conditions, including with alleles of the flanking CD28 and inducible costimulator (ICOS) genes, suggesting that different variants within this extended locus could predispose to different autoimmune conditions.15 Etiological variants at the CTLA-4 locus might result in either a deficient or an overexuberant CTLA-4 response relative to CD28, which interferes with immune homeostasis. Furthermore, CTLA-4 is constitutively expressed on regulatory T cells (Tregs). It has been suggested that this is one of the crucial, innate protective mechanisms against infectious agents and prevents the expansion of pathogenic T-cell clones.16 Blockade of CTLA-4 activity has also been shown to abrogate Treg function in several experimental situations, perhaps promoting an environment permissive to autoreactivity.17, 18

The association between CTLA-4 gene variants and AAD has previously been reported in some but not all studies of different European populations.3, 11, 12, 13, 19, 20, 21, 22 Of particular interest is the common nonsynonymous CTLA-4 polymorphism rs231775, also called exon 1 +49A→G (Ala17Thr). Although this SNP was found to be associated with AAD in Italy,3 this finding was not replicated in studies from Germany,21 Spain22 or Norway.20 In the UK population, results have been conflicting with one study finding an association12 and another failing to replicate this finding.20 Other polymorphisms in CTLA-4 have shown similar inconclusive patterns of association.3, 11, 20

The main limitation of all the previous studies of CTLA-4 gene variation in AAD has been insufficient statistical power due to the small patient cohorts studied, owing to the rarity of AAD. To overcome this, we have now collected DNA from 691 AAD patients from Norway and the UK and explored several polymorphisms in the CD28-CTLA-4-ICOS region within these cohorts using a tag-SNP approach. We also performed a meta-analysis of studies that have examined the CTLA-4 rs231775 variant in European populations.

Sixteen SNPs were analysed in the extended CTLA-4 gene locus in the UK and Norwegian AAD cohorts and compared with the allele and genotype frequencies of healthy controls of the same ethnic origins (Allele distribution with re-labelling #1–16 shown in Figure 1). Two SNPs did not meet the call rate criteria (that is, combined call rate>95%) from the UK cohort (rs11571308 (#6) and rs231730 (#12)) and from the Norwegian cohort (rs1181389 (#3) and rs5742909 (#7)). These were excluded from further analysis.Figure 1An overview of the SNPs that were analysed in this study. For simplicity, the SNPs are also labelled 1–16. *5′ near gene; **3′ near gene.

An overview of the SNPs that were analysed in this study. For simplicity, the SNPs are also labelled 1–16. *5′ near gene; **3′ near gene.

Genotypes and allele frequencies of one SNP, rs3181096 (#2), located close to the start of the CD28 gene (Figure 1), were associated with AAD in the combined cohort (all UK and Norwegian patients), and in the Norwegian cohort alone (Table 1). When we further separated the Norwegian patients into isolated AAD (iAAD) or APS II, or those with 21-hydroxylase (21-OH) antibodies, all subgroups remained significantly associated. However, there was no association in the UK patients when analysed alone (Table 1).Table 1The distribution of alleles of 16 SNPs in the extended CTLA-4 gene locus encompassing CD28, CTLA-4 and ICOS in the Norwegian and UK patients and controls cohortsDIL107AAD GAAD TCtrls GCtrls TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4891295001500.9851.140.950.34448.24P=0.52P=0.39Norway5751775911550.980.85−1.290.19951.76P=0.99P=0.57P=0.22Meta random0.98−0.140.8852.450.12P=0.65P=0.06rs3181096AAD CAAD TCtrls CCtrls TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4121984122180.9651.10.80.42445.6P=0.31P=0.73Norway5042364642840.9751.312.460.01454.4P=0.054P=0.041P=0.006Meta random1.212.220.0271.120.29P=0.03P=0.13rs1181389AAD TAAD CCtrl TCtrl CCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4761045831020.950.8−1.460.14545.98P=0.43P=0.46rs1181388AAD GAAD ACtrl GCtrl ACall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK515815511050.971.211.20.23147.55P=0.29P=0.15Norway6361146311030.9750.91−0.640.52552.45P=0.50P=0.69P=0.52Meta random1.040.30.7671.720.19P=0.84P=0.77rs3116507AAD AAAD GCtrl ACtrl GCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4751414961440.980.98−0.160.86945.380.40P=0.57Norway5651835701640.9750.89−0.960.33554.620.54P=0.35P=0.21Meta random0.93−0.820.410.280.6P=0.92P=0.18rs11571308AAD CAAD TCtrl CCtrl TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NNorway64692651870.970.94−0.40.6959.04P=0.47P=1.0P=0.90rs5742909AAD CAAD TCtrl CCtrl TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK56044588360.9550.78−1.070.28337.72P=0.57P=0.11Promoter_1661AAD AAAD GCtrl ACtrl GCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK48694556940.9560.87−0.850.39642.941.00P=0.92Norway6111355981220.9650.92−0.580.56257.060.53P=0.73P=0.61Meta random0.9−0.990.320.070.79P=0.63P=0.27rs231775AAD AAAD GCtrl ACtrl GCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK3612574002600.9950.91−0.80.42547.520.19P=0.013Norway3823704373130.9850.74−2.90.00452.480.10P=0.003P=0.003Meta random0.82−1.910.0551.870.17P=0.04P=0.27rs231726AAD CAAD TCtrl CCtrl TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK3872114102220.9550.99−0.060.95448.49P=0.67P=0.008Norway4113254672690.9650.73−2.970.00351.51P=0.10P=0.002P=0.002Meta random0.85−1.080.2823.750.05P=0.16P=0.40rs231727AAD GAAD ACtrl GCtrl ACall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK3812074332250.9650.96−0.370.70947.47P=0.41P=0.023Norway4203284572690.970.75−2.660.00852.53P=0.09P=0.007P=0.007Meta Random0.84−1.430.1542.220.14P=0.10P=0.36rs231730AAD TAAD ACtrl TCtrl ACall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NNorway5861505791410.9550.95−0.380.70456.97P=0.35P=0.85P=0.85rs2882973AAD TAAD CCtrl TCtrl CCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4012154312250.9850.97−0.230.82146.53P=0.56P=0.13Norway4533014902620.990.8−2.040.04253.47P=0.20P=0.048P=0.02Meta random0.88−1.350.1761.440.23P=0.18P=0.37rs3116505AAD CAAD TCtrl CCtrl TCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK3952114112170.960.99-0.10.92246.71P=0.68P=0.10Norway4282864772590.950.81-1.910.05653.29P=0.30P=0.045P=0.03Meta random0.89-1.190.2351.470.23P=0.26P=0.46rs10183087AAD AAAD CCtrl ACtrl CCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK4441584561720.9551.060.450.65150.91P=0.89P=0.66Norway5941465951430.970.98−0.170.86449.09P=0.45P=0.39P=0.94Meta random1.020.20.8390.190.66P=0.54P=0.83rs1896286AAD TAAD GCtrl TCtrl GCall rateORZPQ heteroP heteroWeightiAADAPS II21-OH+ NUK3722383822740.9851.1210.31946.82P=0.67P=0.42Norway4452754442900.9551.060.510.60753.18P=0.38P=0.96P=0.46Meta random1.091.060.290.140.71P=0.35P=0.42Abbreviations: AAD, autoimmune Addison’s disease; APS II, autoimmune polyendocrinopathy type II; APS II, P-value for analysis of only APS II; Ctrl, control subjects; OR, odds ratio; Z, Z-value for meta-analysis; P, P-value for the different cohorts and for the meta-analysis using the random model for calculations; Q hetero, Q-value for heterogeneity; P hetero, P-value for heterogeneity; Weight, relative weight using the random model; iAAD, P-value for analysis of only isolated AAD; 21-OH+ N, P-value for analysis of Norwegian patients with positive anti-21-OH autoantibodies.The distribution of alleles is given in numbers for the cohort of patients with AAD (cases) and for control subjects (Ctrl). The call rates given in the table are the combined call rate for controls and patients. All SNPs with a combined call rate of <95% have been rejected. Analyses have been executed by the program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com) using a random effects model. P-values which mark a significant association (P<0.05) are marked in bold text. Norwegian cohort: Total AAD cases: N=382 (iAAD N=162; APS II N=220). Total controls: N=380. UK cohort: Total AAD cases: N=309 (iAAD N=135; APS II N=174). Total controls: N=335.

The distribution of alleles of 16 SNPs in the extended CTLA-4 gene locus encompassing CD28, CTLA-4 and ICOS in the Norwegian and UK patients and controls cohorts

Abbreviations: AAD, autoimmune Addison’s disease; APS II, autoimmune polyendocrinopathy type II; APS II, P-value for analysis of only APS II; Ctrl, control subjects; OR, odds ratio; Z, Z-value for meta-analysis; P, P-value for the different cohorts and for the meta-analysis using the random model for calculations; Q hetero, Q-value for heterogeneity; P hetero, P-value for heterogeneity; Weight, relative weight using the random model; iAAD, P-value for analysis of only isolated AAD; 21-OH+ N, P-value for analysis of Norwegian patients with positive anti-21-OH autoantibodies.

The distribution of alleles is given in numbers for the cohort of patients with AAD (cases) and for control subjects (Ctrl). The call rates given in the table are the combined call rate for controls and patients. All SNPs with a combined call rate of <95% have been rejected. Analyses have been executed by the program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com) using a random effects model. P-values which mark a significant association (P<0.05) are marked in bold text. Norwegian cohort: Total AAD cases: N=382 (iAAD N=162; APS II N=220). Total controls: N=380. UK cohort: Total AAD cases: N=309 (iAAD N=135; APS II N=174). Total controls: N=335.

Five SNPs, all located within, or very close to, CTLA-4, were associated with AAD in the Norwegian cohort: rs231775 (#9), rs231726 (10), rs231727 (#11), rs2882973 (#13) and rs3116505 (#14) (Table 1). In the combined cohort, the G-allele of rs231775 (#1), was associated. This association was also seen in both the Norwegian and UK APS II subgroups, and in the 21-OH positive Norwegian subgroup. At rs231726 (#10), rs231727 (#11), rs2882973 (#13) and rs3116505 (#14), the association was specific for Norwegian APS II patients, rather than for iAAD. SNPs in ICOS were not associated with AAD in the either cohort.

A three-marker haplotype, comprising PROMOTER_1661 (#7), rs231726 (#10) and rs1896286 (#16) was found to be associated with AAD in the Norwegian cohort only. The major haplotype (A–C–T) was found with a frequency of 34.9% in the Norwegian control cohort. Association was found with the A–T–T haplotype, found in 25.2% of Norwegian cases compared with only 15.8% of Norwegian controls (odds ratio (OR) 2.43 (confidence interval (CI) 1.68–3.51)). The stepwise inclusion of additional markers did not improve the model, and when the data set was conditioned for SNPs within the CTLA-4 gene, markers in the CD28 gene were not found to be exerting an independent effect.

The SNP rs231775 (#9) has previously been studied in other AAD populations and found to be associated with disease. We confirm a strong correlation between this SNP and development of AAD in a random effects meta-analysis comprising 1002 patients and 1614 controls (Table 2), of which ~60% came from our current study. Overall, the OR for the association of the G-allele with AAD was 1.37 (CI 1.13–1.66), P=0.002.Table 2Meta-analysis on the effect of the CTLA-4 SNP rs231775 for AADStudy nameA AADG AADA CtrlsG CtrlsOdds ratio for G (95% CI)Z-valueP-v
alueRelative weightUK EuradrenalAAD N=309;Ctrls N=3353612574002601.10 (0.88–1.37)−0.800.42502528.13Norway EuradrenalAAD, N=382;Ctrls, N=3803823704373131.35 (1.10–1.66)−2.900.00368833.87Brozzetti et al.3AAD, N=180;Ctrls, N=3942141465782101.88 (1.44–2.44)−4.690.00000320.31Donner et al.21AAD, N=76Ctrls; N=39485675813581.28 (0.91–1.81)−1.390.16329811.73Perez de Nanclares et al.22AAD, N=57;Ctrls, N=1117539160621.34 (0.83-2.18)−1.190.2350725.96TOTAL random modelaAAD, N=1002;Ctrls, N=16141121885213611861.37 (1.13–1.66)−3.150.002Meta-analysis was performed using the program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com), random effects model. Significant associations are marked with bold text in the P-value column.aBeslow-Day test of heterogeneity: P=0.05; Hedges’s g test of heterogeneity: P=0.05.

Meta-analysis on the effect of the CTLA-4 SNP rs231775 for AAD

AAD N=309;

Ctrls N=335

AAD, N=382;

Ctrls, N=380

AAD, N=180;

Ctrls, N=394

AAD, N=76

Ctrls; N=394

AAD, N=57;

Ctrls, N=111

AAD, N=1002;

Ctrls, N=1614

Meta-analysis was performed using the program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com), random effects model. Significant associations are marked with bold text in the P-value column.

aBeslow-Day test of heterogeneity: P=0.05; Hedges’s g test of heterogeneity: P=0.05.

Variants in the CTLA-4 gene have been found to have a role in susceptibility to several autoimmune diseases in different populations, including autoimmune endocrinopathies such as Graves’ disease, autoimmune hypothyroidism and type 1 diabetes, (reviewed by Kavvoura et al.23, Kristiansen et al.24 and Ueda et al.25). The significance of CTLA-4 polymorphisms in AAD susceptibility remains controversial, as different studies have reported conflicting findings.3, 11, 12, 13, 19, 20, 21, 22 We have studied the contribution of SNPs in and around the CTLA-4 gene to AAD development in a large population comprising almost 700 patients from Norway and the UK. We conclude that several of the SNPs located in close proximity to the CD28 gene and to CTLA-4, including the G-allele of the common polymorphism rs231775 (#9), were significantly associated with AAD in the Norwegian population, but not in the UK patient cohort. This finding is in agreement with those of a previous study by Vaidya from 2000 which included 90 patients,12 but conflicts with the findings reported by Blomhoff from 2004 on 134 UK patients.20 However, these previous studies were conducted on much smaller cohorts, and hence are significantly less powerful than the current study. Other studies have indicated that CTLA-4, and not the neighbouring genes encoding CD28 and ICOS, are involved in susceptibility to autoimmune diseases.25, 26, 27 This study shows an association with CTLA-4 and AAD but no independent association between CD28 or ICOS and AAD, the latter of which had been previously suggested. We further show that for some of the SNPs in CTLA-4, the association with AAD is only true when AAD presents in combination with type 1 diabetes or thyroid disease (that is, APS II). It is possible that the positive association with these variants is due to both or one of these accompanying endocrine autoimmune conditions rather than to AAD in particular.

Haplotype analysis revealed that the haplotype PROMOTER_1661(A)-rs231726 (T)-rs1896286 (T) (#7#10#16) was significantly associated with AAD in the Norwegian cohort, strengthening the hypothesis of a role for CTLA-4 in AAD susceptibility. Markers in the CD28 gene were not found to exert an independent effect; hence, we speculate that the CD28 association with AAD is the result of linkage disequilibrium between CTLA-4 and CD28 alleles. A recent study suggested that the conflicting results of CD28, CTLA-4 and ICOS candidate gene association studies is due to population stratification. That is, the presence of systematic differences in allele frequencies between covert subpopulations within a population.28 This phenomenon is likely to explain the differing results in the UK and Norwegian cohorts studied here, even though the formal tests for genetic heterogeneity between the two populations were not overtly discrepant. It is also plausible that the real percentage of patients with Addison’s disease in whom the disease is in fact caused by pathological autoimmunity differ. However, we have tested the Norwegian cohort in this study and in a pilot subgroup of British AAD patients for autoantibodies against 21-OH, a positive marker for an ongoing autoimmune condition against the adrenals, and found similar results (data not shown). Deviations of results in different populations have also been reported for CTLA-4 associations with, for example, type 1 diabetes and autoimmune thyroid diseases (reviewed in Hou et al.29 and Wang et al.30). Notably, however, when the data indicate an association, it is most often with the same allele (for example, the G-allele concerning rs231775). As most autoimmune diseases are thought to be caused by a combination of unfavourable genetic factors (in addition to environmental triggers), a possibility exist that different genes have different impact on the association risk based on underlying genetics specific for each populations and ethnic groups. Hence, CTLA polymorphisms might be important for autoimmunity susceptibility in some populations, but overruled by other factors in other populations.

This study also confirms that the rs231775 (#9) SNP in CTLA-4 is significantly associated with AAD in a European meta-analysis. Interestingly, a previous report has demonstrated that this polymorphism exerts an effect on AAD susceptibility independent of prevalent HLA haplotypes.3 CD28 and CTLA-4 are important for T-cell function and signalling, and their regulation is therefore important for maintenance of normal lymphocyte biology. A polymorphism that diminishes CTLA-4 function or which increases CD28 function would be predicted to result in down-regulation of T-cell responses, favouring autoreactivity and therefore the development of autoimmune conditions. One important mechanism for controlling CTLA-4 and CD28 activity is by altering their cell-surface expression.31 In 2003, Ueda et al.25 demonstrated that a 3′-UTR haplotype of CTLA-4, including the CT60 SNP (rs30807243), was associated with altered levels of full-length (transmembrane) and soluble (s)CTLA-4 isoforms, the susceptibility haplotype being associated with reduced sCTLA-4 mRNA levels in man. Paradoxically, several studies have found increased levels of sCTLA-4 in the sera of individuals with autoimmune diseases, although newer data have challenged this hypothesis and the effect of certain genotypes on sCTLA-4 concentration remain controversial.32, 33, 34, 35, 36, 37, 38 People over 70 years of age have also been reported to have an increased serum level of sCTLA-4,39 which could be a contributing factor to the high prevalence of autoimmune diseases amongst the elderly. In accordance with our results, the G-allele of rs231775 (#9) has been shown to confer higher T-cell proliferation than the A-allele, indicating a lower degree of inhibition of T-cell activity, which may promote autoimmunity.40 However, this SNP, which is located on the peptide leader sequence of CTLA-4, may potentially also interfere with mRNA stability.

The impact of heterogeneity in our meta-analysis was borderline significant (P=0.05). We therefore used a random effects model which does not make the assumption that all studies included had a uniform direction of effect. The varying inclusion criteria between different studies and the differing frequency of HLA haplotypes between different European subpopulations are factors which might explain the genetic heterogeneity between the included studies. Nevertheless, all studies included in the meta-analysis indicate the same trend; that the G-allele of CTLA-4 SNP rs231775 (#9) contributes to AAD susceptibility.

In conclusion, this is the largest genetic study that has ever been conducted in AAD in relation to the CTLA-4 gene and we strengthen the evidence of an association between the CTLA-4 gene and AAD. In particular, the rs231775 (#9) polymorphism in the CTLA-4 signal peptide is associated with AAD in the European population, which supports a role for CTLA-4 in the pathogenesis of AAD.

AAD patients were recruited from the UK (N=309: iAAD N=135, autoimmune polyendocrinopathy type II (APS II) N=174) and Norway (N=382: iAAD N=162, APS II N=220). For the Norwegian cohort, 290 patients (76%) were 21-OH autoantibody positive, measured by radioimmunoassay as described elsewhere.41 In each AAD subject, the clinical diagnosis was confirmed by either a low basal cortisol with a high ACTH level or a subnormal response to the ACTH1–24 stimulation test (short synacthen test using 250 μg parenteral tetracosactide). Patients with primary adrenal failure owing to infiltrative or infective causes or secondary adrenal failure were excluded. In addition, patients with APS1, on the grounds of mucocutaneous candidiasis, hypoparathyroidism and ectodermal dystrophy, were also excluded and to our knowledge, all included individuals were nonconsanguineous. Patients with type 1 diabetes and/or autoimmune thyroid disease in addition to AAD were classified as having APS II. As controls, blood samples were collected from 335 and 380 healthy people from the UK and Norway, respectively. All individuals participating in this study gave informed, written consent. The study was approved by the Regional Ethical Committees (Leeds East in the UK; Western Norway in Norway).

DNA was extracted from peripheral blood. Sixteen SNPs within the extended CTLA-4 gene locus (encompassing CD28, CTLA-4 and ICOS) were chosen (Figure 1). Fourteen tag-SNPs were selected from the HapMap database, and an additional two SNPs (DIL107, promoter_1661) were chosen for their association with the autoimmune condition systemic lupus erythematosus.15 The DNA sequence flanking each SNP was found in the Ensembl database (~300 bp in length) and this sequence data was then used to generate a Sequenom iPlex assay, using the MassARRAY Designer software (Sequenom, San Diego, CA, USA). Compatibility of the multiplex assays was checked in silico. SNPs were genotyped using the Sequenom MassARRAY technology (Sequenom). The Iplex assay was followed according to manufacturer’s instructions (www.sequenom.com) using 30 ng of genomic DNA template.

The UK and Norwegian cohort data were analysed separately and then analysed in combination by meta-analysis. For each SNP a count was made of each genotype (AA, AB and BB) in cases and controls. Genotype frequencies were used to calculate call rates for each SNP (AA+AB+BB/sample size × 100). Genotypes at all markers were checked for Hardy–Weinberg Equilibrium (threshold P<0.01). SNPs with a combined call rate (healthy controls and patients) of <95% were excluded from the analysis. The program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com) using a random effects model was used in order to calculate ORs, z-scores, P-values. A subgroup analysis was then performed, whereby each cohort was divided into iAAD and APS II, (with and without 21-OH antibodies in the Norwegian cohort only).

Haploview42 was used to calculate nonrandom association of alleles (linkage disequilibrium expressed as r2 in the UK and Norwegian control cohorts (Figure 2)). The UNPHASED program43 was used to estimate haplotype frequencies and association in the UK and Norwegian cases and controls.Figure 2Haploview plots of linkage disequilibrium (LD; r2) between the SNPs for the UK control cohort analysed in this study. The plot for the Norwegian controls looked very similar. Some of the SNPs for each control cohort were excluded due to low call rate and these were not included in the Figure. To clarify which SNPs that are included in the graph, the SNPs have also been relabelled 1–16, which corresponds to the numbers given in Figure 1.

Haploview plots of linkage disequilibrium (LD; r2) between the SNPs for the UK control cohort analysed in this study. The plot for the Norwegian controls looked very similar. Some of the SNPs for each control cohort were excluded due to low call rate and these were not included in the Figure. To clarify which SNPs that are included in the graph, the SNPs have also been relabelled 1–16, which corresponds to the numbers given in Figure 1.

‘CTLA-4’ and ‘autoimmune Addison’s disease’ were used as search terms in Pubmed aiming to find all studies examining this gene in AAD patients. Studies comprising over 50 patients which investigated rs231775, and which showed no obvious overlap with the AAD cohorts described here, were included in the meta-analysis. The program ‘Comprehensive meta-analysis version 2.0’ (www.meta-analysis.com) using a random effects model was used in order to calculate ORs, z-scores and P-values. The impact of heterogeneity in the material was calculated by Hegdes’s G’ test and by Breslow–Day test of heterogeneity.