---
reference_id: "PMID:33545711"
title: SARS-CoV-2 evolution during treatment of chronic infection.
authors:
- Kemp SA
- Collier DA
- Datir RP
- Ferreira IATM
- Gayed S
- Jahun A
- Hosmillo M
- Rees-Spear C
- Mlcochova P
- Lumb IU
- Roberts DJ
- Chandra A
- Temperton N
- CITIID-NIHR BioResource COVID-19 Collaboration
- COVID-19 Genomics UK (COG-UK) Consortium
- Sharrocks K
- Blane E
- Modis Y
- Leigh KE
- Briggs JAG
- van Gils MJ
- Smith KGC
- Bradley JR
- Smith C
- Doffinger R
- Ceron-Gutierrez L
- Barcenas-Morales G
- Pollock DD
- Goldstein RA
- Smielewska A
- Skittrall JP
- Gouliouris T
- Goodfellow IG
- Gkrania-Klotsas E
- Illingworth CJR
- McCoy LE
- Gupta RK
journal: Nature
year: '2021'
doi: 10.1038/s41586-021-03291-y
content_type: full_text_xml
---

# SARS-CoV-2 evolution during treatment of chronic infection.
**Authors:** Kemp SA, Collier DA, Datir RP, Ferreira IATM, Gayed S, Jahun A, Hosmillo M, Rees-Spear C, Mlcochova P, Lumb IU, Roberts DJ, Chandra A, Temperton N, CITIID-NIHR BioResource COVID-19 Collaboration, COVID-19 Genomics UK (COG-UK) Consortium, Sharrocks K, Blane E, Modis Y, Leigh KE, Briggs JAG, van Gils MJ, Smith KGC, Bradley JR, Smith C, Doffinger R, Ceron-Gutierrez L, Barcenas-Morales G, Pollock DD, Goldstein RA, Smielewska A, Skittrall JP, Gouliouris T, Goodfellow IG, Gkrania-Klotsas E, Illingworth CJR, McCoy LE, Gupta RK
**Journal:** Nature (2021)
**DOI:** [10.1038/s41586-021-03291-y](https://doi.org/10.1038/s41586-021-03291-y)

## Content

1. Nature. 2021 Apr;592(7853):277-282. doi: 10.1038/s41586-021-03291-y. Epub 2021
 Feb 5.

SARS-CoV-2 evolution during treatment of chronic infection.

Kemp SA(#)(1), Collier DA(#)(1)(2)(3), Datir RP(#)(2)(3), Ferreira IATM(2)(3), 
Gayed S(4), Jahun A(5), Hosmillo M(5), Rees-Spear C(1), Mlcochova P(2)(3), Lumb 
IU(6), Roberts DJ(6), Chandra A(2)(3), Temperton N(7); CITIID-NIHR BioResource 
COVID-19 Collaboration; COVID-19 Genomics UK (COG-UK) Consortium; Sharrocks 
K(4), Blane E(3), Modis Y(2)(3)(8), Leigh KE(2)(3)(8), Briggs JAG(8), van Gils 
MJ(9), Smith KGC(2)(3), Bradley JR(3)(10), Smith C(11), Doffinger R(12), 
Ceron-Gutierrez L(12), Barcenas-Morales G(12)(13), Pollock DD(14), Goldstein 
RA(1), Smielewska A(5)(11), Skittrall JP(4)(15)(16), Gouliouris T(4), Goodfellow 
IG(5), Gkrania-Klotsas E(4), Illingworth CJR(15)(17), McCoy LE(1), Gupta 
RK(18)(19)(20).

Collaborators: Baker S, Dougan G, Hess C, Kingston N, Lehner PJ, Lyons PA, 
Matheson NJ, Owehand WH, Saunders C, Summers C, Thaventhiran JED, Toshner M, 
Weekes MP, Bucke A, Calder J, Canna L, Domingo J, Elmer A, Fuller S, Harris J, 
Hewitt S, Kennet J, Jose S, Kourampa J, Meadows A, O'Brien C, Price J, Publico 
C, Rastall R, Ribeiro C, Rowlands J, Ruffolo V, Tordesillas H, Bullman B, 
Dunmore BJ, Fawke S, Gräf S, Hodgson J, Huang C, Hunter K, Jones E, Legchenko E, 
Matara C, Martin J, Mescia F, O'Donnell C, Pointon L, Pond N, Shih J, Sutcliffe 
R, Tilly T, Treacy C, Tong Z, Wood J, Wylot M, Bergamaschi L, Betancourt A, 
Bower G, Cossetti C, De Sa A, Epping M, Fawke S, Gleadall N, Grenfell R, Hinch 
A, Huhn O, Jackson S, Jarvis I, Lewis D, Marsden J, Nice F, Okecha G, Omarjee O, 
Perera M, Richoz N, Romashova V, Yarkoni NS, Sharma R, Stefanucci L, Stephens J, 
Strezlecki M, Turner L, De Bie EMDD, Bunclark K, Josipovic M, Mackay M, Rossi S, 
Selvan M, Spencer S, Yong C, Ansaripour A, Michael A, Mwaura L, Patterson C, 
Polwarth G, Polgarova P, di Stefano G, Fahey C, Michel R, Bong SH, Coudert JD, 
Holmes E, Allison J, Butcher H, Caputo D, Clapham-Riley D, Dewhurst E, Furlong 
A, Graves B, Gray J, Ivers T, Kasanicki M, Le Gresley E, Linger R, Meloy S, 
Muldoon F, Ovington N, Papadia S, Phelan I, Stark H, Stirrups KE, Townsend P, 
Walker N, Webster J, Robson SC, Loman NJ, Connor TR, Golubchik T, Martinez Nunez 
RT, Ludden C, Corden S, Johnston I, Bonsall D, Smith CP, Awan AR, Bucca G, Estee 
Torok M, Saeed K, Prieto JA, Jackson DK, Hamilton WL, Snell LB, Moore C, 
Harrison EM, Goncalves S, Fairley DJ, Loose MW, Watkins J, Livett R, Moses S, 
Amato R, Nicholls S, Bull M, Smith DL, Barrett J, Aanensen DM, Curran MD, Parmar 
S, Aggarwal D, Shepherd JG, Parker MD, Glaysher S, Bashton M, Underwood AP, 
Pacchiarini N, Loveson KF, Carabelli AM, Templeton KE, Langford CF, Sillitoe J, 
de Silva TI, Wang D, Kwiatkowski D, Rambaut A, O'Grady J, Cottrell S, Holden 
MTG, Thomson EC, Osman H, Andersson M, Chauhan AJ, Hassan-Ibrahim MO, Lawniczak 
M, Alderton A, Chand M, Constantinidou C, Unnikrishnan M, Darby AC, Hiscox JA, 
Paterson S, Martincorena I, Robertson DL, Volz EM, Page AJ, Pybus OG, Bassett 
AR, Ariani CV, Spencer Chapman MH, Li KK, Shah RN, Jesudason NG, Taha Y, McHugh 
MP, Dewar R, Jahun AS, McMurray C, Pandey S, McKenna JP, Nelson A, Young GR, 
McCann CM, Elliott S, Lowe H, Temperton B, Roy S, Price A, Rey S, Wyles M, Rooke 
S, Shaaban S, de Cesare M, Letchford L, Silveira S, Pelosi E, Wilson-Davies E, 
Hosmillo M, O'Toole Á, Hesketh AR, Stark R, du Plessis L, Ruis C, Adams H, 
Bourgeois Y, Michell SL, Grammatopoulos D, Edgeworth J, Breuer J, Todd JA, 
Fraser C, Buck D, John M, Kay GL, Palmer S, Peacock SJ, Heyburn D, Weldon D, 
Robinson E, McNally A, Muir P, Vipond IB, Boyes J, Sivaprakasam V, Salluja T, 
Dervisevic S, Meader EJ, Park NR, Oliver K, Jeffries AR, Ott S, da Silva Filipe 
A, Simpson DA, Williams C, Masoli JAH, Knight BA, Jones CR, Koshy C, Ash A, 
Casey A, Bosworth A, Ratcliffe L, Xu-McCrae L, Pymont HM, Hutchings S, Berry L, 
Jones K, Halstead F, Davis T, Holmes C, Iturriza-Gomara M, Lucaci AO, Randell 
PA, Cox A, Madona P, Harris KA, Brown JR, Mahungu TW, Irish-Tavares D, Haque T, 
Hart J, Witele E, Fenton ML, Liggett S, Graham C, Swindells E, Collins J, 
Eltringham G, Campbell S, McClure PC, Clark G, Sloan TJ, Jones C, Lynch J, Warne 
B, Leonard S, Durham J, Williams T, Haldenby ST, Storey N, Alikhan NF, Holmes N, 
Moore C, Carlile M, Perry M, Craine N, Lyons RA, Beckett AH, Goudarzi S, Fearn 
C, Cook K, Dent H, Paul H, Davies R, Blane B, Girgis ST, Beale MA, Bellis KL, 
Dorman MJ, Drury E, Kane L, Kay S, McGuigan S, Nelson R, Prestwood L, Rajatileka 
S, Batra R, Williams RJ, Kristiansen M, Green A, Justice A, Mahanama AIK, 
Samaraweera B, Hadjirin NF, Quick J, Poplawski R, Kermack LM, Reynolds N, Hall 
G, Chaudhry Y, Pinckert ML, Georgana I, Moll RJ, Thornton A, Myers R, Stockton 
J, Williams CA, Yew WC, Trotter AJ, Trebes A, MacIntyre-Cockett G, Birchley A, 
Adams A, Plimmer A, Gatica-Wilcox B, McKerr C, Hilvers E, Jones H, Asad H, 
Coombes J, Evans JM, Fina L, Gilbert L, Graham L, Cronin M, Kumziene-Summerhayes 
S, Taylor S, Jones S, Groves DC, Zhang P, Gallis M, Louka SF, Starinskij I, 
Jackson C, Gourtovaia M, Tonkin-Hill G, Lewis K, Tovar-Corona JM, James K, 
Baxter L, Alam MT, Orton RJ, Hughes J, Vattipally S, Ragonnet-Cronin M, 
Nascimento FF, Jorgensen D, Boyd O, Geidelberg L, Zarebski AE, Raghwani J, 
Kraemer MUG, Southgate J, Lindsey BB, Freeman TM, Keatley JP, Singer JB, de 
Oliveira Martins L, Yeats CA, Abudahab K, Taylor BEW, Menegazzo M, Danesh J, 
Hogsden W, Eldirdiri S, Kenyon A, Mason J, Robinson TI, Holmes A, Price J, 
Hartley JA, Curran T, Mather AE, Shankar G, Jones R, Howe R, Morgan S, Wastenge 
E, Chapman MR, Mookerjee S, Stanley R, Smith W, Peto T, Eyre D, Crook D, Vernet 
G, Kitchen C, Gulliver H, Merrick I, Guest M, Munn R, Bradley DT, Wyatt T, 
Beaver C, Foulser L, Palmer S, Churcher CM, Brooks E, Smith KS, Galai K, McManus 
GM, Bolt F, Coll F, Meadows L, Attwood SW, Davies A, De Lacy E, Downing F, 
Edwards S, Scarlett GP, Jeremiah S, Smith N, Leek D, Sridhar S, Forrest S, 
Cormie C, Gill HK, Dias J, Higginson EE, Maes M, Young J, Wantoch M, Jamrozy D, 
Lo S, Patel M, Hill V, Bewshea CM, Ellard S, Auckland C, Harrison I, Bishop C, 
Chalker V, Richter A, Beggs A, Best A, Percival B, Mirza J, Megram O, Mayhew M, 
Crawford L, Ashcroft F, Moles-Garcia E, Cumley N, Hopes R, Asamaphan P, Niebel 
MO, Gunson RN, Bradley A, Maclean A, Mollett G, Blacow R, Bird P, Helmer T, 
Fallon K, Tang J, Hale AD, Macfarlane-Smith LR, Harper KL, Carden H, Machin NW, 
Jackson KA, Ahmad SSY, George RP, Turtle L, O'Toole E, Watts J, Breen C, Cowell 
A, Alcolea-Medina A, Charalampous T, Patel A, Levett LJ, Heaney J, Rowan A, 
Taylor GP, Shah D, Atkinson L, Lee JCD, Westhorpe AP, Jannoo R, Lowe HL, 
Karamani A, Ensell L, Chatterton W, Pusok M, Dadrah A, Symmonds A, Sluga G, 
Molnar Z, Baker P, Bonner S, Essex S, Barton E, Padgett D, Scott G, Greenaway J, 
Payne BAI, Burton-Fanning S, Waugh S, Raviprakash V, Sheriff N, Blakey V, 
Williams LA, Moore J, Stonehouse S, Smith L, Davidson RK, Bedford L, Coupland L, 
Wright V, Chappell JG, Tsoleridis T, Ball J, Khakh M, Fleming VM, Lister MM, 
Howson-Wells HC, Berry L, Boswell T, Joseph A, Willingham I, Duckworth N, Walsh 
S, Wise E, Moore N, Mori M, Cortes N, Kidd S, Williams R, Gifford L, Bicknell K, 
Wyllie S, Lloyd A, Impey R, Malone CS, Cogger BJ, Levene N, Monaghan L, Keeley 
AJ, Partridge DG, Raza M, Evans C, Johnson K, Abnizova I, Aigrain L, Alderton A, 
Ali M, Allen L, Amato R, Anderson R, Ariani C, Austin-Guest S, Bala S, Barrett 
J, Bassett A, Battleday K, Beal J, Beale M, Beaver C, Bellany S, Bellerby T, 
Bellis K, Berger D, Berriman M, Betteridge E, Bevan P, Binley S, Bishop J, 
Blackburn K, Bonfield J, Boughton N, Bowker S, Brendler-Spaeth T, Bronner I, 
Brooklyn T, Buddenborg SK, Bush R, Caetano C, Cagan A, Carter N, Cartwright J, 
Monteiro TC, Chapman L, Chillingworth TJ, Clapham P, Clark R, Clarke A, Clarke 
C, Cole D, Cook E, Coppola M, Cornell L, Cornwell C, Corton C, Crackett A, 
Cranage A, Craven H, Craw S, Crawford M, Cutts T, Dabrowska M, Davies M, Davies 
R, Dawson J, Day C, Densem A, Dibling T, Dockree C, Dodd D, Dogga S, Dorman M, 
Dougan G, Dougherty M, Dove A, Drummond L, Drury E, Dudek M, Durham J, Durrant 
L, Easthope E, Eckert S, Ellis P, Farr B, Fenton M, Ferrero M, Flack N, Fordham 
H, Forsythe G, Foulser L, Francis M, Fraser A, Freeman A, Galvin A, 
Garcia-Casado M, Gedny A, Girgis S, Glover J, Goncalves S, Goodwin S, Gould O, 
Gourtovaia M, Gray A, Gray E, Griffiths C, Gu Y, Guerin F, Hamilton W, Hanks H, 
Harrison E, Harrott A, Harry E, Harvison J, Heath P, Hernandez-Koutoucheva A, 
Hobbs R, Holland D, Holmes S, Hornett G, Hough N, Huckle L, Hughes-Hallet L, 
Hunter A, Inglis S, Iqbal S, Jackson A, Jackson D, James K, Jamrozy D, Verdejo 
CJ, Johnston I, Jones M, Kallepally K, Kane L, Kay K, Kay S, Keatley J, Keith A, 
King A, Kitchin L, Kleanthous M, Klimekova M, Korlevic P, Krasheninnkova K, 
Kwiatkowski D, Lane G, Langford C, Laverack A, Law K, Lawniczak M, Lensing S, 
Leonard S, Letchford L, Lewis K, Lewis-Wade A, Liddle J, Lin Q, Lindsay S, 
Linsdell S, Livett R, Lo S, Long R, Lovell J, Lovell J, Ludden C, Mack J, 
Maddison M, Makunin A, Mamun I, Mansfield J, Marriott N, Martin M, Martincorena 
I, Mayho M, McCarthy S, McClintock J, McGuigan S, McHugh S, McMinn L, Meadows C, 
Mobley E, Moll R, Morra M, Morrow L, Murie K, Nash S, Nathwani C, Naydenova P, 
Neaverson A, Nelson R, Nerou E, Nicholson J, Nimz T, Noell GG, O'Meara S, Ohan 
V, Oliver K, Olney C, Ormond D, Oszlanczi A, Palmer S, Pang YF, Pardubska B, 
Park N, Parmar A, Patel G, Patel M, Payne M, Peacock S, Petersen A, Plowman D, 
Preston T, Prestwood L, Puethe C, Quail M, Rajan D, Rajatileka S, Rance R, 
Rawlings S, Redshaw N, Reynolds J, Reynolds M, Rice S, Richardson M, Roberts C, 
Robinson K, Robinson M, Robinson D, Rogers H, Rojo EM, Roopra D, Rose M, Rudd L, 
Sadri R, Salmon N, Saul D, Schwach F, Scott C, Seekings P, Shirley L, Sillitoe 
J, Simms A, Sinnott M, Sivadasan S, Siwek B, Sizer D, Skeldon K, Skelton J, 
Slater-Tunstill J, Sloper L, Smerdon N, Smith C, Smith C, Smith J, Smith K, 
Smith M, Smith S, Smith T, Sneade L, Soria CD, Sousa C, Souster E, Sparkes A, 
Spencer-Chapman M, Squares J, Stanley R, Steed C, Stickland T, Still I, Stratton 
M, Strickland M, Swann A, Swiatkowska A, Sycamore N, Swift E, Symons E, Szluha 
S, Taluy E, Tao N, Taylor K, Taylor S, Thompson S, Thompson M, Thomson M, 
Thomson N, Thurston S, Tonkin-Hill G, Toombs D, Topping B, Tovar-Corona J, 
Ungureanu D, Uphill J, Urbanova J, Jansen Van P, Vancollie V, Voak P, Walker D, 
Walker M, Waller M, Ward G, Weatherhogg C, Webb N, Weldon D, Wells A, Wells E, 
Westwood L, Whipp T, Whiteley T, Whitton G, Whitwham A, Widaa S, Williams M, 
Wilson M, Wright S, Betteridge E, Farr BW, Goodwin S, Quail MA, Scott C, Shirley 
L, Thurston SAJ, Rajan D, Bronner IF, Aigrain L, Redshaw NM, Lensing SV, 
McCarthy S, Makunin A, Balcazar CE, Gallagher MD, Williamson KA, Stanton TD, 
Michelsen ML, Warwick-Dugdale J, Manley R, Farbos A, Harrison JW, Sambles CM, 
Studholme DJ, Lackenby A, Mbisa T, Platt S, Miah S, Bibby D, Manso C, Hubb J, 
Dabrera G, Ramsay M, Bradshaw D, Schaefer U, Groves N, Gallagher E, Lee D, 
Williams D, Ellaby N, Hartman H, Manesis N, Patel V, Ledesma J, Twohig KA, 
Allara E, Pearson C, Cheng JKJ, Bridgewater HE, Frost LR, Taylor-Joyce G, Brown 
PE, Tong L, Broos A, Mair D, Nichols J, Carmichael SN, Smollett KL, Nomikou K, 
Aranday-Cortes E, Johnson N, Nickbakhsh S, Vamos EE, Hughes M, Rainbow L, Eccles 
R, Nelson C, Whitehead M, Gregory R, Gemmell M, Wierzbicki C, Webster HJ, Fisher 
CL, Signell AW, Betancor G, Wilson HD, Nebbia G, Flaviani F, Cerda AC, Merrill 
TV, Wilson RE, Cotic M, Bayzid N, Thompson T, Acheson E, Rushton S, O'Brien S, 
Baker DJ, Rudder S, Aydin A, Sang F, Debebe J, Francois S, Vasylyeva TI, Zamudio 
ME, Gutierrez B, Marchbank A, Maksimovic J, Spellman K, McCluggage K, Morgan M, 
Beer R, Afifi S, Workman T, Fuller W, Bresner C, Angyal A, Green LR, Parsons PJ, 
Tucker RM, Brown R, Whiteley M, Bonfield J, Puethe C, Whitwham A, Liddle J, Rowe 
W, Siveroni I, Le-Viet T, Gaskin A, Johnson R.

Author information:
(1)Division of Infection and Immunity, University College London, London, UK.
(2)Cambridge Institute of Therapeutic Immunology & Infectious Disease (CITIID), 
Cambridge, UK.
(3)Department of Medicine, University of Cambridge, Cambridge, UK.
(4)Department of Infectious Diseases, Cambridge University NHS Hospitals 
Foundation Trust, Cambridge, UK.
(5)Department of Pathology, University of Cambridge, Cambridge, UK.
(6)NHS Blood and Transplant, Oxford and BRC Haematology Theme, University of 
Oxford, Oxford, UK.
(7)Viral Pseudotype Unit, Medway School of Pharmacy, University of Kent, 
Canterbury, UK.
(8)Medical Research Council Laboratory of Molecular Biology, Cambridge, UK.
(9)Department of Medical Microbiology, Academic Medical Center, University of 
Amsterdam, Amsterdam, The Netherlands.
(10)NIHR Cambridge Bioresource, Cambridge, UK.
(11)Department of Virology, Cambridge University NHS Hospitals Foundation Trust, 
Cambridge, UK.
(12)Department of Clinical Biochemistry and Immunology, Addenbrooke's Hospital, 
Cambridge, UK.
(13)FES-Cuautitlán, UNAM, Cuautitlán Izcalli, Mexico.
(14)Biochemistry and Molecular Genetics, University of Colorado School of 
Medicine, Aurora, CO, USA.
(15)Department of Applied Mathematics and Theoretical Physics, University of 
Cambridge, Cambridge, UK.
(16)Clinical Microbiology and Public Health Laboratory, Addenbrooke's Hospital, 
Cambridge, UK.
(17)MRC Biostatistics Unit, University of Cambridge, Cambridge, UK.
(18)Cambridge Institute of Therapeutic Immunology & Infectious Disease (CITIID), 
Cambridge, UK. rkg20@cam.ac.uk.
(19)Department of Medicine, University of Cambridge, Cambridge, UK. 
rkg20@cam.ac.uk.
(20)Africa Health Research Institute, Durban, South Africa. rkg20@cam.ac.uk.
(#)Contributed equally

Erratum in
    Nature. 2022 Aug;608(7922):E23. doi: 10.1038/s41586-022-05104-2.

Update of
    medRxiv. 2020 Dec 29:2020.12.05.20241927. doi: 10.1101/2020.12.05.20241927.

The spike protein of severe acute respiratory syndrome coronavirus 2 
(SARS-CoV-2) is critical for virus infection through the engagement of the human 
ACE2 protein1 and is a major antibody target. Here we show that chronic 
infection with SARS-CoV-2 leads to viral evolution and reduced sensitivity to 
neutralizing antibodies in an immunosuppressed individual treated with 
convalescent plasma, by generating whole-genome ultra-deep sequences for 23 time 
points that span 101 days and using in vitro techniques to characterize the 
mutations revealed by sequencing. There was little change in the overall 
structure of the viral population after two courses of remdesivir during the 
first 57 days. However, after convalescent plasma therapy, we observed large, 
dynamic shifts in the viral population, with the emergence of a dominant viral 
strain that contained a substitution (D796H) in the S2 subunit and a deletion 
(ΔH69/ΔV70) in the S1 N-terminal domain of the spike protein. As passively 
transferred serum antibodies diminished, viruses with the escape genotype were 
reduced in frequency, before returning during a final, unsuccessful course of 
convalescent plasma treatment. In vitro, the spike double mutant bearing both 
ΔH69/ΔV70 and D796H conferred modestly decreased sensitivity to convalescent 
plasma, while maintaining infectivity levels that were similar to the wild-type 
virus.The spike substitution mutant D796H appeared to be the main contributor to 
the decreased susceptibility to neutralizing antibodies, but this mutation 
resulted in an infectivity defect. The spike deletion mutant ΔH69/ΔV70 had a 
twofold higher level of infectivity than wild-type SARS-CoV-2, possibly 
compensating for the reduced infectivity of the D796H mutation. These data 
reveal strong selection on SARS-CoV-2 during convalescent plasma therapy, which 
is associated with the emergence of viral variants that show evidence of reduced 
susceptibility to neutralizing antibodies in immunosuppressed individuals.

DOI: 10.1038/s41586-021-03291-y
PMCID: PMC7610568
PMID: 33545711 [Indexed for MEDLINE]

Conflict of interest statement: Competing interests: the authors declare no 
competing interests

A septuagenarian male was admitted to a tertiary hospital in summer of 2020 and had tested positive for SARS-CoV-2 RT-PCR 35 days previously on a nasopharyngeal swab (Day 1) at a local hospital (Extended data 1 and 2). His past medical history was significant for marginal B cell lymphoma diagnosed in 2012, with previous chemotherapy including vincristine, prednisolone, cyclophosphamide and anti-CD20 B cell depletion with rituximab. It is likely that both chemotherapy and underlying lymphoma contributed to B and T cell combined immunodeficiency (Extended data 2 and 3, Supplementary Table 1). Computed tomography (CT) of the chest showed widespread abnormalities consistent with COVID-19 pneumonia (Supplementary Figure 1). Treatment included two 10-day courses of remdesivir with a five day gap in between (Extended data 1). Two units of convalescent plasma were administered on days 63 and 65 (Extended data 3). Following clinical deterioration, remdesivir and a unit of convalescent plasma were administered on day 95, but the individual unfortunately died on day 102 (Supplementary text).

The majority of samples were respiratory samples from nose and throat or endotracheal aspirates during the period of intubation (Supplementary Table 3). Ct values ranged from 16-34 and all 23 respiratory samples were successfully sequenced by standard single molecule sequencing approach as per the ARTIC protocol implemented by COG-UK; of these 20 additionally underwent short-read deep sequencing using the Illumina platform (Supplementary table 4). There was general agreement between the two methods (Extended data 4). However due to the higher reliability of Illumina for low frequency variants, this was used for formal analysis2,3. Additionally, single genome amplification and sequencing of Spike using extracted RNA from respiratory samples was used as an independent method to detect mutations observed (Extended data 4). Finally, we detected no evidence of recombination, based on two independent methods.

Maximum likelihood analysis of patient-derived whole genome consensus sequences demonstrated clustering with other local sequences from the same region (Figure 1). The infecting strain was assigned to lineage 20B bearing the D614G Spike variant. Environmental sampling showed evidence of virus on surfaces such as telephone and call bell. Sequencing of these surface viruses showed clustering with those derived from the respiratory tract (Extended data 2). All samples were consistent with having arisen from a single underlying viral population. In our phylogenetic analysis, we included sequential sequences from three other local patients identified with persistent viral RNA shedding over a period of 4 weeks or more as well as two long term immunosuppressed SARS-CoV-2 ‘shedders’ recently reported4,5, (Extended data 2, Supplementary Table 2). While the sequences from the three local patients as well as from Avanzato et al5 showed little divergence with no amino acid changes in Spike over time, the case patient showed significant diversification. The Choi et al report4 showed similar degree of diversification as the case patient. Further investigation of the sequence data suggested the existence of an underlying structure to the viral population in our patient, with samples collected at days 93 and 95 being rooted within, but significantly divergent from the original population (Extended data 5 and 6). The relationship of the divergent samples to those at earlier time points argues against superinfection.

All samples tested positive by RT-PCR and there was no sustained change in Ct values throughout the 101 days following the first two courses of remdesivir (days 41 and 54), or the first two units of convalescent plasma with polyclonal antibodies (days 63 and 65, Extended data 3). Of note we were not able to culture virus from stored swab samples. Consensus sequences from short read deep sequence Illumina data revealed dynamic population changes after day 65, as shown by a highlighter plot (Extended data 6). In addition, we were also able to follow the dynamics of virus populations down to low frequencies during the entire period (Figure 2, Supplementary Table 4). Following remdesivir at day 41 the low frequency variant analysis allowed us to observe transient amino acid changes in populations at below 50% abundance in Orf 1b, 3a and Spike, with a T39I (C27509T) mutation in ORF7a reaching 79% on day 45 (Figure 2, pink, supplementary information). At day 66 we noted I513T in NSP2 (T2343C) and V157L (G13936T) in RdRp had emerged from undetectable at day 54 to almost 100% frequency (Figure 2, red and green dashed lines), with the polymerase being the more plausible candidate for driving this sweep. Notably, spike variant N501Y, which can increase the ACE2 receptor affinity6, and which is present in the new UK B1.1.7 lineage7, was observed on day 55 at 33% frequency, but was eliminated by the sweep of the NSP2/RdRp variant.

In contrast to the early period of infection, between days 66 and 82, following the first two administrations of convalescent sera, a shift in the virus population was observed, with a variant bearing D796H in S2 and ΔH69/ΔV70 in the S1 N-terminal domain (NTD) becoming the dominant population at day 82. This was identified in a nose and throat swab sample with high viral load as indicated by Ct of 23 (Figure 3A). The deletion was detected transiently at baseline according to short read deep sequencing. ΔH69/ΔV70 was due to an out of frame six nucleotide deletion resulting in the sequence of codon 68 changing from ATA to ATC.

On Days 86 and 89, viruses obtained from upper respiratory tract samples were characterised by the Spike mutations Y200H and T240I, with the deletion/mutation pair observed on day 82 having fallen to frequencies of 10% or less (Figure 2 and 3). The Spike mutations Y200H and T240I were accompanied at high frequency by two other non-synonymous variants with similar allele frequencies, coding for I513T in NSP2, V157L in RdRp and N177S in NSP15 (Figure 2A). Both of these were also previously observed at >98% frequency in the sample on day 66 (Figure 2A, red and green lines), arguing that this new lineage emerged out of a previously existing population.

Sequencing of a nose and throat swab sample at day 93 identified viruses characterised by Spike mutations P330S at the edge of the RBD and W64G in S1 NTD at close to 100% abundance, with D796H along with ΔH69/ΔV70 at <1% abundance and the variants Y200H and T240I at frequencies of <2%. Viruses with the P330S variant were detected in two independent samples from different sampling sites, arguing against the possibility of contamination. The divergence of these samples from the remainder of the population (Figure 2, 3B and Extended data 5 and 6) suggests the possibility that they represent a compartmentalised subpopulation.

Patterns in the variant frequencies suggest competition between virus populations carrying different mutations, viruses with the D796H/ΔH69/ΔV70 deletion/mutation pair rising to high frequency during CP therapy, then being outcompeted by another population in the absence of therapy. Specifically, these data are consistent with a lineage of viruses with the NSP2 I513T and RdRp V157L variant, dominant on day 66, being outcompeted during therapy by the mutation/deletion variant. With the lapse in therapy, the original strain, having acquired NSP15 N1773S and the Spike mutations Y200H and T240I, regained dominance, followed by the emergence of a separate population with the W64G and P330S mutations.

In a final attempt to reduce the viral load, a third course of remdesivir (day 93) and third dose of CP (day 95) were administered. We observed a re-emergence of the D796H + ΔH69/ΔV70 viral population (Figure 2, 3). The inferred linkage of D796H and ΔH69/ΔV70 was maintained as evidenced by the highly similar frequencies of the two variants, suggesting that the third unit of CP led to the re-emergence of this population under renewed positive selection. In further support of our proposed idea of competition, noted above, frequencies of these two variants appeared to mirror changes in the NSP2 I513T mutation (Figure 2), suggesting these as markers of opposing clades in the viral population. Ct values remained low throughout this period with hyperinflammation, eventually leading to multi-organ failure and death at day 102. The repeated increase in frequency of the viral population with CP therapy strongly supports the hypothesis that the deletion/mutation combination conferred selective advantage.

Using lentiviral pseudotyping we generated wild type, ΔH69/ΔV70 + D796H and single mutant Spike proteins in enveloped virions in order to measure neutralisation activity of CP against these viruses (Figure 4). This system has been shown to give generally similar results to replication competent virus8,9. Spike protein from each mutant was detected in pelleted virions (Figure 4A). We also probed with an HIV-1 p24 antibody to monitor levels of lentiviral particle production (Figure 4A, Supplementary Figure 2). We then measured infectivity of the pseudoviruses, correcting for virus input using reverse transcriptase activity measurement, and found that ΔH69/ΔV70 appeared to have two-fold higher infectivity over a single round of infection compared to wild type (Figure 4B, Extended data 7). By contrast, the D796H single mutant had significantly lower infectivity as compared to wild type and double mutant had similar infectivity to wild type (Figure 4B, Extended data 7).

We found that D796H alone and the D796H + ΔH69/ΔV70 double mutant were less sensitive to neutralisation by convalescent plasma samples (Figure 4C-E, Extended data 7). By contrast the ΔH69/ΔV70 single mutant did not reduce neutralisation sensitivity. In addition, patient derived serum from days 64 and 66 (one day either side of CP2 infusion) similarly showed lower potency against the D796H + ΔH69/ΔV70 mutants (Figure 4F, G).

A panel of nineteen monoclonal antibodies (mAbs) isolated from three donors was previously identified to neutralize SARS-CoV-2. To establish if the mutations incurring in vivo (D796H and ΔH69/ΔV70) resulted in a global change in neutralization sensitivity we tested neutralising mAbs targeting the seven major epitope clusters previously described (excluding non-neutralising clusters II, V and small [n =<2] neutralising clusters IV, X). The eight RBD-specific mAbs (Extended data 8) exhibited no major change in neutralisation potency and non-RBD specific COVA1-21 showing 3-5 fold reduction in potency against ΔH69/ΔV70+D796H and ΔH69/ΔV70, but not D796H alone9 (Extended data 8). We observed no differences in neutralisation between single/double mutants and wild type, suggesting that the mechanism of escape was likely outside these epitopes in the RBD. These data confirm the specificity of the findings from convalescent plasma and suggest that mutations observed are related to antibodies targeting regions outside the RBD. Interestingly, ΔH69/ΔV70 containing viruses showed reduced neutralisation sensitivity to the mAb COVA1-21, targeting an as yet undefined epitope outside the RBD. 10.

To understand how the ΔH69/ΔV70 and D796H might confer antibody resistance, we assessed how they might affect the Spike structure (Extended data 9). We based this analysis primarily on a structure lacking stabilising modifications (PDB 6xr8)11, but also referred to stabilised structures determined at different pH values12. ΔH69/ΔV70 is located in a disordered, glycosylated loop at the distal surface of the NTD, near the binding site of polyclonal antibodies derived from COV57 plasma13,14 (Extended data 9). As this loop is flexible and highly accessible, ΔH69/V70 could in principle affect antibody binding in this region. D796 is located near the base of Spike, in a surface loop that is structurally somewhat disordered in the prefusion conformation and becomes part of a large disordered region in the post fusion S2 trimer11 (Extended data 9). The loop containing residue 796 is proposed to be targeted by antibodies15, despite mutations at position 796 being relatively uncommon (Extended data 9). In the RBD-down Spike structures11,12, D796 forms contacts with residues in the neighbouring protomer, including the glycosylated residue N709 (Extended data 9).

Here we have documented a repeated evolutionary response by SARS-CoV-2 in the presence of antibody therapy during the course of a persistent infection in an immunocompromised host. The observation of potential selection for specific variants coinciding with the presence of antibodies from convalescent plasma is supported by the experimental finding of two-fold reduced susceptibility of these viruses to convalescent plasma containing polyclonal antibodies. In this case the emergence of the variant was not the primary reason for treatment failure. We have noted in our analysis signs of compartmentalised viral replication based on the sequences recovered in upper respiratory tract samples. Both population genetic and small animal studies have shown a lack of reassortment between influenza viruses within a single host during an infection, suggesting that acute respiratory viral infection may be characterised by spatially distinct viral populations16,17. In the analysis of data, it is important to distinguish genetic changes which occur in the primary viral population from apparent changes that arise from the stochastic observation of spatially distinct subpopulations in the host. While the samples we observe on days 93 and 95 of infection are genetically distinct from the others, the remaining samples are consistent with arising from a consistent viral population. We note that Choi et al reported the detection in post-mortem tissue of viral RNA not only in lung tissue, but also in the spleen, liver, and heart4. Mixing of virus from different compartments, for example via blood, or movement of secretions from lower to upper respiratory tract, could lead to fluctuations in viral populations at particular sampling sites.

This is a single case report and therefore limited conclusions can be drawn about generalisability.

An important limitation is that the data were derived from sampling from the upper respiratory tract and not the lower tract, thus limiting the inferences that can be drawn regarding viral populations in this single case.

In addition to documenting the emergence of SARS-CoV-2 Spike ΔH69/ΔV70 in vivo, we show that this mutation modestly increases infectivity of the Spike protein in a pseudotyping assay. The deletion was observed contemporaneously with the rare S2 mutation D796H after two separate courses of CP, with other viral populations emerging. D796H, but not ΔH69/ΔV70, conferred reduction in susceptibility to polyclonal antibodies in the units of CP administered, though we cannot speculate as to their individual impacts on sera from other individuals. It is intriguing that the ΔH69/ΔV70 + D796H double mutant diminished in between CP courses, suggesting that there were other selective forces at play in the intervening period, possibly driven by the inflammation observed in the individual. This includes the possibility that the haplotype with ΔH69/ΔV70 + D796H may have carried mutations in other regions deleterious during that intervening period. Although ΔH69/V70 is expanding at a high rate18, D796 mutations are also increasing. D796H has been documented in 0.02% of global sequences and D796Y appears in 0.05% of global sequences (Extended data 9).

The effects of CP on virus evolution seen here are unlikely to apply in immune competent hosts where viral diversity is likely to be lower due to better immune control. Our data highlight that infection control measures may need to be tailored to the needs of immunocompromised patients and also caution in interpretation of CDC guidelines that recommend 20 days as the upper limit of infection prevention precautions in immune compromised patients who are afebrile19. Due to the difficulty with culturing clinical isolates, use of surrogates are warranted20. However, where detection of ongoing viral evolution is possible, this serves as a clear proxy for the existence of infectious virus. In our case we detected environmental contamination whilst in a single occupancy room and the patient was moved to a negative-pressure high air-change infectious disease isolation room.

Clinical efficacy of convalescent plasma in severe COVID-19 has not been demonstrated21, and its use in different stages of infection and disease remains experimental; as such, we suggest that it should be reserved for use within clinical trials, with rigorous monitoring of clinical and virological parameters. The data from this single case report might warrant caution in use of convalescent plasma in patients with immune suppression of both T cell and B cell arms; in such cases, the antibodies administered have little support from cytotoxic T cells, thereby reducing chances of clearance and theoretically raising the potential for escape mutations. Whilst we await further data, where clinical trial enrolment is not possible, convalescent plasma administered for clinical need in immune suppression should ideally only be considered as part of observational studies, undertaken preferably in single occupancy rooms with enhanced infection control precautions, including SARS-CoV-2 environmental sampling and real-time sequencing. Understanding of viral dynamics and characterisation of viral evolution in response to different selection pressures in the immunocompromised host is necessary not only for improved patient management but also for public health benefit.

Serial samples were collected from the patient periodically from the lower respiratory tract (sputum or endotracheal aspirate), upper respiratory tract (throat and nasal swab), and from stool. Nucleic acid extraction was done from 500μl of sample with a dilution of MS2 bacteriophage to act as an internal control, using the easyMAG platform (Biomerieux, Marcy-l'Étoile) according to the manufacturers’ instructions. All samples were tested for presence of SARS-CoV-2 with a validated one-step RT q-PCR assay developed in conjunction with the Public Health England Clinical Microbiology22. Amplification reaction were all performed on a Rotorgene™ PCR instrument. Samples which generated a CT of ≤36 were considered to be positive.

Sera from recovered patients in the COVIDx study23 were used for testing of neutralisation activity by SARS-CoV-2 mutants.

Recombinant SARS-CoV-2 N, S and RBD were covalently coupled to distinct carboxylated bead sets (Luminex; Netherlands) to form a 3-plex and analyzed as previously described (Xiong et al. 2020). Specific binding was reported as mean fluorescence intensities (MFI).

Whole blood was diluted 1:5 in RPMI into 96-well F plates (Corning) and activated by single stimulation with phytohemagglutinin (PHA; 10 μg/ml; Sigma-Aldrich), or LPS (1 μg/ml, List Biochemicals) or by co-stimulating with anti-CD3 (MEM57, Abcam, 200 ng/ml, 1:1000) and IL-2 (Immunotools, 1430U/ml, 1:1000). Supernatants were taken after 24 hours. Levels (pg/ml) are shown for IFNg, IL17, IL2, TNFa, IL6, IL1b and IL10. Cytokines were measured by multiplexed particle based Flow cytometry on a Luminex analyzer (Bio-Plex, Bio-Rad, UK) using an R&D Systems custom kit (R&D Systems, UK).

For viral genomic sequencing, total RNA was extracted from samples as described. Samples were sequenced using MinION flow cells version 9.4.1 (Oxford Nanopore Technologies) following the ARTICnetwork V3 protocol (https://dx.doi.org/10.17504/protocols.io.bbmuik6w) and BAM files assembled using the ARTICnetwork assembly pipeline (https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html). A representative set of 10 sequences were selected and also sequenced using the Illumina MiSeq platform. Amplicons were diluted to 2 ng/μl and 25 μl (50 ng) were used as input for each library preparation reaction. The library preparation used KAPA Hyper Prep kit (Roche) according to manufacturer’s instructions. Briefly, amplicons were end-repaired and had A-overhang added; these were then ligated with 15mM of NEXTflex DNA Barcodes (Bio Scientific, Texas, USA). Post-ligation products were cleaned using AMPure beads and eluted in 25 μl. Then, 20 μl were used for library amplification by 5 cycles of PCR. For the negative controls, 1ng was used for ligation-based library preparation. All libraries were assayed using TapeStation (Agilent Technologies, California, USA) to assess fragment size and quantified by QPCR. All libraries were then pooled in equimolar accordingly. Libraries were loaded at 15nM and spiked in 5% PhiX (Illumina, California, USA) and sequenced on one MiSeq 500 cycle using a Miseq Nano v2 with 2x 250 paired-end sequencing. A minimum of ten reads were required for a variant call.

For long-read sequencing, genomes were assembled with reference-based assembly and a curated bioinformatics pipeline with 20x minimum coverage across the whole-genome24. For short-read sequencing, FASTQs were downloaded, poor-quality reads were identified and removed, and both Illumina and PHiX adapters were removed using TrimGalore v0.6.625. Trimmed paired-end reads were mapped to the National Center for Biotechnology Information SARS-CoV-2 reference sequence MN908947.3 using MiniMap2-2.17 with arguments -ax and sr26. BAM files were then sorted and indexed with samtools v1.11 and PCR optical duplicates removed using Picard (http://broadinstitute.github.io/picard). A consensus sequences of nucleic acids with a minimum whole-genome coverage of at least 20× were generated with BCFtools using a 0% majority threshold.

Variant frequencies were validated using custom code as part of the AnCovMulti package (github.com/PollockLaboratory/AnCovMulti). The main idea behind this validation was to identify and remove consistent potential amplification errors and mutability near the end of Illumina reads. Furthermore, stringent filtering was applied to remove biased amplification of early laboratory-induced mutations or very low copy variations.

Filtering consisted of requiring exact initiation at a primer within two bp of the start of a read, a minimum of 247 bp length read, fewer than four well-separated sites divergent from the reference sequence, a maximum insertion size of three nucleotides, a maximum deletion size of 11 bp, and resolution of conflicting signal from different primers.

Viral RNA extracts were reverse transcribed from each sample to sufficiently capture the diversity of the viral population without introducing resampling bias. SuperScript IV (Thermofisher Scientific) and the gene specific primers were used for reverse transcription. Template RNA was degraded with RNAse H (Thermofisher Scientific). All primers used were ‘in-house’ primers designed using the multiple sequence alignment of the patient’s consensus NGS sequences. Partial Spike (amino acids 21-800) was amplified as 1 continuous length of DNA (Spike ~ 1.8 kb) by nested PCR. Terminally diluted cDNA was PCR-amplified using Platinum® Taq DNA Polymerase High Fidelity (Invitrogen, Carlsbad, CA) so that 30% of reactions were positive27. By Poisson statistics, sequences were deemed ≥80% likely to be derived from HIV-1 single genomes. We obtained between 20–60 single genomes at each sample time point to achieve 90% confidence of detecting variants present at ≥8% of the viral population in vivo28,29. Partial spike amplicons obtained from terminal dilution PCR amplification were Sanger sequenced to form a contiguous sequence using another set of 8 in-house primers. Sanger sequencing was provided by Genewiz UK and manual sequence editing was performed using DNA Dynamo software (Blue Tractor Software Ltd, UK).

All available full-genome SARS-CoV-2 sequences were downloaded from the GISAID database (http://gisaid.org/)30 on 16th December. Duplicate and low-quality sequences (>5% N regions) were removed, leaving a dataset of 212,297 sequences with a length of >29,000bp. All sequences were sorted by name and only sequences sequenced with United Kingdom / England identifiers were retained. From this dataset, sequences were de-duplicated and where background sequences were required in figures, randomly subsampled using seqtk (https://github.com/lh3/seqtk). All sequences were aligned to the SARS-CoV-2 reference strain MN908947.3, using MAFFT v7.475 with automatic flavour selection31. Major SARS-CoV-2 clade memberships were assigned to all sequences using both the Nextclade server v0.9 (https://clades.nextstrain.org/) and Phylogenetic Assignment Of Named Global Outbreak Lineages (pangolin)32.

Maximum likelihood phylogenetic trees were produced using the above curated dataset using IQ-TREE v2.1.233. Evolutionary model selection for trees were inferred using ModelFinder34 and trees were estimated using the GTR+F+I model with 1000 ultrafast bootstrap replicates35. All trees were visualised with Figtree v.1.4.4 (http://tree.bio.ed.ac.uk/software/figtree/), rooted on the SARS-CoV-2 reference sequence and nodes arranged in descending order. Nodes with bootstraps values of <50 were collapsed using an in-house script.

The SAMFIRE package version 1.06 36 was used to call allele frequency trajectories from BAM file data. Reads were included in this analysis if they had a median PHRED score of at least 30, trimming the ends of reads to achieve this if necessary. Nucleotides were then filtered to have a PHRED score of at least 30; reads with fewer than 30 such reads were discarded. Distances between sequences, accounting for low-frequency variant information, was also conducted using SAMFIRE. The sequence distance metric, described in an earlier paper37, combines allele frequencies across the whole genome. Where L is the length of the genome, we define q(t) as a 4 x L element vector describing the frequencies of each of the nucleotides A, C, G, and T at each locus in the viral genome sampled at time t. For any given locus i in the genome we calculate the change in allele frequencies between the times t1 and t2 via a generalisation of the Hamming distance d(qi(t1),qi(t2))=12∑a∈{A,C,G,T}|qia(t1)−qia(t2)| where the vertical lines indicate the absolute value of the difference. These statistics were then combined across the genome to generate the pairwise sequence distance metric D(q(t1),q(t2))=∑id(qi(t1),qi(t2))


The Mathematica software package was to conduct a regression analysis of pairwise sequence distances against time, leading to an estimate of a mean rate of within-host sequence evolution. In contrast to the phylogenetic analysis, this approach assumed the samples collected on days 93 and 95 to arise via stochastic emission from a spatially separated subpopulation within the host, leading to a lower inferred rate of viral evolution for the bulk of the viral population.

All variants were indecently validated using custom code as part of the AnCovMulti package, found at https://github.com/PollockLaboratory/AnCovMulti.

Forty-eight hours after transfection of cells with plasmid preparations, the culture supernatant was harvested and passed through a 0.45-μm-pore-size filter to remove cellular debris. The filtrate was centrifuged at 15,000 rpm for 120 min to pellet virions. The pelleted virions were lysed in Laemmli reducing buffer (1 M Tris-HCl [pH 6.8], SDS, 100% glycerol, β-mercaptoethanol, and bromophenol blue). Pelleted virions were subjected to electrophoresis on SDS–4 to 12% bis-Tris protein gels (Thermo Fisher Scientific) under reducing conditions. This was followed by electroblotting onto polyvinylidene difluoride (PVDF) membranes. The SARS-CoV-2 Spike proteins were visualized by a ChemiDoc> MP imaging system (Biorad) using anti-Spike S2 (Invitrogen at 1:1000 dilution) and anti-p24 Gag antibodies (NIH AIDS Reagents 1:1000 dilution).

All sequences were tested for potential recombination, as this would impact on evolutionary estimates. Potential recombination events were explored with nine algorithms (RDP, MaxChi, SisScan, GeneConv, Bootscan, PhylPro, Chimera, LARD and 3SEQ), implemented in RDP5 with default settings38. To corroborate any findings, ClonalFrameML v1.1239 was also used to infer recombination breakpoints. Neither programs indicated evidence of recombination in our data.

The Pymol Molecular Graphics System v2.4.0 (https://github.com/schrodinger/pymol-open-source/releases) was used to map the location of the four spike mutations of interested onto a SARS-CoV-2 spike structure visualised by Wrobel et al (PDB: 6ZGE)40.

The Anti-SARS-CoV-2 ELISA (IgG) assay used to test CP for antibody titres was Euroimmun Medizinische Labordiagnostika AG. This indirect ELISA based assay uses a recombinant structural spike 1 (S1) protein of SARS-CoV-2 expressed in the human cell line HEK 293 for the detection of SARS-CoV2 IgG.

Amino acid substitutions were introduced into the D614G pCDNA_SARS-CoV-2_Spike plasmid as previously described41 using the QuikChange Lightening Site-Directed Mutagenesis kit, following the manufacturer’s instructions (Agilent Technologies, Inc., Santa Clara, CA).

Viral vectors were prepared by transfection of 293T cells by using Fugene HD transfection reagent (Promega). 293T cells were transfected with a mixture of 11ul of Fugene HD, 1μg of pCDNAp19Spike-HA, 1ug of p8.91 HIV-1 gag-pol expression vector42,43, and 1.5μg of pCSFLW (expressing the firefly luciferase reporter gene with the HIV-1 packaging signal). Viral supernatant was collected at 48 and 72h after transfection, filtered through 0.45um filter and stored at -8°C. The 50% tissue culture infectious dose (TCID50) of SARS-CoV-2 pseudovirus was determined using Steady-Glo Luciferase assay system (Promega).

The reverse transcriptase activity of virus preparations was determined by qPCR using a SYBR Green-based product-enhanced PCR assay (SG-PERT) as previously described44. Briefly, 10-fold dilutions of virus supernatant were lysed in a 1:1 ratio in a 2x lysis solution (made up of 40% glycerol v/v 0.25% Trition X-100 v/v 100mM KCl, RNase inhibitor 0.8 U/ml, TrisHCL 100mM, buffered to pH7.4) for 10 minutes at room temperature.

12μl of each sample lysate was added to thirteen 13μl of a SYBR Green master mix (containing 0.5μM of MS2-RNA Fwd and Rev primers, 3.5pmol/ml of MS2-RNA, and 0.125U/μl of Ribolock RNAse inhibitor and cycled in a QuantStudio. Relative amounts of reverse transcriptase activity were determined as the rate of transcription of bacteriophage MS2 RNA, with absolute RT activity calculated by comparing the relative amounts of RT to an RT standard of known activity.

Spike pseudotype assays have been shown to have similar characteristics as neutralisation testing using fully infectious wild type SARS-CoV-28.Virus neutralisation assays were performed on 293T cell transiently transfected with ACE2 and TMPRSS2 using SARS-CoV-2 Spike pseudotyped virus expressing luciferase45. Pseudotyped virus was incubated with serial dilution of heat inactivated human serum samples or convalescent plasma in duplicate for 1h at 37°C. Virus and cell only controls were also included. Then, freshly trypsinized 293T ACE2/TMPRSS2 expressing cells were added to each well. Following 48h incubation in a 5% CO2 environment at 37°C, the luminescence was measured using Steady-Glo Luciferase assay system (Promega).

Virus neutralisation assays were performed on HeLa cells stably expressing ACE2 and using SARS-CoV-2 Spike pseudotyped virus expressing luciferase as previously described46. Pseudotyped virus was incubated with serial dilution of purified mAbs9 in duplicate for 1h at 37°C. Then, freshly trypsinized HeLa ACE2-expressing cells were added to each well. Following 48h incubation in a 5% CO2 environment at 37°C, the luminescence was measured using Bright-Glo Luciferase assay system (Promega) and neutralization calculated relative to virus only controls. IC50 values were calculated in GraphPad Prism.


Clinical time line of events with longitudinal respiratory sample CT values. CT – cycle threshold.


A. Blood parameters over time in patient case: White cell count (WCC) and lymphocyte counts are expressed as x103 Cells/mm3. CRP: C reactive protein. B. Assessment of T cell and innate function. Whole blood cytokines were measured in whole blood after 24 hours stimulation either after T-cell stimulation with PHA or anti CD3/IL2 or innate stimulation with LPS. Healthy controls are shown as grey circles (N=15), Patient at d71 and d98 is shown as blue circles or red circles respectively. Cytokine levels are shown as pg/ml stimulation. Mean is shown by line and whiskers representing standard deviation.


A. Serum SARS-CoV-2 antibody levels and virus population changes in chronic SARS-CoV-2 infection. Anti SARS-CoV2 IgG antibodies in patient and pre/post convalescent plasma compared to RNA+ Covid19 patients and prepandemic healthy controls: Red, grey and gold: IgG antibodies to SARS-CoV2 nucleocapsid protein (N), trimeric S protein (S) and the receptor binding domain (RBD) were measured by multiplexed particle based flow cytometry (Luminex) in RNA+ COVID-19 patients (N=20, red dots), Pre-pandemic healthy controls (N=20, grey dots) and in the convalescent donor plasma (orange dots); Results are shown as mean fluorescent intensity (MFI) +/- SD. Patient sera over time in blue: Anti SARS-CoV2 IgG to N (blue squares), S (blue circles) and RBD (blue triangles). Timing of CP units is also shown. B. SARS-CoV-2 antibody titres in patient and in convalescent plasma. Measurement of SARS-CoV-2 specific IgG antibody titres in three units of convalescent plasma (CP) by Euroimmun assay.

Concordance was generally good between the majority of timepoints, however due to large discrepancies in a number of timepoints, we suggest that due to the high base calling error rate, Nanopore is not yet suitable for calling minority variants. As such, all figures in the main paper were produced using Illumina data only. B. Single genome sequencing (SGS) data from respiratory samples at indicated days. Indicated are the number of single genomes obtained at each time point with the mutations of interest (identified by deep sequencing). *denominator is 19 as for 2 samples the primer reads were poor quality at amino acid 796 at day 98. Amino acid variant and corresponding nucleotide position: S:W64G = 21752, S:Δ69 = 21765-21770, S:Y200H = 22160, S:T240I = 22281, S:P330S = 22550, S:D795H = 23948


A. Pairwise distances between samples measured using the all-locus distance metric plotted against pairwise distances in time (measured in days) between samples being collected. Internal distances between samples in the proposed main clade are shown in black, distances between samples in the main clade and samples collected on days 93 and 95 are shown in red, and internal distances between samples collected on days 93 and 95 are shown in green. B. Pairwise distances between samples in the larger clade (black) and between these samples and those collected on days 93 and 95 (red). The median values of the distributions of these values are significantly different according to a Mann Whitney test. C. Pairwise distances between samples in the main clade, once those collected on days 86, 89, 93, 95 have been removed (black) and between these samples and those collected on days 86 and 89 (red). The median values of the distributions of these values are not significantly different at the 5 level according to a Mann Whitney test.


A. Close-view maximum-likelihood phylogenetic tree indicating the diversity of the case patient and three other long-term shedders from the local area (red, blue and purple), compared to recently published sequences from Choi et al (orange) and Avanzato et al (gold). Control patients generally showed limited diversity temporally, though the Choi et al sequences were highly divergent. Environmental samples (patient’s call bell, and patient’s mobile phone) are indicated. Tree branched have been collapsed where bootstrap support was <60.


B. Highlighter plot indicating nucleotide changes at consensus level in sequential respiratory samples compared to the consensus sequence at first diagnosis of COVID-19. Each row indicates the timepoint the sample was collected (number of days from first positive SARS-CoV-2 RT-PCR). Black dashed lines indicate the RNA-dependent RNA polymerase (RdRp) and Spike regions of the genome. There were few nucleotide substitutions between days 1-54, despite the patient receiving two courses of remdesivir. The first major changes in the spike genome occurred on day 82, following convalescent plasma given on days 63 and 65. The amino acid deletion in S1, ΔH69/V70 is indicated by the black lines. Sites: Endotracheal aspirate (ETA) or Nose/throat swabs (N+T).


A. infection of target 293T cells expressing TMPRSS2 and ACE2 receptors using equal amounts of virus as determined by reverse transcriptase activity. Data points represent technical replicates (n=2), with mean shown with error bars representing standard deviation. Data are representative of n=2 independent experiments (n=2). B. Representative Inverse dilution plots for Spike variants against convalescent plasma units 1-3. Data points represent mean neutralisation of technical replicates and error bars represent standard error of the mean of replicates. Data are representative of two independent experiments (n=2).


A. Neutralization potency of a panel of monoclonal antibodies targeting the RBD is not impacted by Spike mutations D796H or ΔH69/V70. Lentivirus pseudotyped with SARS-CoV-2 Spike protein: WT (D614G background), D796H, ΔH69/V70, D796H+ΔH69/V70 were produced in 293T cells and used to infect target Hela cells stably expressing ACE2 in the presence of serial dilutions of indicated monoclonal antibodies. Data are means of technical replicates with error bars representing SD. Data are representative of at least two independent experiments. RBD: receptor binding domain. B. Classes of RBD binding antibodies and fold changes for Spike mutations D796H or ΔH69/V70 are indicated based Bouwer et al. Clusters II, V contain only non-neutralising mAbs, smaller neutralising mAb clusters IV (n=2) and X (n=1) were not tested. Red indicates significant fold changes.


A. The SARS-CoV-2 spike trimer (PDB ID: 6xr8) with two protomers represented as surfaces and one protomer represented as a ribbon. The NTD is coloured in light blue, the RBD in light pink, the fusion peptide in dark pink, the HR1 domain in yellow, the CH domain in pale green, and the CD domain in brown. The location of D796 and H69 are indicated by red spheres. The loop connecting D796 to the fusion peptide is coloured magenta to improve visibility. The double grey lines provide orientation relative to the membrane. B. A close-up of the region defined by the box around H69 in panel A. H69 is highlighted in yellow. Residues containing atoms that are within 6 Å of H69 are highlighted in cyan. C. A close-up of the region defined by the box around D796 in panel A. D796 is highlighted in yellow. Residues containing atoms that are within 6 Å of D796 are highlighted in cyan. Hydrogen bonds are indicated by dashed yellow lines. Hydrophobic residues in the vicinity of D796 have been labelled. Y707 is from the neighbouring protomer. D. Global prevalence of selected spike mutations detailed in this paper. All high coverage sequences were downloaded from the GISAID database on 6th January and aligned using MAFFT; as of this date there were 298254 sequences available. The global prevalence of each of the six spike mutations W64G, ΔH69/V70, Y200H, T240I, P330S and D796H were assessed by viewing the multiple sequence alignment in AliView, sorting by the column of interest, and counting the number of mutations.